{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8270eb45",
   "metadata": {},
   "source": [
    "A Python script for searching through all gff files in a directory. The script searches through a given gff file, notes the count of a given gene in a pathway, and then outputs that number into the correct cell associated with that gene and pathway in the excel spreadsheet."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aa0ed9",
   "metadata": {},
   "source": [
    "How it works:\n",
    "A. Flip the csv vertical genes to be names of columns. \n",
    "B. Write all species ids to the given csv (should be names of rows). \n",
    "C. Create a list of all gene names to search for called to_search.txt\n",
    "\n",
    "1. Search through all files in a directory and find all gff files(even if they are in a sub-folder)\n",
    "- unzip the file in the same folder\n",
    "- write the path of the file to a list\n",
    "- trim path name to part that is necessary for operating \n",
    "\n",
    "2. for each file in the list, perform the following:\n",
    "- for each line in to_search.txt, find number of lines in which that word shows up at least once (counts of lines, not words!)\n",
    "- create a dataframe with the columns as gene names from to_search.txt \n",
    "- add row to dataframe for each species searched, that row has all the counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a84427e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/gracedai/Documents/MethanogeneDB\n"
     ]
    }
   ],
   "source": [
    "#Check that we are in the correct working directory\n",
    "import os \n",
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d4ad666f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving notices: ...working... done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.3.1\n",
      "  latest version: 23.5.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.5.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - sh\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    certifi-2023.5.7           |  py310hca03da5_0         153 KB\n",
      "    openssl-1.1.1u             |       h1a28f6b_0         2.6 MB\n",
      "    sh-1.14.2                  |  py310hca03da5_0          67 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         2.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  sh                 pkgs/main/osx-arm64::sh-1.14.2-py310hca03da5_0 \n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    anaconda::ca-certificates-2022.4.26-h~ --> pkgs/main::ca-certificates-2023.05.30-hca03da5_0 \n",
      "  certifi            anaconda::certifi-2022.6.15-py310hca0~ --> pkgs/main::certifi-2023.5.7-py310hca03da5_0 \n",
      "  openssl                                 1.1.1t-h1a28f6b_0 --> 1.1.1u-h1a28f6b_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "certifi-2023.5.7     | 153 KB    |                                       |   0% \n",
      "openssl-1.1.1u       | 2.6 MB    |                                       |   0% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     |                                       |   0% \u001b[A\u001b[A\n",
      "certifi-2023.5.7     | 153 KB    | ##################################### | 100% \u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     | ########8                             |  24% \u001b[A\u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: \\ WARNING conda.core.path_actions:verify(1093): Unable to create environments file. Path not writable.\n",
      "  environment location: /Users/gracedai/.conda/environments.txt\n",
      "\n",
      "done\n",
      "Executing transaction: / WARNING conda.core.envs_manager:register_env(49): Unable to register environment. Path not writable or missing.\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "  registry file: /Users/gracedai/.conda/environments.txt\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# Install a conda package in the current Jupyter kernel\n",
    "import sys\n",
    "!conda install --yes --prefix {sys.prefix} sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "df450b61",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/gracedai/Documents/MethanogeneDB\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0c2c402a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#unzips a file name, given the path, and deletes the zipped version\n",
    "from sh import gunzip\n",
    "#this line is for testing purposes\n",
    "#gunzip('GCF_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "01379dfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#unzip whatever files needed\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gz\"):\n",
    "            #os.remove(os.path.join(root, file))\n",
    "            gunzip(os.path.join(root, file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f642bd3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/gracedai/Documents/MethanogeneDB/GCA_000970225.1/GCF_000970225.1_ASM97022v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000092305.1/GCF_000092305.1_ASM9230v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970165.1/GCF_000970165.1_ASM97016v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170265.1/GCF_024170265.1_ASM2417026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502925.1/GCA_002502925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000091665.1/GCF_000091665.1_ASM9166v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_020885915.1/GCF_020885915.1_ASM2088591v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017535165.1/GCA_017535165.1_ASM1753516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000166095.1/GCF_000166095.1_ASM16609v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412935.1/GCA_017412935.1_ASM1741293v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508885.1/GCA_002508885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633965.1/GCF_029633965.1_ASM2963396v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114585.1/GCF_900114585.1_IMG-taxon_2593339150_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744315.1/GCF_000744315.1_ASM74431v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_028743435.1/GCF_028743435.1_ASM2874343v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002356395.1/GCA_002356395.1_ASM235639v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168575.1/GCA_030168575.1_ASM3016857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970325.1/GCF_000970325.1_ASM97032v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502965.1/GCA_002502965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970265.1/GCF_000970265.1_ASM97026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025525.1/GCF_000025525.1_ASM2552v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001729965.1/GCF_001729965.1_ASM172996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900215215.1/GCF_900215215.1_IMG-taxon_2703719067_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902762095.1/GCA_902762095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000746075.1/GCF_000746075.1_ASM74607v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000196655.1/GCF_000196655.1_ASM19665v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013725.1/GCF_000013725.1_ASM1372v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_946463545.1/GCF_946463545.1_Mth-DSM-2095-CCS-WG_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003111625.1/GCF_003111625.1_ASM311162v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002252585.1/GCF_002252585.1_ASM225258v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017935945.1/GCA_017935945.1_ASM1793594v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_021184045.1/GCF_021184045.1_ASM2118404v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017539065.1/GCA_017539065.1_ASM1753906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503105.1/GCA_002503105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000217995.1/GCF_000217995.1_ASM21799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017447225.1/GCA_017447225.1_ASM1744722v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000251105.1/GCF_000251105.1_ASM25110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711905.1/GCF_000711905.1_ASM71190v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011005.1/GCF_000011005.1_ASM1100v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024167805.1/GCA_024167805.1_ASM2416780v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495885.1/GCA_002495885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024808815.1/GCF_024808815.1_ASM2480881v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017540705.1/GCA_017540705.1_ASM1754070v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025397995.1/GCF_025397995.1_ASM2539799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000621965.1/GCF_000621965.1_ASM62196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813675.1/GCF_002813675.1_ASM281367v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017432245.1/GCA_017432245.1_ASM1743224v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468935.1/GCA_017468935.1_ASM1746893v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873415.1/GCF_017873415.1_ASM1787341v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744455.1/GCF_000744455.1_ASM74445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969925.1/GCF_000969925.1_ASM96992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000327485.1/GCF_000327485.1_ASM32748v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902763685.1/GCA_902763685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496295.1/GCA_002496295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900111645.1/GCF_900111645.1_IMG-taxon_2642422537_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024625.1/GCF_000024625.1_ASM2462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900109595.1/GCF_900109595.1_IMG-taxon_2654588138_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508925.1/GCA_002508925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584625.1/GCA_003584625.1_ASM358462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017504565.1/GCA_017504565.1_ASM1750456v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026016205.1/GCA_026016205.1_ASM2601620v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002287175.1/GCF_002287175.1_ASM228717v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151225.1/GCA_000151225.1_ASM15122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902795935.1/GCA_902795935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875315.1/GCF_017875315.1_ASM1787531v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555045.1/GCF_027555045.1_ASM2755504v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874455.1/GCF_017874455.1_ASM1787445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563305.1/GCA_001563305.1_ASM156330v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506325.1/GCA_002506325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508825.1/GCA_002508825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017399555.1/GCA_017399555.1_ASM1739955v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000306725.1/GCA_000306725.1_ASM30672v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506905.1/GCA_002506905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827225.1/GCF_902827225.1_Methanocaldococcus_lauensis_SG7_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214725.1/GCF_000214725.1_ASM21472v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009914355.1/GCF_009914355.1_ASM991435v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491285.1/GCA_003491285.1_ASM349128v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017512765.1/GCA_017512765.1_ASM1751276v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001481295.1/GCF_001481295.1_ASM148129v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509405.1/GCA_002509405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015765.1/GCF_000015765.1_ASM1576v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000300255.2/GCF_000300255.2_ASM30025v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171695.1/GCA_905171695.1_GoM-Arc1_E37_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000014945.1/GCF_000014945.1_ASM1494v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496945.1/GCA_002496945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970185.1/GCF_000970185.1_ASM97018v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000220645.1/GCF_000220645.1_ASM22064v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168595.1/GCA_030168595.1_ASM3016859v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002973515.1/GCF_002973515.1_ASM297351v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001560915.1/GCA_001560915.1_ASM156091v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000179575.2/GCF_000179575.2_ASM17957v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509645.1/GCA_002509645.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001889405.1/GCF_001889405.1_ASM188940v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017549625.1/GCA_017549625.1_ASM1754962v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029854155.1/GCF_029854155.1_ASM2985415v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000235565.1/GCF_000235565.1_ASM23556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813695.1/GCF_002813695.1_ASM281369v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247815.1/GCA_030247815.1_ASM3024781v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003052125.1/GCF_003052125.1_ASM305212v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171685.1/GCA_905171685.1_GoM-Arc1_E50_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900177455.1/GCF_900177455.1_IMG-taxon_2706794887_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828535.1/GCF_000828535.1_ASM82853v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017461225.1/GCA_017461225.1_ASM1746122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017524805.1/GCA_017524805.1_ASM1752480v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017508065.1/GCA_017508065.1_ASM1750806v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017408885.1/GCA_017408885.1_ASM1740888v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018141185.1/GCF_018141185.1_ASM1814118v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509655.1/GCA_002509655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023016325.1/GCF_023016325.1_ASM2301632v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000021965.1/GCF_000021965.1_ASM2196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505245.1/GCA_002505245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009917665.1/GCF_009917665.1_ASM991766v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404225.1/GCF_000404225.1_ASM40422v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508935.1/GCA_002508935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813085.1/GCF_002813085.1_ASM281308v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969885.1/GCF_000969885.1_ASM96988v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828575.1/GCF_000828575.1_ASM82857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498705.1/GCA_002498705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095295.1/GCF_900095295.1_MCBB_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900100715.1/GCF_900100715.1_IMG-taxon_2642422561_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434555.1/GCA_017434555.1_ASM1743455v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498605.1/GCA_002498605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900101055.1/GCF_900101055.1_IMG-taxon_2634166298_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026625165.1/GCA_026625165.1_ASM2662516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002504405.1/GCA_002504405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506565.1/GCA_002506565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505345.1/GCA_002505345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_016907005.1/GCF_016907005.1_ASM1690700v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114835.1/GCF_900114835.1_IMG-taxon_2642422540_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900116525.1/GCF_900116525.1_IMG-taxon_2595698250_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002945325.1/GCF_002945325.1_ASM294532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509425.1/GCA_002509425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011585.1/GCF_000011585.1_ASM1158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902782945.1/GCA_902782945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017537385.1/GCA_017537385.1_ASM1753738v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023167465.1/GCF_023167465.1_ASM2316746v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017625.1/GCF_000017625.1_ASM1762v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494805.1/GCA_002494805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669925.1/GCF_019669925.1_ASM1966992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017622775.1/GCA_017622775.1_ASM1762277v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019263745.1/GCF_019263745.1_ASM1926374v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902775005.1/GCA_902775005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498365.1/GCA_002498365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434525.1/GCA_017434525.1_ASM1743452v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151245.1/GCA_000151245.1_ASM15124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508845.1/GCA_002508845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505275.1/GCA_002505275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508905.1/GCA_002508905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000328665.1/GCF_000328665.1_ASM32866v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553445.1/GCA_017553445.1_ASM1755344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669965.1/GCF_019669965.1_ASM1966996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498805.1/GCA_002498805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503785.1/GCA_002503785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016125.1/GCF_000016125.1_ASM1612v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875815.1/GCF_017875815.1_ASM1787581v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584605.1/GCA_003584605.1_ASM358460v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019262145.1/GCA_019262145.1_ASM1926214v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900103415.1/GCF_900103415.1_IMG-taxon_2593339167_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875625.1/GCF_017875625.1_ASM1787562v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875565.1/GCF_017875565.1_ASM1787556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555065.1/GCF_027555065.1_ASM2755506v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017357525.1/GCF_017357525.1_ASM1735752v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000350305.1/GCA_000350305.1_ASM35030v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827205.1/GCA_902827205.1_Methanocaldococcus_lauensis_SG1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247745.1/GCA_030247745.1_ASM3024774v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508625.1/GCA_002508625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468685.1/GCA_017468685.1_ASM1746868v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828555.1/GCF_000828555.1_ASM82855v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000275865.1/GCA_000275865.1_ASM27586v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015825.1/GCF_000015825.1_ASM1582v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027554905.1/GCF_027554905.1_ASM2755490v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875325.1/GCF_017875325.1_ASM1787532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024662215.1/GCF_024662215.1_ASM2466221v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016525.1/GCF_000016525.1_ASM1652v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026684035.1/GCF_026684035.1_ASM2668403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553245.1/GCA_017553245.1_ASM1755324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000063445.1/GCF_000063445.1_ASM6344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498765.1/GCA_002498765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874375.1/GCF_017874375.1_ASM1787437v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495325.1/GCA_002495325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000739065.1/GCF_000739065.1_ASM73906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498425.1/GCA_002498425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846155.1/GCF_022846155.1_ASM2284615v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017561385.1/GCA_017561385.1_ASM1756138v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498375.1/GCA_002498375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007185.1/GCF_000007185.1_ASM718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017165.1/GCF_000017165.1_ASM1716v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003711245.1/GCF_003711245.1_ASM371124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000147875.1/GCF_000147875.1_ASM14787v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000745485.1/GCF_000745485.1_ASM74548v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508855.1/GCA_002508855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214415.1/GCF_000214415.1_ASM21441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013388255.1/GCF_013388255.1_ASM1338825v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000376965.1/GCF_000376965.1_ASM37696v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902383905.1/GCF_902383905.1_UHGG_MGYG-HGUT-02160_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495685.1/GCA_002495685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503135.1/GCA_002503135.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970245.1/GCF_000970245.1_ASM97024v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000304355.2/GCF_000304355.2_Mb_MS2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013445.1/GCF_000013445.1_ASM1344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017411925.1/GCA_017411925.1_ASM1741192v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873625.1/GCF_017873625.1_ASM1787362v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711215.1/GCF_000711215.1_ASM71121v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491305.1/GCA_003491305.1_ASM349130v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000320505.1/GCF_000320505.1_ASM32050v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017407825.1/GCA_017407825.1_ASM1740782v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970005.1/GCF_000970005.1_ASM97000v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761105.1/GCF_013761105.1_ASM1376110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970205.1/GCF_000970205.1_ASM97020v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_004363215.1/GCF_004363215.1_ASM436321v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024807035.1/GCF_024807035.1_ASM2480703v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017519045.1/GCA_017519045.1_ASM1751904v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761045.1/GCF_013761045.1_ASM1376104v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873855.1/GCF_017873855.1_ASM1787385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001304615.2/GCF_001304615.2_ASM130461v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412625.1/GCA_017412625.1_ASM1741262v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023617305.1/GCF_023617305.1_ASM2361730v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970045.1/GCF_000970045.1_ASM97004v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009911715.1/GCA_009911715.1_ASM991171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025449255.1/GCF_025449255.1_ASM2544925v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875395.1/GCF_017875395.1_ASM1787539v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025914035.1/GCF_025914035.1_ASM2591403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003385755.1/GCF_003385755.1_ASM338575v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017633855.1/GCA_017633855.1_ASM1763385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000191585.1/GCF_000191585.1_ASM19158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761055.1/GCF_013761055.1_ASM1376105v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007065.1/GCF_000007065.1_ASM706v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498785.1/GCA_002498785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900079125.1/GCA_900079125.1_MBBA_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813655.1/GCF_002813655.1_ASM281365v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017940815.1/GCA_017940815.1_ASM1794081v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017387505.1/GCA_017387505.1_ASM1738750v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000204415.1/GCF_000204415.1_ASM20441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506995.1/GCA_002506995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000800805.1/GCF_000800805.1_ASM80080v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003351865.1/GCF_003351865.1_ASM335186v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017410345.1/GCA_017410345.1_ASM1741034v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494585.1/GCA_002494585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017185.1/GCF_000017185.1_ASM1718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017473945.1/GCA_017473945.1_ASM1747394v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502465.1/GCA_002502465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170505.1/GCF_024170505.1_ASM2417050v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017409525.1/GCA_017409525.1_ASM1740952v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001458655.1/GCF_001458655.1_Mb9_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017406825.1/GCA_017406825.1_ASM1740682v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000008645.1/GCF_000008645.1_ASM864v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000145295.1/GCF_000145295.1_ASM14529v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900036045.1/GCF_900036045.1_Methanoculleus_sp_MAB1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900106905.1/GCF_900106905.1_IMG-taxon_2642422606_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000023985.1/GCF_000023985.1_ASM2398v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014203295.1/GCF_014203295.1_ASM1420329v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000341715.1/GCF_000341715.1_ASM34171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018502485.1/GCF_018502485.1_ASM1850248v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633915.1/GCF_029633915.1_ASM2963391v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_008033705.1/GCF_008033705.1_ASM803370v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014207315.1/GCF_014207315.1_ASM1420731v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902787415.1/GCA_902787415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025865.1/GCF_000025865.1_ASM2586v1_genomic.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "275"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# list to store paths to gff files\n",
    "gff_files = []\n",
    "# iterate through current working directory and for each file \n",
    "# unzip the file and append the path to a list \n",
    "#run this twice ?\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gff.gz\"):\n",
    "            gunzip(os.path.join(root, file))\n",
    "        if file.endswith(\".gff\"):\n",
    "            gff_files.append(os.path.join(root, file))\n",
    "print(gff_files)\n",
    "len(gff_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a4818c09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCA_000970225.1/GCF_000970225.1_ASM97022v1_genomic.gff', 'GCA_000092305.1/GCF_000092305.1_ASM9230v1_genomic.gff', 'GCA_000970165.1/GCF_000970165.1_ASM97016v1_genomic.gff', 'GCA_024170265.1/GCF_024170265.1_ASM2417026v1_genomic.gff', 'GCA_002502925.1/GCA_002502925.1.gff', 'GCA_000091665.1/GCF_000091665.1_ASM9166v1_genomic.gff', 'GCA_020885915.1/GCF_020885915.1_ASM2088591v1_genomic.gff', 'GCA_017535165.1/GCA_017535165.1_ASM1753516v1_genomic.gff', 'GCA_000166095.1/GCF_000166095.1_ASM16609v1_genomic.gff', 'GCA_017412935.1/GCA_017412935.1_ASM1741293v1_genomic.gff', 'GCA_002508885.1/GCA_002508885.1.gff', 'GCA_029633965.1/GCF_029633965.1_ASM2963396v1_genomic.gff', 'GCA_900114585.1/GCF_900114585.1_IMG-taxon_2593339150_annotated_assembly_genomic.gff', 'GCA_000744315.1/GCF_000744315.1_ASM74431v1_genomic.gff', 'GCA_028743435.1/GCF_028743435.1_ASM2874343v1_genomic.gff', 'GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', 'GCA_002356395.1/GCA_002356395.1_ASM235639v1_genomic.gff', 'GCA_030168575.1/GCA_030168575.1_ASM3016857v1_genomic.gff', 'GCA_000970325.1/GCF_000970325.1_ASM97032v1_genomic.gff', 'GCA_002502965.1/GCA_002502965.1.gff', 'GCA_000970265.1/GCF_000970265.1_ASM97026v1_genomic.gff', 'GCA_000025525.1/GCF_000025525.1_ASM2552v1_genomic.gff', 'GCA_001729965.1/GCF_001729965.1_ASM172996v1_genomic.gff', 'GCA_900215215.1/GCF_900215215.1_IMG-taxon_2703719067_annotated_assembly_genomic.gff', 'GCA_902762095.1/GCA_902762095.1.gff', 'GCA_000746075.1/GCF_000746075.1_ASM74607v1_genomic.gff', 'GCA_000196655.1/GCF_000196655.1_ASM19665v1_genomic.gff', 'GCA_000013725.1/GCF_000013725.1_ASM1372v1_genomic.gff', 'GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff', 'GCA_946463545.1/GCF_946463545.1_Mth-DSM-2095-CCS-WG_genomic.gff', 'GCA_003111625.1/GCF_003111625.1_ASM311162v1_genomic.gff', 'GCA_002252585.1/GCF_002252585.1_ASM225258v1_genomic.gff', 'GCA_017935945.1/GCA_017935945.1_ASM1793594v1_genomic.gff', 'GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff', 'GCA_021184045.1/GCF_021184045.1_ASM2118404v1_genomic.gff', 'GCA_017539065.1/GCA_017539065.1_ASM1753906v1_genomic.gff', 'GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff', 'GCA_002503105.1/GCA_002503105.1.gff', 'GCA_000217995.1/GCF_000217995.1_ASM21799v1_genomic.gff', 'GCA_017447225.1/GCA_017447225.1_ASM1744722v1_genomic.gff', 'GCA_000251105.1/GCF_000251105.1_ASM25110v1_genomic.gff', 'GCA_000711905.1/GCF_000711905.1_ASM71190v1_genomic.gff', 'GCA_000011005.1/GCF_000011005.1_ASM1100v1_genomic.gff', 'GCA_024167805.1/GCA_024167805.1_ASM2416780v1_genomic.gff', 'GCA_002495885.1/GCA_002495885.1.gff', 'GCA_024808815.1/GCF_024808815.1_ASM2480881v1_genomic.gff', 'GCA_017540705.1/GCA_017540705.1_ASM1754070v1_genomic.gff', 'GCA_025397995.1/GCF_025397995.1_ASM2539799v1_genomic.gff', 'GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff', 'GCA_000621965.1/GCF_000621965.1_ASM62196v1_genomic.gff', 'GCA_002813675.1/GCF_002813675.1_ASM281367v1_genomic.gff', 'GCA_017432245.1/GCA_017432245.1_ASM1743224v1_genomic.gff', 'GCA_017468935.1/GCA_017468935.1_ASM1746893v1_genomic.gff', 'GCA_017873415.1/GCF_017873415.1_ASM1787341v1_genomic.gff', 'GCA_000744455.1/GCF_000744455.1_ASM74445v1_genomic.gff', 'GCA_000969925.1/GCF_000969925.1_ASM96992v1_genomic.gff', 'GCA_000327485.1/GCF_000327485.1_ASM32748v1_genomic.gff', 'GCA_902763685.1/GCA_902763685.1.gff', 'GCA_002496295.1/GCA_002496295.1.gff', 'GCA_900111645.1/GCF_900111645.1_IMG-taxon_2642422537_annotated_assembly_genomic.gff', 'GCA_000024625.1/GCF_000024625.1_ASM2462v1_genomic.gff', 'GCA_900109595.1/GCF_900109595.1_IMG-taxon_2654588138_annotated_assembly_genomic.gff', 'GCA_002508925.1/GCA_002508925.1.gff', 'GCA_003584625.1/GCA_003584625.1_ASM358462v1_genomic.gff', 'GCA_017504565.1/GCA_017504565.1_ASM1750456v1_genomic.gff', 'GCA_026016205.1/GCA_026016205.1_ASM2601620v1_genomic.gff', 'GCA_002287175.1/GCF_002287175.1_ASM228717v1_genomic.gff', 'GCA_000151225.1/GCA_000151225.1_ASM15122v1_genomic.gff', 'GCA_902795935.1/GCA_902795935.1.gff', 'GCA_017875315.1/GCF_017875315.1_ASM1787531v1_genomic.gff', 'GCA_027555045.1/GCF_027555045.1_ASM2755504v1_genomic.gff', 'GCA_017874455.1/GCF_017874455.1_ASM1787445v1_genomic.gff', 'GCA_001563305.1/GCA_001563305.1_ASM156330v1_genomic.gff', 'GCA_002506325.1/GCA_002506325.1.gff', 'GCA_002508825.1/GCA_002508825.1.gff', 'GCA_017399555.1/GCA_017399555.1_ASM1739955v1_genomic.gff', 'GCA_000306725.1/GCA_000306725.1_ASM30672v1_genomic.gff', 'GCA_002506905.1/GCA_002506905.1.gff', 'GCA_902827225.1/GCF_902827225.1_Methanocaldococcus_lauensis_SG7_genomic.gff', 'GCA_000214725.1/GCF_000214725.1_ASM21472v1_genomic.gff', 'GCA_009914355.1/GCF_009914355.1_ASM991435v1_genomic.gff', 'GCA_003491285.1/GCA_003491285.1_ASM349128v1_genomic.gff', 'GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', 'GCA_017512765.1/GCA_017512765.1_ASM1751276v1_genomic.gff', 'GCA_001481295.1/GCF_001481295.1_ASM148129v1_genomic.gff', 'GCA_002509405.1/GCA_002509405.1.gff', 'GCA_000015765.1/GCF_000015765.1_ASM1576v1_genomic.gff', 'GCA_000300255.2/GCF_000300255.2_ASM30025v2_genomic.gff', 'GCA_905171695.1/GCA_905171695.1_GoM-Arc1_E37_genomic.gff', 'GCA_000014945.1/GCF_000014945.1_ASM1494v1_genomic.gff', 'GCA_002496945.1/GCA_002496945.1.gff', 'GCA_000970185.1/GCF_000970185.1_ASM97018v1_genomic.gff', 'GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff', 'GCA_000220645.1/GCF_000220645.1_ASM22064v1_genomic.gff', 'GCA_030168595.1/GCA_030168595.1_ASM3016859v1_genomic.gff', 'GCA_002973515.1/GCF_002973515.1_ASM297351v1_genomic.gff', 'GCA_001560915.1/GCA_001560915.1_ASM156091v1_genomic.gff', 'GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff', 'GCA_000179575.2/GCF_000179575.2_ASM17957v2_genomic.gff', 'GCA_002509645.1/GCA_002509645.1.gff', 'GCA_001889405.1/GCF_001889405.1_ASM188940v1_genomic.gff', 'GCA_017549625.1/GCA_017549625.1_ASM1754962v1_genomic.gff', 'GCA_029854155.1/GCF_029854155.1_ASM2985415v1_genomic.gff', 'GCA_000235565.1/GCF_000235565.1_ASM23556v1_genomic.gff', 'GCA_002813695.1/GCF_002813695.1_ASM281369v1_genomic.gff', 'GCA_030247815.1/GCA_030247815.1_ASM3024781v1_genomic.gff', 'GCA_003052125.1/GCF_003052125.1_ASM305212v1_genomic.gff', 'GCA_905171685.1/GCA_905171685.1_GoM-Arc1_E50_genomic.gff', 'GCA_900177455.1/GCF_900177455.1_IMG-taxon_2706794887_annotated_assembly_genomic.gff', 'GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff', 'GCA_000828535.1/GCF_000828535.1_ASM82853v1_genomic.gff', 'GCA_017461225.1/GCA_017461225.1_ASM1746122v1_genomic.gff', 'GCA_017524805.1/GCA_017524805.1_ASM1752480v1_genomic.gff', 'GCA_017508065.1/GCA_017508065.1_ASM1750806v1_genomic.gff', 'GCA_017408885.1/GCA_017408885.1_ASM1740888v1_genomic.gff', 'GCA_018141185.1/GCF_018141185.1_ASM1814118v1_genomic.gff', 'GCA_002509655.1/GCA_002509655.1.gff', 'GCA_023016325.1/GCF_023016325.1_ASM2301632v1_genomic.gff', 'GCA_000021965.1/GCF_000021965.1_ASM2196v1_genomic.gff', 'GCA_002505245.1/GCA_002505245.1.gff', 'GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff', 'GCA_009917665.1/GCF_009917665.1_ASM991766v1_genomic.gff', 'GCA_000404225.1/GCF_000404225.1_ASM40422v1_genomic.gff', 'GCA_002508935.1/GCA_002508935.1.gff', 'GCA_002813085.1/GCF_002813085.1_ASM281308v1_genomic.gff', 'GCA_000969885.1/GCF_000969885.1_ASM96988v1_genomic.gff', 'GCA_000828575.1/GCF_000828575.1_ASM82857v1_genomic.gff', 'GCA_002498705.1/GCA_002498705.1.gff', 'GCA_900095295.1/GCF_900095295.1_MCBB_genomic.gff', 'GCA_900100715.1/GCF_900100715.1_IMG-taxon_2642422561_annotated_assembly_genomic.gff', 'GCA_017434555.1/GCA_017434555.1_ASM1743455v1_genomic.gff', 'GCA_002498605.1/GCA_002498605.1.gff', 'GCA_900101055.1/GCF_900101055.1_IMG-taxon_2634166298_annotated_assembly_genomic.gff', 'GCA_026625165.1/GCA_026625165.1_ASM2662516v1_genomic.gff', 'GCA_002504405.1/GCA_002504405.1.gff', 'GCA_002506565.1/GCA_002506565.1.gff', 'GCA_002505345.1/GCA_002505345.1.gff', 'GCA_016907005.1/GCF_016907005.1_ASM1690700v1_genomic.gff', 'GCA_900114835.1/GCF_900114835.1_IMG-taxon_2642422540_annotated_assembly_genomic.gff', 'GCA_900116525.1/GCF_900116525.1_IMG-taxon_2595698250_annotated_assembly_genomic.gff', 'GCA_002945325.1/GCF_002945325.1_ASM294532v1_genomic.gff', 'GCA_002509425.1/GCA_002509425.1.gff', 'GCA_000011585.1/GCF_000011585.1_ASM1158v1_genomic.gff', 'GCA_902782945.1/GCA_902782945.1.gff', 'GCA_017537385.1/GCA_017537385.1_ASM1753738v1_genomic.gff', 'GCA_023167465.1/GCF_023167465.1_ASM2316746v1_genomic.gff', 'GCA_000017625.1/GCF_000017625.1_ASM1762v1_genomic.gff', 'GCA_002494805.1/GCA_002494805.1.gff', 'GCA_019669925.1/GCF_019669925.1_ASM1966992v1_genomic.gff', 'GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff', 'GCA_017622775.1/GCA_017622775.1_ASM1762277v1_genomic.gff', 'GCA_019263745.1/GCF_019263745.1_ASM1926374v1_genomic.gff', 'GCA_902775005.1/GCA_902775005.1.gff', 'GCA_002498365.1/GCA_002498365.1.gff', 'GCA_017434525.1/GCA_017434525.1_ASM1743452v1_genomic.gff', 'GCA_000151245.1/GCA_000151245.1_ASM15124v1_genomic.gff', 'GCA_002508845.1/GCA_002508845.1.gff', 'GCA_002505275.1/GCA_002505275.1.gff', 'GCA_002508905.1/GCA_002508905.1.gff', 'GCA_000328665.1/GCF_000328665.1_ASM32866v1_genomic.gff', 'GCA_017553445.1/GCA_017553445.1_ASM1755344v1_genomic.gff', 'GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff', 'GCA_019669965.1/GCF_019669965.1_ASM1966996v1_genomic.gff', 'GCA_002498805.1/GCA_002498805.1.gff', 'GCA_002503785.1/GCA_002503785.1.gff', 'GCA_000016125.1/GCF_000016125.1_ASM1612v1_genomic.gff', 'GCA_017875815.1/GCF_017875815.1_ASM1787581v1_genomic.gff', 'GCA_003584605.1/GCA_003584605.1_ASM358460v1_genomic.gff', 'GCA_019262145.1/GCA_019262145.1_ASM1926214v1_genomic.gff', 'GCA_900103415.1/GCF_900103415.1_IMG-taxon_2593339167_annotated_assembly_genomic.gff', 'GCA_017875625.1/GCF_017875625.1_ASM1787562v1_genomic.gff', 'GCA_017875565.1/GCF_017875565.1_ASM1787556v1_genomic.gff', 'GCA_027555065.1/GCF_027555065.1_ASM2755506v1_genomic.gff', 'GCA_017357525.1/GCF_017357525.1_ASM1735752v1_genomic.gff', 'GCA_000350305.1/GCA_000350305.1_ASM35030v1_genomic.gff', 'GCA_902827205.1/GCA_902827205.1_Methanocaldococcus_lauensis_SG1_genomic.gff', 'GCA_030247745.1/GCA_030247745.1_ASM3024774v1_genomic.gff', 'GCA_002508625.1/GCA_002508625.1.gff', 'GCA_017468685.1/GCA_017468685.1_ASM1746868v1_genomic.gff', 'GCA_000828555.1/GCF_000828555.1_ASM82855v1_genomic.gff', 'GCA_000275865.1/GCA_000275865.1_ASM27586v1_genomic.gff', 'GCA_000015825.1/GCF_000015825.1_ASM1582v1_genomic.gff', 'GCA_027554905.1/GCF_027554905.1_ASM2755490v1_genomic.gff', 'GCA_017875325.1/GCF_017875325.1_ASM1787532v1_genomic.gff', 'GCA_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff', 'GCA_024662215.1/GCF_024662215.1_ASM2466221v1_genomic.gff', 'GCA_000016525.1/GCF_000016525.1_ASM1652v1_genomic.gff', 'GCA_026684035.1/GCF_026684035.1_ASM2668403v1_genomic.gff', 'GCA_017553245.1/GCA_017553245.1_ASM1755324v1_genomic.gff', 'GCA_000063445.1/GCF_000063445.1_ASM6344v1_genomic.gff', 'GCA_002498765.1/GCA_002498765.1.gff', 'GCA_017874375.1/GCF_017874375.1_ASM1787437v1_genomic.gff', 'GCA_002495325.1/GCA_002495325.1.gff', 'GCA_000739065.1/GCF_000739065.1_ASM73906v1_genomic.gff', 'GCA_002498425.1/GCA_002498425.1.gff', 'GCA_022846155.1/GCF_022846155.1_ASM2284615v1_genomic.gff', 'GCA_017561385.1/GCA_017561385.1_ASM1756138v1_genomic.gff', 'GCA_002498375.1/GCA_002498375.1.gff', 'GCA_000007185.1/GCF_000007185.1_ASM718v1_genomic.gff', 'GCA_000017165.1/GCF_000017165.1_ASM1716v1_genomic.gff', 'GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', 'GCA_003711245.1/GCF_003711245.1_ASM371124v1_genomic.gff', 'GCA_000147875.1/GCF_000147875.1_ASM14787v1_genomic.gff', 'GCA_000745485.1/GCF_000745485.1_ASM74548v1_genomic.gff', 'GCA_002508855.1/GCA_002508855.1.gff', 'GCA_000214415.1/GCF_000214415.1_ASM21441v1_genomic.gff', 'GCA_013388255.1/GCF_013388255.1_ASM1338825v1_genomic.gff', 'GCA_000376965.1/GCF_000376965.1_ASM37696v1_genomic.gff', 'GCA_902383905.1/GCF_902383905.1_UHGG_MGYG-HGUT-02160_genomic.gff', 'GCA_002495685.1/GCA_002495685.1.gff', 'GCA_002503135.1/GCA_002503135.1.gff', 'GCA_000970245.1/GCF_000970245.1_ASM97024v1_genomic.gff', 'GCA_000304355.2/GCF_000304355.2_Mb_MS2_genomic.gff', 'GCA_000013445.1/GCF_000013445.1_ASM1344v1_genomic.gff', 'GCA_017411925.1/GCA_017411925.1_ASM1741192v1_genomic.gff', 'GCA_017873625.1/GCF_017873625.1_ASM1787362v1_genomic.gff', 'GCA_000711215.1/GCF_000711215.1_ASM71121v1_genomic.gff', 'GCA_003491305.1/GCA_003491305.1_ASM349130v1_genomic.gff', 'GCA_000320505.1/GCF_000320505.1_ASM32050v2_genomic.gff', 'GCA_017407825.1/GCA_017407825.1_ASM1740782v1_genomic.gff', 'GCA_000970005.1/GCF_000970005.1_ASM97000v1_genomic.gff', 'GCA_013761105.1/GCF_013761105.1_ASM1376110v1_genomic.gff', 'GCA_000970205.1/GCF_000970205.1_ASM97020v1_genomic.gff', 'GCA_004363215.1/GCF_004363215.1_ASM436321v1_genomic.gff', 'GCA_024807035.1/GCF_024807035.1_ASM2480703v1_genomic.gff', 'GCA_017519045.1/GCA_017519045.1_ASM1751904v1_genomic.gff', 'GCA_013761045.1/GCF_013761045.1_ASM1376104v1_genomic.gff', 'GCA_017873855.1/GCF_017873855.1_ASM1787385v1_genomic.gff', 'GCA_001304615.2/GCF_001304615.2_ASM130461v2_genomic.gff', 'GCA_017412625.1/GCA_017412625.1_ASM1741262v1_genomic.gff', 'GCA_023617305.1/GCF_023617305.1_ASM2361730v1_genomic.gff', 'GCA_000970045.1/GCF_000970045.1_ASM97004v1_genomic.gff', 'GCA_009911715.1/GCA_009911715.1_ASM991171v1_genomic.gff', 'GCA_025449255.1/GCF_025449255.1_ASM2544925v1_genomic.gff', 'GCA_017875395.1/GCF_017875395.1_ASM1787539v1_genomic.gff', 'GCA_025914035.1/GCF_025914035.1_ASM2591403v1_genomic.gff', 'GCA_003385755.1/GCF_003385755.1_ASM338575v1_genomic.gff', 'GCA_017633855.1/GCA_017633855.1_ASM1763385v1_genomic.gff', 'GCA_000191585.1/GCF_000191585.1_ASM19158v1_genomic.gff', 'GCA_013761055.1/GCF_013761055.1_ASM1376105v1_genomic.gff', 'GCA_000007065.1/GCF_000007065.1_ASM706v1_genomic.gff', 'GCA_002498785.1/GCA_002498785.1.gff', 'GCA_900079125.1/GCA_900079125.1_MBBA_genomic.gff', 'GCA_002813655.1/GCF_002813655.1_ASM281365v1_genomic.gff', 'GCA_017940815.1/GCA_017940815.1_ASM1794081v1_genomic.gff', 'GCA_017387505.1/GCA_017387505.1_ASM1738750v1_genomic.gff', 'GCA_000204415.1/GCF_000204415.1_ASM20441v1_genomic.gff', 'GCA_002506995.1/GCA_002506995.1.gff', 'GCA_000800805.1/GCF_000800805.1_ASM80080v1_genomic.gff', 'GCA_003351865.1/GCF_003351865.1_ASM335186v1_genomic.gff', 'GCA_017410345.1/GCA_017410345.1_ASM1741034v1_genomic.gff', 'GCA_002494585.1/GCA_002494585.1.gff', 'GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff', 'GCA_000017185.1/GCF_000017185.1_ASM1718v1_genomic.gff', 'GCA_017473945.1/GCA_017473945.1_ASM1747394v1_genomic.gff', 'GCA_002502465.1/GCA_002502465.1.gff', 'GCA_024170505.1/GCF_024170505.1_ASM2417050v1_genomic.gff', 'GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', 'GCA_017409525.1/GCA_017409525.1_ASM1740952v1_genomic.gff', 'GCA_001458655.1/GCF_001458655.1_Mb9_genomic.gff', 'GCA_017406825.1/GCA_017406825.1_ASM1740682v1_genomic.gff', 'GCA_000008645.1/GCF_000008645.1_ASM864v1_genomic.gff', 'GCA_000145295.1/GCF_000145295.1_ASM14529v1_genomic.gff', 'GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', 'GCA_900036045.1/GCF_900036045.1_Methanoculleus_sp_MAB1_genomic.gff', 'GCA_900106905.1/GCF_900106905.1_IMG-taxon_2642422606_annotated_assembly_genomic.gff', 'GCA_000023985.1/GCF_000023985.1_ASM2398v1_genomic.gff', 'GCA_014203295.1/GCF_014203295.1_ASM1420329v1_genomic.gff', 'GCA_000341715.1/GCF_000341715.1_ASM34171v1_genomic.gff', 'GCA_018502485.1/GCF_018502485.1_ASM1850248v1_genomic.gff', 'GCA_029633915.1/GCF_029633915.1_ASM2963391v1_genomic.gff', 'GCA_008033705.1/GCF_008033705.1_ASM803370v1_genomic.gff', 'GCA_014207315.1/GCF_014207315.1_ASM1420731v1_genomic.gff', 'GCA_902787415.1/GCA_902787415.1.gff', 'GCA_000025865.1/GCF_000025865.1_ASM2586v1_genomic.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "275"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_paths = []\n",
    "for file in gff_files:\n",
    "    gff_paths.append(file[40:])\n",
    "print(gff_paths)\n",
    "len(gff_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1316e212",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/Users/gracedai/Documents/asmg labs files/optimize/to_search_co2.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[11], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mpd\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m#read the text file of gene names into a list\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m gene_file \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m/Users/gracedai/Documents/asmg labs files/optimize/to_search_co2.txt\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      4\u001b[0m data \u001b[38;5;241m=\u001b[39m gene_file\u001b[38;5;241m.\u001b[39mread()\n\u001b[1;32m      5\u001b[0m to_search \u001b[38;5;241m=\u001b[39m data\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/IPython/core/interactiveshell.py:284\u001b[0m, in \u001b[0;36m_modified_open\u001b[0;34m(file, *args, **kwargs)\u001b[0m\n\u001b[1;32m    277\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m {\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m2\u001b[39m}:\n\u001b[1;32m    278\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m    279\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIPython won\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt let you open fd=\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m by default \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    280\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mas it is likely to crash IPython. If you know what you are doing, \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    281\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124myou can use builtins\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m open.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    282\u001b[0m     )\n\u001b[0;32m--> 284\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mio_open\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfile\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/Users/gracedai/Documents/asmg labs files/optimize/to_search_co2.txt'"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "#read the text file of gene names into a list\n",
    "gene_file = open(\"/Users/gracedai/Documents/asmg labs files/optimize/to_search_co2.txt\", \"r\")\n",
    "data = gene_file.read()\n",
    "to_search = data.split(\"\\n\")\n",
    "print(to_search)\n",
    "gene_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "84a557dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(to_search)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b63f759",
   "metadata": {},
   "source": [
    "GFF File Format: \n",
    "- Fields must be tab-separated. Also, all but the final field in each feature line must contain a value; \"empty\" columns should be denoted with a '.'\n",
    "[0] seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.\n",
    "[1] source - name of the program that generated this feature, or the data source (database or project name)\n",
    "[2] feature - feature type name, e.g. Gene, Variation, Similarity\n",
    "[3] start - Start position* of the feature, with sequence numbering starting at 1.\n",
    "[4] end - End position* of the feature, with sequence numbering starting at 1.\n",
    "[5] score - A floating point value.\n",
    "[6] strand - defined as + (forward) or - (reverse).\n",
    "[7] frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..\n",
    "[8] attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e3288df2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create an empty dictionary to store the gene counts for each species\n",
    "gene_counts = {}\n",
    "\n",
    "#for gene pathways: iterate through all species gff files\n",
    "# for path_name in gff_paths:\n",
    "#     species = path_name[:15]  #keep species name from the path_name\n",
    "#     gene_counts[species] = {}\n",
    "    \n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 for gene in to_search:\n",
    "#                     if \"gene=\" + gene in line:\n",
    "#                         if gene not in gene_counts[species]:\n",
    "#                             gene_counts[species][gene] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species][gene] += 1\n",
    "    #read the GFF file\n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 fields = line.strip().split('\\t')\n",
    "#                 if (fields[2] == 'gene'):\n",
    "#                     attributes = fields[8] #field number 8 contains attributes such as gene name\n",
    "#                     for gene in to_search:\n",
    "#                         if gene in attributes:\n",
    "#                             if gene not in gene_counts[species]:\n",
    "#                                 gene_counts[species][gene] = 1\n",
    "#                             else:\n",
    "#                                 gene_counts[species][gene] += 1\n",
    "\n",
    "# for 16s and tRNA\n",
    "# for path_name in gff_paths:\n",
    "#     species = path_name[:15]  #keep species name from the path_name\n",
    "#     gene_counts[species] = {}\n",
    "\n",
    "#     #read the GFF file\n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 fields = line.strip().split('\\t')\n",
    "#                 attributes = fields[8] #field number 8 contains attributes such as gene name\n",
    "#                 if fields[2] == \"gene\":\n",
    "#                     if 'tRNA' in attributes:\n",
    "#                         if 'tRNA' not in gene_counts[species]:\n",
    "#                             gene_counts[species]['tRNA'] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species]['tRNA'] += 1\n",
    "#                 if fields[2] == \"rRNA\":\n",
    "#                     if '16S' in attributes:\n",
    "#                         if '16S' not in gene_counts[species]:\n",
    "#                             gene_counts[species]['16S'] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species]['16S'] += 1\n",
    "\n",
    "for path_name in gff_paths:\n",
    "    species = path_name[:15]  #keep species name from the path_name\n",
    "    gene_counts[species] = {}\n",
    "\n",
    "    #read the GFF file\n",
    "    with open(path_name, 'r') as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                if \"Name=mrtA\" in line or \"Name=MrtA\" in line:\n",
    "                    if 'mrtA' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtA'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtA'] += 1\n",
    "                if \"Name=mrtB\" in line or \"Name=MrtB\" in line: \n",
    "                    if 'mrtB' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtB'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtB'] += 1\n",
    "                if \"Name=mrtG\" in line or \"Name=MrtG\" in line: \n",
    "                    if 'mrtG' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtG'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtG'] += 1\n",
    "\n",
    "# Create a dataframe from the gene counts\n",
    "df = pd.DataFrame.from_dict(gene_counts, orient='index')\n",
    "df.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d146647d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mrtG</th>\n",
       "      <th>mrtB</th>\n",
       "      <th>mrtA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCA_002502965.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902762095.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_000953115.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_900095815.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902763685.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902795935.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002506325.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002506905.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002509405.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002496945.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002505245.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498705.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498605.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002504405.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002506565.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002505345.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002494805.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498365.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002505275.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498805.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002503785.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002508625.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498765.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002495325.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498425.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498375.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002498785.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002506995.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002494585.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002502465.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_000499765.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902787415.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_905171695.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_905171685.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    mrtG  mrtB  mrtA\n",
       "assembly_accession                  \n",
       "GCA_002502965.1      1.0   0.0   0.0\n",
       "GCA_902762095.1      1.0   1.0   1.0\n",
       "GCA_000953115.1      1.0   1.0   1.0\n",
       "GCA_900095815.1      1.0   1.0   1.0\n",
       "GCA_902763685.1      1.0   0.0   0.0\n",
       "GCA_902795935.1      1.0   1.0   1.0\n",
       "GCA_002506325.1      1.0   0.0   0.0\n",
       "GCA_002506905.1      1.0   0.0   0.0\n",
       "GCA_002509405.1      1.0   0.0   0.0\n",
       "GCA_002496945.1      1.0   0.0   0.0\n",
       "GCA_002505245.1      1.0   0.0   0.0\n",
       "GCA_002498705.1      1.0   0.0   0.0\n",
       "GCA_002498605.1      1.0   0.0   0.0\n",
       "GCA_002504405.1      1.0   0.0   0.0\n",
       "GCA_002506565.1      1.0   0.0   0.0\n",
       "GCA_002505345.1      1.0   0.0   0.0\n",
       "GCA_002494805.1      1.0   0.0   0.0\n",
       "GCA_002498365.1      1.0   0.0   0.0\n",
       "GCA_002505275.1      1.0   0.0   0.0\n",
       "GCA_002498805.1      1.0   0.0   0.0\n",
       "GCA_002503785.1      1.0   0.0   0.0\n",
       "GCA_002508625.1      1.0   0.0   0.0\n",
       "GCA_002498765.1      1.0   0.0   0.0\n",
       "GCA_002495325.1      1.0   0.0   0.0\n",
       "GCA_002498425.1      1.0   0.0   0.0\n",
       "GCA_002498375.1      1.0   0.0   0.0\n",
       "GCA_002498785.1      1.0   0.0   0.0\n",
       "GCA_002506995.1      1.0   0.0   0.0\n",
       "GCA_002494585.1      1.0   0.0   0.0\n",
       "GCA_002502465.1      1.0   0.0   0.0\n",
       "GCA_000499765.1      1.0   1.0   1.0\n",
       "GCA_902787415.1      1.0   0.0   0.0\n",
       "GCA_905171695.1      0.0   1.0   0.0\n",
       "GCA_905171685.1      0.0   1.0   0.0"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename_axis('assembly_accession')\n",
    "df"
   ]
  },
  {
   "cell_type": "raw",
   "id": "03930dc3",
   "metadata": {},
   "source": [
    "df.shape[0] #missing 5 rows (species) \n",
    "df.shape[1] #missing 5 columns (genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4132f9d3",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[43mto_search\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m      2\u001b[0m df\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/frame.py:3813\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3811\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_iterator(key):\n\u001b[1;32m   3812\u001b[0m         key \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(key)\n\u001b[0;32m-> 3813\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_indexer_strict\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcolumns\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m[\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m   3815\u001b[0m \u001b[38;5;66;03m# take() does not accept boolean indexers\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(indexer, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdtype\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mbool\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6070\u001b[0m, in \u001b[0;36mIndex._get_indexer_strict\u001b[0;34m(self, key, axis_name)\u001b[0m\n\u001b[1;32m   6067\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   6068\u001b[0m     keyarr, indexer, new_indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex_non_unique(keyarr)\n\u001b[0;32m-> 6070\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_raise_if_missing\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkeyarr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis_name\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   6072\u001b[0m keyarr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtake(indexer)\n\u001b[1;32m   6073\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(key, Index):\n\u001b[1;32m   6074\u001b[0m     \u001b[38;5;66;03m# GH 42790 - Preserve name from an Index\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6133\u001b[0m, in \u001b[0;36mIndex._raise_if_missing\u001b[0;34m(self, key, indexer, axis_name)\u001b[0m\n\u001b[1;32m   6130\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNone of [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mkey\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m] are in the [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00maxis_name\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m]\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   6132\u001b[0m not_found \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(ensure_index(key)[missing_mask\u001b[38;5;241m.\u001b[39mnonzero()[\u001b[38;5;241m0\u001b[39m]]\u001b[38;5;241m.\u001b[39munique())\n\u001b[0;32m-> 6133\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mnot_found\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not in index\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\""
     ]
    }
   ],
   "source": [
    "df = df[to_search]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "ec15cc8d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mcrA</th>\n",
       "      <th>mcrG</th>\n",
       "      <th>mcrB</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>mtbC</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>mtbA</th>\n",
       "      <th>...</th>\n",
       "      <th>mtmB</th>\n",
       "      <th>mtbB</th>\n",
       "      <th>mttC</th>\n",
       "      <th>mttB</th>\n",
       "      <th>hdrA1</th>\n",
       "      <th>mtmC</th>\n",
       "      <th>hdrB1</th>\n",
       "      <th>hdrC1</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_001458655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000008645.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000145295.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008033705.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002813655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001477655.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_028743435.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000970325.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001027005.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_000306725.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    mvhA  mcrA  mcrG  mcrB  mvhG  hdrA2  mtbC  hdrE  hdrD  \\\n",
       "assembly_accession                                                          \n",
       "GCF_001458655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000008645.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000145295.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_008033705.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_002813655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "...                  ...   ...   ...   ...   ...    ...   ...   ...   ...   \n",
       "GCF_001477655.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_028743435.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000970325.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   0.0   \n",
       "GCF_001027005.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   1.0   \n",
       "GCA_000306725.1      0.0   0.0   0.0   0.0   0.0    0.0   0.0   0.0   0.0   \n",
       "\n",
       "                    mtbA  ...  mtmB  mtbB  mttC  mttB  hdrA1  mtmC  hdrB1  \\\n",
       "assembly_accession        ...                                               \n",
       "GCF_001458655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000008645.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000145295.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_008033705.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_002813655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "...                  ...  ...   ...   ...   ...   ...    ...   ...    ...   \n",
       "GCF_001477655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_028743435.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000970325.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_001027005.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCA_000306725.1      1.0  ...   4.0   3.0   1.0   2.0    0.0   0.0    0.0   \n",
       "\n",
       "                    hdrC1  hdrB2  hdrC2  \n",
       "assembly_accession                       \n",
       "GCF_001458655.1       0.0    0.0    0.0  \n",
       "GCF_000008645.1       0.0    0.0    0.0  \n",
       "GCF_000145295.1       0.0    0.0    0.0  \n",
       "GCF_008033705.1       0.0    0.0    0.0  \n",
       "GCF_002813655.1       0.0    0.0    0.0  \n",
       "...                   ...    ...    ...  \n",
       "GCF_001477655.1       0.0    0.0    0.0  \n",
       "GCF_028743435.1       0.0    0.0    0.0  \n",
       "GCF_000970325.1       0.0    0.0    0.0  \n",
       "GCF_001027005.1       0.0    0.0    0.0  \n",
       "GCA_000306725.1       0.0    0.0    0.0  \n",
       "\n",
       "[199 rows x 23 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add columns that are missing, fill in with 0\n",
    "missing = ['mtmC','hdrB1', 'hdrC1', 'hdrB2', 'hdrC2']\n",
    "for i in missing: \n",
    "    df[i] = 0.0\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "ef5985d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002495685.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Accession                  organism_name   Isolate\n",
       "0  GCA_902788255.1  uncultured Methanobrevibacter  RUG13354\n",
       "1  GCA_002494785.1   Methanobacteriaceae archaeon    UBA117\n",
       "2  GCA_902784195.1  uncultured Methanobrevibacter  RUG12946\n",
       "3  GCA_002495685.1             Methanobrevibacter    UBA412\n",
       "4  GCA_902771225.1  uncultured Methanobrevibacter  RUG11657"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#match species to the assembly acession numbers \n",
    "#first import csv with species and their accession numbers \n",
    "species = pd.read_csv('/Users/gracedai/Documents/MAGS.csv')\n",
    "species = species[['Accession', 'organism_name', 'Isolate']]\n",
    "species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4c6d603f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mrtG</th>\n",
       "      <th>mrtB</th>\n",
       "      <th>mrtA</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002502965.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCA_902795935.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002506325.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002506905.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002509405.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002496945.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002505245.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498605.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002504405.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002506565.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002505345.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002494805.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498365.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002505275.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498805.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002503785.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002508625.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498765.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002495325.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498425.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002498785.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002506995.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002494585.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002502465.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    mrtG  mrtB  mrtA        Accession                      organism_name  \\\n",
       "0    1.0   0.0   0.0  GCA_002502965.1  Methanomassiliicoccaceae archaeon   \n",
       "1    1.0   1.0   1.0  GCA_902762095.1      uncultured Methanobrevibacter   \n",
       "2    1.0   1.0   1.0              NaN                                NaN   \n",
       "3    1.0   1.0   1.0              NaN                                NaN   \n",
       "4    1.0   0.0   0.0  GCA_902763685.1              methanogenic archaeon   \n",
       "5    1.0   1.0   1.0  GCA_902795935.1      uncultured Methanobrevibacter   \n",
       "6    1.0   0.0   0.0  GCA_002506325.1  Methanomassiliicoccaceae archaeon   \n",
       "7    1.0   0.0   0.0  GCA_002506905.1  Methanomassiliicoccaceae archaeon   \n",
       "8    1.0   0.0   0.0  GCA_002509405.1  Methanomassiliicoccaceae archaeon   \n",
       "9    1.0   0.0   0.0  GCA_002496945.1  Methanomassiliicoccaceae archaeon   \n",
       "10   1.0   0.0   0.0  GCA_002505245.1  Methanomassiliicoccaceae archaeon   \n",
       "11   1.0   0.0   0.0  GCA_002498705.1                 Methanocorpusculum   \n",
       "12   1.0   0.0   0.0  GCA_002498605.1  Methanomassiliicoccaceae archaeon   \n",
       "13   1.0   0.0   0.0  GCA_002504405.1  Methanomassiliicoccaceae archaeon   \n",
       "14   1.0   0.0   0.0  GCA_002506565.1  Methanomassiliicoccaceae archaeon   \n",
       "15   1.0   0.0   0.0  GCA_002505345.1  Methanomassiliicoccaceae archaeon   \n",
       "16   1.0   0.0   0.0  GCA_002494805.1  Methanomassiliicoccaceae archaeon   \n",
       "17   1.0   0.0   0.0  GCA_002498365.1  Methanomassiliicoccaceae archaeon   \n",
       "18   1.0   0.0   0.0  GCA_002505275.1  Methanomassiliicoccaceae archaeon   \n",
       "19   1.0   0.0   0.0  GCA_002498805.1  Methanomassiliicoccaceae archaeon   \n",
       "20   1.0   0.0   0.0  GCA_002503785.1  Methanomassiliicoccaceae archaeon   \n",
       "21   1.0   0.0   0.0  GCA_002508625.1  Methanomassiliicoccaceae archaeon   \n",
       "22   1.0   0.0   0.0  GCA_002498765.1  Methanomassiliicoccaceae archaeon   \n",
       "23   1.0   0.0   0.0  GCA_002495325.1  Methanomassiliicoccaceae archaeon   \n",
       "24   1.0   0.0   0.0  GCA_002498425.1  Methanomassiliicoccaceae archaeon   \n",
       "25   1.0   0.0   0.0  GCA_002498375.1                 Methanocorpusculum   \n",
       "26   1.0   0.0   0.0  GCA_002498785.1  Methanomassiliicoccaceae archaeon   \n",
       "27   1.0   0.0   0.0  GCA_002506995.1  Methanomassiliicoccaceae archaeon   \n",
       "28   1.0   0.0   0.0  GCA_002494585.1  Methanomassiliicoccaceae archaeon   \n",
       "29   1.0   0.0   0.0  GCA_002502465.1  Methanomassiliicoccaceae archaeon   \n",
       "30   1.0   1.0   1.0              NaN                                NaN   \n",
       "31   1.0   0.0   0.0  GCA_902787415.1              methanogenic archaeon   \n",
       "32   0.0   1.0   0.0              NaN                                NaN   \n",
       "33   0.0   1.0   0.0              NaN                                NaN   \n",
       "\n",
       "     Isolate  \n",
       "0     UBA381  \n",
       "1   RUG10736  \n",
       "2        NaN  \n",
       "3        NaN  \n",
       "4   RUG10898  \n",
       "5   RUG14110  \n",
       "6     UBA314  \n",
       "7     UBA328  \n",
       "8      UBA72  \n",
       "9     UBA408  \n",
       "10    UBA422  \n",
       "11    UBA362  \n",
       "12    UBA404  \n",
       "13     UBA71  \n",
       "14    UBA333  \n",
       "15    UBA370  \n",
       "16    UBA482  \n",
       "17    UBA421  \n",
       "18    UBA371  \n",
       "19    UBA399  \n",
       "20     UBA75  \n",
       "21    UBA344  \n",
       "22    UBA343  \n",
       "23    UBA593  \n",
       "24    UBA317  \n",
       "25    UBA424  \n",
       "26    UBA400  \n",
       "27    UBA427  \n",
       "28    UBA409  \n",
       "29    UBA296  \n",
       "30       NaN  \n",
       "31  RUG13270  \n",
       "32       NaN  \n",
       "33       NaN  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='Accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "950cd6a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('mrt1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "dc9fa0b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#resort column order\n",
    "cols = ['Accession', 'organism_name', 'Isolate', \n",
    "    '16S', 'tRNA']\n",
    "df = df[cols]\n",
    "\n",
    "#add row with kegg numbers\n",
    "# kegg_row = [0,0,0,\n",
    "#     'K14082',\n",
    "#     'K16176',\n",
    "#     'K16177',\n",
    "#     'K16178',\n",
    "#     'K16179',\n",
    "#     'K14083',\n",
    "#     'K14084',\n",
    "#     'K00399',\n",
    "#     'K00401',\n",
    "#     'K00402',\n",
    "#     'K22480',\n",
    "#     'K22481',\n",
    "#     'K22482',\n",
    "#     'K03388',\n",
    "#     'K03389',\n",
    "#     'K03390',\n",
    "#     'K08264',\n",
    "#     'K08265',\n",
    "#     'K14126',\n",
    "#     'K14128',\n",
    "#     'K14127',\n",
    "#     'K22516',\n",
    "#     'K00125']\n",
    "\n",
    "# kegg_row_df = pd.DataFrame([kegg_row], columns=df.columns)\n",
    "#add row with gene description\n",
    "# descrip = [0, 0, 0, '[methyl-Co(III) methylamine-specific corrinoid protein]:coenzyme M methyltransferase [EC:2.1.1.247]',\n",
    "#     'methylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.248]',\n",
    "#     'monomethylamine corrinoid protein',\n",
    "#     'dimethylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.249]',\n",
    "#     'dimethylamine corrinoid protein',\n",
    "#     'trimethylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.250]',\n",
    "#     'trimethylamine corrinoid protein',\n",
    "#     'methyl-coenzyme M reductase alpha subunit [EC:2.8.4.1]',\n",
    "#     'methyl-coenzyme M reductase beta subunit [EC:2.8.4.1]',\n",
    "#     'methyl-coenzyme M reductase gamma subunit [EC:2.8.4.1]',\n",
    "#     'heterodisulfide reductase subunit A1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit B1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit C1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit A2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit B2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit C2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit D [EC:1.8.98.1]',\n",
    "#     'heterodisulfide reductase subunit E [EC:1.8.98.1]',\n",
    "#     'F420-non-reducing hydrogenase large subunit [EC:1.12.99.- 1.8.98.5]',\n",
    "#     'F420-non-reducing hydrogenase small subunit [EC:1.12.99.- 1.8.98.5]',\n",
    "#     'F420-non-reducing hydrogenase iron-sulfur subunit [EC:1.12.99.- 1.8.98.5 1.8.98.6]',\n",
    "#     'formate dehydrogenase (coenzyme F420) alpha subunit [EC:1.17.98.3 1.8.98.6]',\n",
    "#     'formate dehydrogenase (coenzyme F420) beta subunit [EC:1.17.98.3 1.8.98.6]']\n",
    "# descrip_df = pd.DataFrame([descrip], columns=df.columns)\n",
    "# df = pd.concat([kegg_row_df, descrip_df, df], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "649191c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "      <td>1.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "      <td>2.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>GCA_902764455.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>GCA_902768415.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11363</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                      organism_name   Isolate  16S  tRNA\n",
       "0    GCA_902788045.1      uncultured Methanobrevibacter  RUG13330  0.0    32\n",
       "1    GCA_902774685.1      uncultured Methanobrevibacter  RUG11990  0.0    28\n",
       "2    GCA_002497785.1                       Methanolinea    UBA144  1.0    37\n",
       "3    GCA_002502925.1  Methanomassiliicoccaceae archaeon    UBA414  2.0    41\n",
       "4    GCA_002506795.1                   Methanobacterium    UBA397  0.0    38\n",
       "..               ...                                ...       ...  ...   ...\n",
       "218  GCA_902764455.1      uncultured Methanobrevibacter  RUG10962  0.0    35\n",
       "219  GCA_902802065.1      uncultured Methanobrevibacter  RUG14729  2.0    35\n",
       "220  GCA_902787415.1              methanogenic archaeon  RUG13270  0.0    35\n",
       "221  GCA_902768415.1      uncultured Methanobrevibacter  RUG11363  0.0    27\n",
       "222  GCA_002502855.1                   Methanobacterium    UBA279  0.0    37\n",
       "\n",
       "[223 rows x 5 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "33c18b50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_902770715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11588</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902801725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14697</td>\n",
       "      <td>1.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GCA_902758945.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10414</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_902769175.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11428</td>\n",
       "      <td>0.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>GCA_902777475.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_902759195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10437</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>GCA_902784295.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12954</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>GCA_902762245.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>GCA_902770085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11531</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>GCA_902774705.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11996</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                  organism_name   Isolate  16S  tRNA\n",
       "38   GCA_902770715.1  uncultured Methanobrevibacter  RUG11588  0.0    30\n",
       "37   GCA_902801725.1  uncultured Methanobrevibacter  RUG14697  1.0    30\n",
       "166  GCA_902798335.1  uncultured Methanobrevibacter  RUG14349  0.0    31\n",
       "167  GCA_902757275.1  uncultured Methanobrevibacter  RUG10244  0.0    36\n",
       "75   GCA_902758945.1  uncultured Methanobrevibacter  RUG10414  0.0    33\n",
       "34   GCA_902769175.1  uncultured Methanobrevibacter  RUG11428  0.0    41\n",
       "125  GCA_902777475.1  uncultured Methanobrevibacter  RUG12275  0.0    34\n",
       "32   GCA_902759195.1  uncultured Methanobrevibacter  RUG10437  0.0    33\n",
       "174  GCA_902784295.1  uncultured Methanobrevibacter  RUG12954  0.0    31\n",
       "31   GCA_902788255.1  uncultured Methanobrevibacter  RUG13354  0.0    31\n",
       "177  GCA_902762245.1  uncultured Methanobrevibacter  RUG10743  0.0    30\n",
       "178  GCA_902770085.1  uncultured Methanobrevibacter  RUG11531  0.0    34\n",
       "123  GCA_902775005.1  uncultured Methanobrevibacter  RUG12025  0.0    30\n",
       "25   GCA_902762085.1  uncultured Methanobrevibacter  RUG10738  0.0    23\n",
       "140  GCA_902774705.1  uncultured Methanobrevibacter  RUG11996  0.0    82\n",
       "184  GCA_902765105.1  uncultured Methanobrevibacter  RUG11035  0.0    30\n",
       "185  GCA_902785855.1  uncultured Methanobrevibacter  RUG13115  0.0    28\n",
       "120  GCA_902757365.1  uncultured Methanobrevibacter  RUG10252  0.0    36\n",
       "21   GCA_902784715.1  uncultured Methanobrevibacter  RUG12995  0.0    27\n",
       "0    GCA_902788045.1  uncultured Methanobrevibacter  RUG13330  0.0    32"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_name'] = df['organism_name'].astype(str)\n",
    "df_sorted = df.sort_values('organism_name', ascending=True)\n",
    "df_sorted.tail(n=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "cefb5125",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sorted.to_csv('prokka_16s.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "14289b16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "c18cc498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism_name</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>strain</th>\n",
       "      <th>fwdA</th>\n",
       "      <th>fmdA</th>\n",
       "      <th>fwdB</th>\n",
       "      <th>fmdB</th>\n",
       "      <th>fwdC</th>\n",
       "      <th>fmdC</th>\n",
       "      <th>fwdD</th>\n",
       "      <th>...</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>mvhD</th>\n",
       "      <th>fdhA</th>\n",
       "      <th>fdhB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#CO2 -&gt; methane</td>\n",
       "      <td># KEGG M00567</td>\n",
       "      <td>#</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00203</td>\n",
       "      <td>...</td>\n",
       "      <td>K03388</td>\n",
       "      <td>K03389</td>\n",
       "      <td>K03390</td>\n",
       "      <td>K08264</td>\n",
       "      <td>K08265</td>\n",
       "      <td>K14126</td>\n",
       "      <td>K14128</td>\n",
       "      <td>K14127</td>\n",
       "      <td>K22516</td>\n",
       "      <td>K00125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit D [EC...</td>\n",
       "      <td>...</td>\n",
       "      <td>heterodisulfide reductase subunit A2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit B2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit C2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit D [EC:1.8.98.1]</td>\n",
       "      <td>heterodisulfide reductase subunit E [EC:1.8.98.1]</td>\n",
       "      <td>F420-non-reducing hydrogenase large subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase small subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase iron-sulfur subu...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) alpha su...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) beta sub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Candidatus Methanogranum sp. U3.2.1</td>\n",
       "      <td>GCA_019262145.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Candidatus Methanoliparum sp. LAM-1</td>\n",
       "      <td>GCF_020885915.1</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis</td>\n",
       "      <td>GCF_902383905.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis ...</td>\n",
       "      <td>GCF_000404225.1</td>\n",
       "      <td>Issoire-Mx1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_902387285.1</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_003711245.1</td>\n",
       "      <td>Mx-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus Mx1201</td>\n",
       "      <td>GCF_000300255.2</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Candidatus Methanomethylophilus sp. 1R26 (arch...</td>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>1R26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       organism_name assembly_accession  \\\n",
       "0                                    #CO2 -> methane      # KEGG M00567   \n",
       "1                                                  #                  #   \n",
       "2                Candidatus Methanogranum sp. U3.2.1    GCA_019262145.1   \n",
       "3                Candidatus Methanoliparum sp. LAM-1    GCF_020885915.1   \n",
       "4      Candidatus Methanomassiliicoccus intestinalis    GCF_902383905.1   \n",
       "5  Candidatus Methanomassiliicoccus intestinalis ...    GCF_000404225.1   \n",
       "6              Candidatus Methanomethylophilus alvus    GCF_902387285.1   \n",
       "7              Candidatus Methanomethylophilus alvus    GCF_003711245.1   \n",
       "8       Candidatus Methanomethylophilus alvus Mx1201    GCF_000300255.2   \n",
       "9  Candidatus Methanomethylophilus sp. 1R26 (arch...    GCF_001481295.1   \n",
       "\n",
       "        strain                                               fwdA  \\\n",
       "0            #                                             K00200   \n",
       "1            #  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2           na                                                  0   \n",
       "3        LAM-1                                                  0   \n",
       "4           na                                                  0   \n",
       "5  Issoire-Mx1                                                  0   \n",
       "6       Mx1201                                                  0   \n",
       "7        Mx-05                                                  0   \n",
       "8       Mx1201                                                  0   \n",
       "9         1R26                                                  0   \n",
       "\n",
       "                                                fmdA  \\\n",
       "0                                             K00200   \n",
       "1  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdD  ...  \\\n",
       "0                                             K00203  ...   \n",
       "1  formylmethanofuran dehydrogenase subunit D [EC...  ...   \n",
       "2                                                  0  ...   \n",
       "3                                                  0  ...   \n",
       "4                                                  0  ...   \n",
       "5                                                  0  ...   \n",
       "6                                                  0  ...   \n",
       "7                                                  0  ...   \n",
       "8                                                  0  ...   \n",
       "9                                                  0  ...   \n",
       "\n",
       "                                               hdrA2  \\\n",
       "0                                             K03388   \n",
       "1  heterodisulfide reductase subunit A2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  2   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrB2  \\\n",
       "0                                             K03389   \n",
       "1  heterodisulfide reductase subunit B2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrC2  \\\n",
       "0                                             K03390   \n",
       "1  heterodisulfide reductase subunit C2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrD  \\\n",
       "0                                             K08264   \n",
       "1  heterodisulfide reductase subunit D [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrE  \\\n",
       "0                                             K08265   \n",
       "1  heterodisulfide reductase subunit E [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhA  \\\n",
       "0                                             K14126   \n",
       "1  F420-non-reducing hydrogenase large subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhG  \\\n",
       "0                                             K14128   \n",
       "1  F420-non-reducing hydrogenase small subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhD  \\\n",
       "0                                             K14127   \n",
       "1  F420-non-reducing hydrogenase iron-sulfur subu...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhA  \\\n",
       "0                                             K22516   \n",
       "1  formate dehydrogenase (coenzyme F420) alpha su...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhB  \n",
       "0                                             K00125  \n",
       "1  formate dehydrogenase (coenzyme F420) beta sub...  \n",
       "2                                                  0  \n",
       "3                                                  0  \n",
       "4                                                  0  \n",
       "5                                                  0  \n",
       "6                                                  0  \n",
       "7                                                  0  \n",
       "8                                                  0  \n",
       "9                                                  0  \n",
       "\n",
       "[10 rows x 46 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs = pd.read_csv('/Users/gracedai/Downloads/co2.csv')\n",
    "prokka = pd.read_csv('/Users/gracedai/Downloads/prokkas.csv')\n",
    "rumen_only = pd.read_csv('/Users/gracedai/Downloads/rumen.csv')\n",
    "wgs.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "620a03c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = pd.merge(rumen_only, wgs, on='assembly_accession', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "d7c75738",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>strain_x</th>\n",
       "      <th>host</th>\n",
       "      <th>location</th>\n",
       "      <th>mcrA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000024185.1</td>\n",
       "      <td>Methanobrevibacter ruminantium</td>\n",
       "      <td>M1</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>Germany</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001027005.1</td>\n",
       "      <td>Methanosarcina barkeri</td>\n",
       "      <td>CM1</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001563245.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>Lamb</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000711215.1</td>\n",
       "      <td>Methanomicrobium mobile</td>\n",
       "      <td>BP</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>1R26</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Denmark</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_003111625.1</td>\n",
       "      <td>Methanobrevibacter thaueri</td>\n",
       "      <td>DSM 11995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCF_000320505.1</td>\n",
       "      <td>Methanobrevibacter boviskoreani</td>\n",
       "      <td>JH1</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Korea</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCF_001729965.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>WGK6</td>\n",
       "      <td>Kangaroo</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCF_003268005.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>BMS</td>\n",
       "      <td>Steer</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCF_001548675.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>YE315</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCF_001477655.1</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_030168595.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-DZ1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCF_000404165.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>AbM4</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCF_000621965.1</td>\n",
       "      <td>Methanobrevibacter wolinii</td>\n",
       "      <td>SH</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCF_900116525.1</td>\n",
       "      <td>Methanosarcina</td>\n",
       "      <td>DSM 11855</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCF_900114585.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>DSM 16632</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_030168575.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-RCC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCF_000762265.1</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902789165.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13441</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_902799105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14438</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_902800515.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14577</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_902765875.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11103</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_902781855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12711</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902757195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10238</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GCA_017540705.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6915</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GCA_002509015.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA188</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GCA_902775725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12100</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GCA_902763115.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10830</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GCA_902758425.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10359</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GCA_902786095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13139</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GCA_902759335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10447</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>GCA_017535165.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6752</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GCA_902756745.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10188</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GCA_017524805.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6768</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>GCA_902800035.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14542</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GCA_902800615.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14594</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>GCA_002505605.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA187</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>GCA_902774605.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11987</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>GCA_902775675.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12086</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GCA_902768855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11403</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>GCA_902798395.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14364</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession                              organism_name_x   strain_x  \\\n",
       "0     GCF_000024185.1              Methanobrevibacter ruminantium          M1   \n",
       "1     GCF_001027005.1                       Methanosarcina barkeri        CM1   \n",
       "2     GCF_001563245.1                   Methanobrevibacter olleyae       YLM1   \n",
       "4     GCF_000711215.1                      Methanomicrobium mobile         BP   \n",
       "5     GCF_001481295.1              Candidatus Methanomethylophilus       1R26   \n",
       "6     GCF_003111625.1                   Methanobrevibacter thaueri  DSM 11995   \n",
       "9     GCF_000320505.1              Methanobrevibacter boviskoreani        JH1   \n",
       "10    GCF_001729965.1                               Methanosphaera       WGK6   \n",
       "11    GCF_003268005.1                               Methanosphaera        BMS   \n",
       "12    GCF_001548675.1                           Methanobrevibacter      YE315   \n",
       "13    GCF_001477655.1                  Methanobrevibacter millerae        SM9   \n",
       "14    GCA_030168595.1             Methanomassiliicoccales archaeon    LGM-DZ1   \n",
       "15    GCF_000404165.1                           Methanobrevibacter       AbM4   \n",
       "16    GCF_000621965.1                   Methanobrevibacter wolinii         SH   \n",
       "17    GCF_900116525.1                               Methanosarcina  DSM 11855   \n",
       "18    GCF_900114585.1                  Methanobrevibacter olleyae   DSM 16632   \n",
       "19    GCA_030168575.1             Methanomassiliicoccales archaeon   LGM-RCC1   \n",
       "20    GCF_000762265.1                  Methanobacterium formicicum       BRM9   \n",
       "21    GCA_902789165.1                uncultured Methanobrevibacter   RUG13441   \n",
       "22    GCA_902762085.1                uncultured Methanobrevibacter   RUG10738   \n",
       "23    GCA_902799105.1                uncultured Methanobrevibacter   RUG14438   \n",
       "24    GCA_902774685.1                uncultured Methanobrevibacter   RUG11990   \n",
       "28    GCA_902775005.1                uncultured Methanobrevibacter   RUG12025   \n",
       "30    GCA_902762095.1                uncultured Methanobrevibacter   RUG10736   \n",
       "33    GCA_902800515.1                uncultured Methanobrevibacter   RUG14577   \n",
       "35    GCA_902787415.1                        methanogenic archaeon   RUG13270   \n",
       "36    GCA_902765105.1                uncultured Methanobrevibacter   RUG11035   \n",
       "37    GCA_902763685.1                        methanogenic archaeon   RUG10898   \n",
       "39    GCA_902765875.1                uncultured Methanobrevibacter   RUG11103   \n",
       "41    GCA_902784715.1                uncultured Methanobrevibacter   RUG12995   \n",
       "43    GCA_902781855.1                uncultured Methanobrevibacter   RUG12711   \n",
       "44    GCA_902757195.1                uncultured Methanobrevibacter   RUG10238   \n",
       "47    GCA_017540705.1  Candidatus Methanomethylophilaceae archaeon   RGIG6915   \n",
       "48    GCA_002509015.1                           Methanobrevibacter     UBA188   \n",
       "49    GCA_902775725.1                uncultured Methanobrevibacter   RUG12100   \n",
       "58    GCA_902763115.1                uncultured Methanobrevibacter   RUG10830   \n",
       "59    GCA_902758425.1                uncultured Methanobrevibacter   RUG10359   \n",
       "62    GCA_902786095.1                uncultured Methanobrevibacter   RUG13139   \n",
       "63    GCA_902759335.1                uncultured Methanobrevibacter   RUG10447   \n",
       "65    GCA_017535165.1  Candidatus Methanomethylophilaceae archaeon   RGIG6752   \n",
       "66    GCA_902756745.1                uncultured Methanobrevibacter   RUG10188   \n",
       "67    GCA_902757275.1                uncultured Methanobrevibacter   RUG10244   \n",
       "69    GCA_902785855.1                uncultured Methanobrevibacter   RUG13115   \n",
       "71    GCA_017524805.1  Candidatus Methanomethylophilaceae archaeon   RGIG6768   \n",
       "72    GCA_902802065.1                uncultured Methanobrevibacter   RUG14729   \n",
       "73    GCA_902800035.1                uncultured Methanobrevibacter   RUG14542   \n",
       "74    GCA_902798335.1                uncultured Methanobrevibacter   RUG14349   \n",
       "77    GCA_902800615.1                uncultured Methanobrevibacter   RUG14594   \n",
       "78    GCA_002505605.1                           Methanobrevibacter     UBA187   \n",
       "79    GCA_902757365.1                uncultured Methanobrevibacter   RUG10252   \n",
       "83    GCA_902774605.1                uncultured Methanobrevibacter   RUG11987   \n",
       "84    GCA_902775675.1                uncultured Methanobrevibacter   RUG12086   \n",
       "85    GCA_902768855.1                uncultured Methanobrevibacter   RUG11403   \n",
       "86    GCA_902798395.1                uncultured Methanobrevibacter   RUG14364   \n",
       "88    GCA_902784195.1                uncultured Methanobrevibacter   RUG12946   \n",
       "89    GCA_902771225.1                uncultured Methanobrevibacter   RUG11657   \n",
       "\n",
       "                                                 host     location  mcrA  \n",
       "0                                              Bovine      Germany     1  \n",
       "1                                                 Cow  New Zealand     1  \n",
       "2                                                Lamb  New Zealand     2  \n",
       "4                                                 Cow    Australia     1  \n",
       "5                                                 Cow      Denmark     1  \n",
       "6                                                 Cow          NaN     1  \n",
       "9                                              Cattle        Korea     1  \n",
       "10                                           Kangaroo    Australia     1  \n",
       "11                                              Steer    Australia     1  \n",
       "12                                             Cattle    Australia     1  \n",
       "13                                             Sheep   New Zealand     2  \n",
       "14                                                NaN        China     1  \n",
       "15                                             Sheep   New Zealand     1  \n",
       "16                                             Bovine          USA     1  \n",
       "17                                             Bovine          USA     1  \n",
       "18                                            Bovine           USA     2  \n",
       "19                                                NaN        China     1  \n",
       "20                                                Cow  New Zealand     2  \n",
       "21                                                Cow     Scotland     1  \n",
       "22                                                Cow     Scotland     2  \n",
       "23                                                Cow     Scotland     1  \n",
       "24                                                Cow     Scotland     2  \n",
       "28                                                Cow     Scotland     1  \n",
       "30                                                Cow     Scotland     1  \n",
       "33                                                Cow     Scotland     2  \n",
       "35                                                Cow     Scotland     1  \n",
       "36                                                Cow     Scotland     1  \n",
       "37                                                Cow     Scotland     1  \n",
       "39                                                Cow     Scotland     1  \n",
       "41                                                Cow     Scotland     1  \n",
       "43                                                Cow     Scotland     1  \n",
       "44                                                Cow     Scotland     1  \n",
       "47  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "48                               Cattle; high methane          NaN     1  \n",
       "49                                                Cow     Scotland     1  \n",
       "58                                                Cow     Scotland     2  \n",
       "59                                                Cow     Scotland     1  \n",
       "62                                                Cow     Scotland     1  \n",
       "63                                                Cow     Scotland     1  \n",
       "65  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "66                                                Cow     Scotland     1  \n",
       "67                                                Cow     Scotland     1  \n",
       "69                                                Cow     Scotland     2  \n",
       "71  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "72                                                Cow     Scotland     1  \n",
       "73                                                Cow     Scotland     1  \n",
       "74                                                Cow     Scotland     1  \n",
       "77                                                Cow     Scotland     1  \n",
       "78                               Cattle; high methane          NaN     1  \n",
       "79                                                Cow     Scotland     1  \n",
       "83                                                Cow     Scotland     1  \n",
       "84                                                Cow     Scotland     1  \n",
       "85                                                Cow     Scotland     1  \n",
       "86                                                Cow     Scotland     1  \n",
       "88                                                Cow     Scotland     1  \n",
       "89                                                Cow     Scotland     1  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out['mcrA'] = out['mcrA'].fillna(0).astype(int)\n",
    "out_mcrA = out[out['mcrA'] >= 1]\n",
    "out_mcrA = out_mcrA[['assembly_accession', 'organism_name_x', 'strain_x','host', 'location', 'mcrA']]\n",
    "out_mcrA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "52f52ca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_mcrA.to_csv('out_mcrA.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "4a99dc7c",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/Documents/asmg_labs_files/fasta_sequences/mcrA/out_mcrA.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[76], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m two_only \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m/Documents/asmg_labs_files/fasta_sequences/mcrA/out_mcrA.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/util/_decorators.py:211\u001b[0m, in \u001b[0;36mdeprecate_kwarg.<locals>._deprecate_kwarg.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    209\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    210\u001b[0m         kwargs[new_arg_name] \u001b[38;5;241m=\u001b[39m new_arg_value\n\u001b[0;32m--> 211\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/util/_decorators.py:331\u001b[0m, in \u001b[0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    325\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) \u001b[38;5;241m>\u001b[39m num_allow_args:\n\u001b[1;32m    326\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    327\u001b[0m         msg\u001b[38;5;241m.\u001b[39mformat(arguments\u001b[38;5;241m=\u001b[39m_format_argument_list(allow_args)),\n\u001b[1;32m    328\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[1;32m    329\u001b[0m         stacklevel\u001b[38;5;241m=\u001b[39mfind_stack_level(),\n\u001b[1;32m    330\u001b[0m     )\n\u001b[0;32m--> 331\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/io/parsers/readers.py:950\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    936\u001b[0m     dialect,\n\u001b[1;32m    937\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    946\u001b[0m     defaults\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdelimiter\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m},\n\u001b[1;32m    947\u001b[0m )\n\u001b[1;32m    948\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 950\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/io/parsers/readers.py:605\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    602\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    604\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 605\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    607\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    608\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1442\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1439\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1441\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1442\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1735\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1733\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1734\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1735\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1736\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1737\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1738\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1739\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1740\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1741\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1742\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1743\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1744\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1745\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1746\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/io/common.py:856\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    851\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    852\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    853\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    854\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    855\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 856\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    857\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    858\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    859\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    860\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    861\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    862\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    863\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    864\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    865\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/Documents/asmg_labs_files/fasta_sequences/mcrA/out_mcrA.csv'"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01dd3668",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
