{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8270eb45",
   "metadata": {},
   "source": [
    "This finds the mrt sequences from protein.faa files. This is primarily for those sequences from ncbi where the gff files did not indicate a distinction between mcr and mrt genes."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aa0ed9",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a84427e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/gracedai/Documents/MethanogeneDB\n"
     ]
    }
   ],
   "source": [
    "#Check that we are in the correct working directory\n",
    "import os \n",
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d4ad666f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving notices: ...working... done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.3.1\n",
      "  latest version: 23.5.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.5.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - sh\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    certifi-2023.5.7           |  py310hca03da5_0         153 KB\n",
      "    openssl-1.1.1u             |       h1a28f6b_0         2.6 MB\n",
      "    sh-1.14.2                  |  py310hca03da5_0          67 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         2.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  sh                 pkgs/main/osx-arm64::sh-1.14.2-py310hca03da5_0 \n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    anaconda::ca-certificates-2022.4.26-h~ --> pkgs/main::ca-certificates-2023.05.30-hca03da5_0 \n",
      "  certifi            anaconda::certifi-2022.6.15-py310hca0~ --> pkgs/main::certifi-2023.5.7-py310hca03da5_0 \n",
      "  openssl                                 1.1.1t-h1a28f6b_0 --> 1.1.1u-h1a28f6b_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "certifi-2023.5.7     | 153 KB    |                                       |   0% \n",
      "openssl-1.1.1u       | 2.6 MB    |                                       |   0% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     |                                       |   0% \u001b[A\u001b[A\n",
      "certifi-2023.5.7     | 153 KB    | ##################################### | 100% \u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     | ########8                             |  24% \u001b[A\u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: \\ WARNING conda.core.path_actions:verify(1093): Unable to create environments file. Path not writable.\n",
      "  environment location: /Users/gracedai/.conda/environments.txt\n",
      "\n",
      "done\n",
      "Executing transaction: / WARNING conda.core.envs_manager:register_env(49): Unable to register environment. Path not writable or missing.\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "  registry file: /Users/gracedai/.conda/environments.txt\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# Install a conda package in the current Jupyter kernel\n",
    "import sys\n",
    "!conda install --yes --prefix {sys.prefix} sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "df450b61",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/gracedai/Documents/MethanogeneDB\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "0c2c402a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#unzips a file name, given the path, and deletes the zipped version\n",
    "from sh import gunzip\n",
    "#this line is for testing purposes\n",
    "#gunzip('GCF_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "01379dfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#unzip whatever files needed\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gz\"):\n",
    "            #os.remove(os.path.join(root, file))\n",
    "            gunzip(os.path.join(root, file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f642bd3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/gracedai/Documents/MethanogeneDB/GCA_000970225.1/GCF_000970225.1_ASM97022v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000092305.1/GCF_000092305.1_ASM9230v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970165.1/GCF_000970165.1_ASM97016v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170265.1/GCF_024170265.1_ASM2417026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502925.1/GCA_002502925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000091665.1/GCF_000091665.1_ASM9166v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_020885915.1/GCF_020885915.1_ASM2088591v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017535165.1/GCA_017535165.1_ASM1753516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000166095.1/GCF_000166095.1_ASM16609v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412935.1/GCA_017412935.1_ASM1741293v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508885.1/GCA_002508885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633965.1/GCF_029633965.1_ASM2963396v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114585.1/GCF_900114585.1_IMG-taxon_2593339150_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744315.1/GCF_000744315.1_ASM74431v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_028743435.1/GCF_028743435.1_ASM2874343v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002356395.1/GCA_002356395.1_ASM235639v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168575.1/GCA_030168575.1_ASM3016857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970325.1/GCF_000970325.1_ASM97032v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502965.1/GCA_002502965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970265.1/GCF_000970265.1_ASM97026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025525.1/GCF_000025525.1_ASM2552v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001729965.1/GCF_001729965.1_ASM172996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900215215.1/GCF_900215215.1_IMG-taxon_2703719067_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902762095.1/GCA_902762095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000746075.1/GCF_000746075.1_ASM74607v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000196655.1/GCF_000196655.1_ASM19665v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013725.1/GCF_000013725.1_ASM1372v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_946463545.1/GCF_946463545.1_Mth-DSM-2095-CCS-WG_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003111625.1/GCF_003111625.1_ASM311162v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002252585.1/GCF_002252585.1_ASM225258v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017935945.1/GCA_017935945.1_ASM1793594v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_021184045.1/GCF_021184045.1_ASM2118404v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017539065.1/GCA_017539065.1_ASM1753906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503105.1/GCA_002503105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000217995.1/GCF_000217995.1_ASM21799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017447225.1/GCA_017447225.1_ASM1744722v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000251105.1/GCF_000251105.1_ASM25110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711905.1/GCF_000711905.1_ASM71190v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011005.1/GCF_000011005.1_ASM1100v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024167805.1/GCA_024167805.1_ASM2416780v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495885.1/GCA_002495885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024808815.1/GCF_024808815.1_ASM2480881v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017540705.1/GCA_017540705.1_ASM1754070v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025397995.1/GCF_025397995.1_ASM2539799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000621965.1/GCF_000621965.1_ASM62196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813675.1/GCF_002813675.1_ASM281367v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017432245.1/GCA_017432245.1_ASM1743224v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468935.1/GCA_017468935.1_ASM1746893v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873415.1/GCF_017873415.1_ASM1787341v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744455.1/GCF_000744455.1_ASM74445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969925.1/GCF_000969925.1_ASM96992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000327485.1/GCF_000327485.1_ASM32748v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902763685.1/GCA_902763685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496295.1/GCA_002496295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900111645.1/GCF_900111645.1_IMG-taxon_2642422537_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024625.1/GCF_000024625.1_ASM2462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900109595.1/GCF_900109595.1_IMG-taxon_2654588138_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508925.1/GCA_002508925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584625.1/GCA_003584625.1_ASM358462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017504565.1/GCA_017504565.1_ASM1750456v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026016205.1/GCA_026016205.1_ASM2601620v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002287175.1/GCF_002287175.1_ASM228717v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151225.1/GCA_000151225.1_ASM15122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902795935.1/GCA_902795935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875315.1/GCF_017875315.1_ASM1787531v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555045.1/GCF_027555045.1_ASM2755504v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874455.1/GCF_017874455.1_ASM1787445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563305.1/GCA_001563305.1_ASM156330v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506325.1/GCA_002506325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508825.1/GCA_002508825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017399555.1/GCA_017399555.1_ASM1739955v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000306725.1/GCA_000306725.1_ASM30672v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506905.1/GCA_002506905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827225.1/GCF_902827225.1_Methanocaldococcus_lauensis_SG7_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214725.1/GCF_000214725.1_ASM21472v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009914355.1/GCF_009914355.1_ASM991435v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491285.1/GCA_003491285.1_ASM349128v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017512765.1/GCA_017512765.1_ASM1751276v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001481295.1/GCF_001481295.1_ASM148129v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509405.1/GCA_002509405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015765.1/GCF_000015765.1_ASM1576v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000300255.2/GCF_000300255.2_ASM30025v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171695.1/GCA_905171695.1_GoM-Arc1_E37_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000014945.1/GCF_000014945.1_ASM1494v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496945.1/GCA_002496945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970185.1/GCF_000970185.1_ASM97018v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000220645.1/GCF_000220645.1_ASM22064v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168595.1/GCA_030168595.1_ASM3016859v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002973515.1/GCF_002973515.1_ASM297351v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001560915.1/GCA_001560915.1_ASM156091v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000179575.2/GCF_000179575.2_ASM17957v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509645.1/GCA_002509645.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001889405.1/GCF_001889405.1_ASM188940v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017549625.1/GCA_017549625.1_ASM1754962v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029854155.1/GCF_029854155.1_ASM2985415v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000235565.1/GCF_000235565.1_ASM23556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813695.1/GCF_002813695.1_ASM281369v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247815.1/GCA_030247815.1_ASM3024781v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003052125.1/GCF_003052125.1_ASM305212v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171685.1/GCA_905171685.1_GoM-Arc1_E50_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900177455.1/GCF_900177455.1_IMG-taxon_2706794887_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828535.1/GCF_000828535.1_ASM82853v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017461225.1/GCA_017461225.1_ASM1746122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017524805.1/GCA_017524805.1_ASM1752480v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017508065.1/GCA_017508065.1_ASM1750806v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017408885.1/GCA_017408885.1_ASM1740888v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018141185.1/GCF_018141185.1_ASM1814118v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509655.1/GCA_002509655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023016325.1/GCF_023016325.1_ASM2301632v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000021965.1/GCF_000021965.1_ASM2196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505245.1/GCA_002505245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009917665.1/GCF_009917665.1_ASM991766v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404225.1/GCF_000404225.1_ASM40422v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508935.1/GCA_002508935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813085.1/GCF_002813085.1_ASM281308v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969885.1/GCF_000969885.1_ASM96988v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828575.1/GCF_000828575.1_ASM82857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498705.1/GCA_002498705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095295.1/GCF_900095295.1_MCBB_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900100715.1/GCF_900100715.1_IMG-taxon_2642422561_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434555.1/GCA_017434555.1_ASM1743455v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498605.1/GCA_002498605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900101055.1/GCF_900101055.1_IMG-taxon_2634166298_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026625165.1/GCA_026625165.1_ASM2662516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002504405.1/GCA_002504405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506565.1/GCA_002506565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505345.1/GCA_002505345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_016907005.1/GCF_016907005.1_ASM1690700v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114835.1/GCF_900114835.1_IMG-taxon_2642422540_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900116525.1/GCF_900116525.1_IMG-taxon_2595698250_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002945325.1/GCF_002945325.1_ASM294532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509425.1/GCA_002509425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011585.1/GCF_000011585.1_ASM1158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902782945.1/GCA_902782945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017537385.1/GCA_017537385.1_ASM1753738v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023167465.1/GCF_023167465.1_ASM2316746v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017625.1/GCF_000017625.1_ASM1762v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494805.1/GCA_002494805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669925.1/GCF_019669925.1_ASM1966992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017622775.1/GCA_017622775.1_ASM1762277v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019263745.1/GCF_019263745.1_ASM1926374v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902775005.1/GCA_902775005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498365.1/GCA_002498365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434525.1/GCA_017434525.1_ASM1743452v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151245.1/GCA_000151245.1_ASM15124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508845.1/GCA_002508845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505275.1/GCA_002505275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508905.1/GCA_002508905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000328665.1/GCF_000328665.1_ASM32866v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553445.1/GCA_017553445.1_ASM1755344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669965.1/GCF_019669965.1_ASM1966996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498805.1/GCA_002498805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503785.1/GCA_002503785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016125.1/GCF_000016125.1_ASM1612v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875815.1/GCF_017875815.1_ASM1787581v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584605.1/GCA_003584605.1_ASM358460v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019262145.1/GCA_019262145.1_ASM1926214v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900103415.1/GCF_900103415.1_IMG-taxon_2593339167_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875625.1/GCF_017875625.1_ASM1787562v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875565.1/GCF_017875565.1_ASM1787556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555065.1/GCF_027555065.1_ASM2755506v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017357525.1/GCF_017357525.1_ASM1735752v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000350305.1/GCA_000350305.1_ASM35030v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827205.1/GCA_902827205.1_Methanocaldococcus_lauensis_SG1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247745.1/GCA_030247745.1_ASM3024774v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508625.1/GCA_002508625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468685.1/GCA_017468685.1_ASM1746868v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828555.1/GCF_000828555.1_ASM82855v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000275865.1/GCA_000275865.1_ASM27586v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015825.1/GCF_000015825.1_ASM1582v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027554905.1/GCF_027554905.1_ASM2755490v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875325.1/GCF_017875325.1_ASM1787532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024662215.1/GCF_024662215.1_ASM2466221v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016525.1/GCF_000016525.1_ASM1652v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026684035.1/GCF_026684035.1_ASM2668403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553245.1/GCA_017553245.1_ASM1755324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000063445.1/GCF_000063445.1_ASM6344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498765.1/GCA_002498765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874375.1/GCF_017874375.1_ASM1787437v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495325.1/GCA_002495325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000739065.1/GCF_000739065.1_ASM73906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498425.1/GCA_002498425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846155.1/GCF_022846155.1_ASM2284615v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017561385.1/GCA_017561385.1_ASM1756138v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498375.1/GCA_002498375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007185.1/GCF_000007185.1_ASM718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017165.1/GCF_000017165.1_ASM1716v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003711245.1/GCF_003711245.1_ASM371124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000147875.1/GCF_000147875.1_ASM14787v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000745485.1/GCF_000745485.1_ASM74548v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508855.1/GCA_002508855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214415.1/GCF_000214415.1_ASM21441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013388255.1/GCF_013388255.1_ASM1338825v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000376965.1/GCF_000376965.1_ASM37696v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902383905.1/GCF_902383905.1_UHGG_MGYG-HGUT-02160_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495685.1/GCA_002495685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503135.1/GCA_002503135.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970245.1/GCF_000970245.1_ASM97024v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000304355.2/GCF_000304355.2_Mb_MS2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013445.1/GCF_000013445.1_ASM1344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017411925.1/GCA_017411925.1_ASM1741192v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873625.1/GCF_017873625.1_ASM1787362v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711215.1/GCF_000711215.1_ASM71121v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491305.1/GCA_003491305.1_ASM349130v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000320505.1/GCF_000320505.1_ASM32050v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017407825.1/GCA_017407825.1_ASM1740782v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970005.1/GCF_000970005.1_ASM97000v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761105.1/GCF_013761105.1_ASM1376110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970205.1/GCF_000970205.1_ASM97020v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_004363215.1/GCF_004363215.1_ASM436321v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024807035.1/GCF_024807035.1_ASM2480703v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017519045.1/GCA_017519045.1_ASM1751904v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761045.1/GCF_013761045.1_ASM1376104v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873855.1/GCF_017873855.1_ASM1787385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001304615.2/GCF_001304615.2_ASM130461v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412625.1/GCA_017412625.1_ASM1741262v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023617305.1/GCF_023617305.1_ASM2361730v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970045.1/GCF_000970045.1_ASM97004v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009911715.1/GCA_009911715.1_ASM991171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025449255.1/GCF_025449255.1_ASM2544925v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875395.1/GCF_017875395.1_ASM1787539v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025914035.1/GCF_025914035.1_ASM2591403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003385755.1/GCF_003385755.1_ASM338575v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017633855.1/GCA_017633855.1_ASM1763385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000191585.1/GCF_000191585.1_ASM19158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761055.1/GCF_013761055.1_ASM1376105v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007065.1/GCF_000007065.1_ASM706v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498785.1/GCA_002498785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900079125.1/GCA_900079125.1_MBBA_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813655.1/GCF_002813655.1_ASM281365v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017940815.1/GCA_017940815.1_ASM1794081v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017387505.1/GCA_017387505.1_ASM1738750v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000204415.1/GCF_000204415.1_ASM20441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506995.1/GCA_002506995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000800805.1/GCF_000800805.1_ASM80080v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003351865.1/GCF_003351865.1_ASM335186v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017410345.1/GCA_017410345.1_ASM1741034v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494585.1/GCA_002494585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017185.1/GCF_000017185.1_ASM1718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017473945.1/GCA_017473945.1_ASM1747394v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502465.1/GCA_002502465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170505.1/GCF_024170505.1_ASM2417050v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017409525.1/GCA_017409525.1_ASM1740952v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001458655.1/GCF_001458655.1_Mb9_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017406825.1/GCA_017406825.1_ASM1740682v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000008645.1/GCF_000008645.1_ASM864v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000145295.1/GCF_000145295.1_ASM14529v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900036045.1/GCF_900036045.1_Methanoculleus_sp_MAB1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900106905.1/GCF_900106905.1_IMG-taxon_2642422606_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000023985.1/GCF_000023985.1_ASM2398v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014203295.1/GCF_014203295.1_ASM1420329v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000341715.1/GCF_000341715.1_ASM34171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018502485.1/GCF_018502485.1_ASM1850248v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633915.1/GCF_029633915.1_ASM2963391v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_008033705.1/GCF_008033705.1_ASM803370v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014207315.1/GCF_014207315.1_ASM1420731v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902787415.1/GCA_902787415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025865.1/GCF_000025865.1_ASM2586v1_genomic.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "275"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# list to store paths to gff files\n",
    "faa_files = []\n",
    "# iterate through current working directory and for each file \n",
    "# unzip the file and append the path to a list \n",
    "#run this twice ?\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\"protein.faa\"):\n",
    "            faa_files.append(os.path.join(root, file))\n",
    "print(faa_files)\n",
    "len(faa_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a4818c09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCA_000970225.1/GCF_000970225.1_ASM97022v1_genomic.gff', 'GCA_000092305.1/GCF_000092305.1_ASM9230v1_genomic.gff', 'GCA_000970165.1/GCF_000970165.1_ASM97016v1_genomic.gff', 'GCA_024170265.1/GCF_024170265.1_ASM2417026v1_genomic.gff', 'GCA_002502925.1/GCA_002502925.1.gff', 'GCA_000091665.1/GCF_000091665.1_ASM9166v1_genomic.gff', 'GCA_020885915.1/GCF_020885915.1_ASM2088591v1_genomic.gff', 'GCA_017535165.1/GCA_017535165.1_ASM1753516v1_genomic.gff', 'GCA_000166095.1/GCF_000166095.1_ASM16609v1_genomic.gff', 'GCA_017412935.1/GCA_017412935.1_ASM1741293v1_genomic.gff', 'GCA_002508885.1/GCA_002508885.1.gff', 'GCA_029633965.1/GCF_029633965.1_ASM2963396v1_genomic.gff', 'GCA_900114585.1/GCF_900114585.1_IMG-taxon_2593339150_annotated_assembly_genomic.gff', 'GCA_000744315.1/GCF_000744315.1_ASM74431v1_genomic.gff', 'GCA_028743435.1/GCF_028743435.1_ASM2874343v1_genomic.gff', 'GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', 'GCA_002356395.1/GCA_002356395.1_ASM235639v1_genomic.gff', 'GCA_030168575.1/GCA_030168575.1_ASM3016857v1_genomic.gff', 'GCA_000970325.1/GCF_000970325.1_ASM97032v1_genomic.gff', 'GCA_002502965.1/GCA_002502965.1.gff', 'GCA_000970265.1/GCF_000970265.1_ASM97026v1_genomic.gff', 'GCA_000025525.1/GCF_000025525.1_ASM2552v1_genomic.gff', 'GCA_001729965.1/GCF_001729965.1_ASM172996v1_genomic.gff', 'GCA_900215215.1/GCF_900215215.1_IMG-taxon_2703719067_annotated_assembly_genomic.gff', 'GCA_902762095.1/GCA_902762095.1.gff', 'GCA_000746075.1/GCF_000746075.1_ASM74607v1_genomic.gff', 'GCA_000196655.1/GCF_000196655.1_ASM19665v1_genomic.gff', 'GCA_000013725.1/GCF_000013725.1_ASM1372v1_genomic.gff', 'GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff', 'GCA_946463545.1/GCF_946463545.1_Mth-DSM-2095-CCS-WG_genomic.gff', 'GCA_003111625.1/GCF_003111625.1_ASM311162v1_genomic.gff', 'GCA_002252585.1/GCF_002252585.1_ASM225258v1_genomic.gff', 'GCA_017935945.1/GCA_017935945.1_ASM1793594v1_genomic.gff', 'GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff', 'GCA_021184045.1/GCF_021184045.1_ASM2118404v1_genomic.gff', 'GCA_017539065.1/GCA_017539065.1_ASM1753906v1_genomic.gff', 'GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff', 'GCA_002503105.1/GCA_002503105.1.gff', 'GCA_000217995.1/GCF_000217995.1_ASM21799v1_genomic.gff', 'GCA_017447225.1/GCA_017447225.1_ASM1744722v1_genomic.gff', 'GCA_000251105.1/GCF_000251105.1_ASM25110v1_genomic.gff', 'GCA_000711905.1/GCF_000711905.1_ASM71190v1_genomic.gff', 'GCA_000011005.1/GCF_000011005.1_ASM1100v1_genomic.gff', 'GCA_024167805.1/GCA_024167805.1_ASM2416780v1_genomic.gff', 'GCA_002495885.1/GCA_002495885.1.gff', 'GCA_024808815.1/GCF_024808815.1_ASM2480881v1_genomic.gff', 'GCA_017540705.1/GCA_017540705.1_ASM1754070v1_genomic.gff', 'GCA_025397995.1/GCF_025397995.1_ASM2539799v1_genomic.gff', 'GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff', 'GCA_000621965.1/GCF_000621965.1_ASM62196v1_genomic.gff', 'GCA_002813675.1/GCF_002813675.1_ASM281367v1_genomic.gff', 'GCA_017432245.1/GCA_017432245.1_ASM1743224v1_genomic.gff', 'GCA_017468935.1/GCA_017468935.1_ASM1746893v1_genomic.gff', 'GCA_017873415.1/GCF_017873415.1_ASM1787341v1_genomic.gff', 'GCA_000744455.1/GCF_000744455.1_ASM74445v1_genomic.gff', 'GCA_000969925.1/GCF_000969925.1_ASM96992v1_genomic.gff', 'GCA_000327485.1/GCF_000327485.1_ASM32748v1_genomic.gff', 'GCA_902763685.1/GCA_902763685.1.gff', 'GCA_002496295.1/GCA_002496295.1.gff', 'GCA_900111645.1/GCF_900111645.1_IMG-taxon_2642422537_annotated_assembly_genomic.gff', 'GCA_000024625.1/GCF_000024625.1_ASM2462v1_genomic.gff', 'GCA_900109595.1/GCF_900109595.1_IMG-taxon_2654588138_annotated_assembly_genomic.gff', 'GCA_002508925.1/GCA_002508925.1.gff', 'GCA_003584625.1/GCA_003584625.1_ASM358462v1_genomic.gff', 'GCA_017504565.1/GCA_017504565.1_ASM1750456v1_genomic.gff', 'GCA_026016205.1/GCA_026016205.1_ASM2601620v1_genomic.gff', 'GCA_002287175.1/GCF_002287175.1_ASM228717v1_genomic.gff', 'GCA_000151225.1/GCA_000151225.1_ASM15122v1_genomic.gff', 'GCA_902795935.1/GCA_902795935.1.gff', 'GCA_017875315.1/GCF_017875315.1_ASM1787531v1_genomic.gff', 'GCA_027555045.1/GCF_027555045.1_ASM2755504v1_genomic.gff', 'GCA_017874455.1/GCF_017874455.1_ASM1787445v1_genomic.gff', 'GCA_001563305.1/GCA_001563305.1_ASM156330v1_genomic.gff', 'GCA_002506325.1/GCA_002506325.1.gff', 'GCA_002508825.1/GCA_002508825.1.gff', 'GCA_017399555.1/GCA_017399555.1_ASM1739955v1_genomic.gff', 'GCA_000306725.1/GCA_000306725.1_ASM30672v1_genomic.gff', 'GCA_002506905.1/GCA_002506905.1.gff', 'GCA_902827225.1/GCF_902827225.1_Methanocaldococcus_lauensis_SG7_genomic.gff', 'GCA_000214725.1/GCF_000214725.1_ASM21472v1_genomic.gff', 'GCA_009914355.1/GCF_009914355.1_ASM991435v1_genomic.gff', 'GCA_003491285.1/GCA_003491285.1_ASM349128v1_genomic.gff', 'GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', 'GCA_017512765.1/GCA_017512765.1_ASM1751276v1_genomic.gff', 'GCA_001481295.1/GCF_001481295.1_ASM148129v1_genomic.gff', 'GCA_002509405.1/GCA_002509405.1.gff', 'GCA_000015765.1/GCF_000015765.1_ASM1576v1_genomic.gff', 'GCA_000300255.2/GCF_000300255.2_ASM30025v2_genomic.gff', 'GCA_905171695.1/GCA_905171695.1_GoM-Arc1_E37_genomic.gff', 'GCA_000014945.1/GCF_000014945.1_ASM1494v1_genomic.gff', 'GCA_002496945.1/GCA_002496945.1.gff', 'GCA_000970185.1/GCF_000970185.1_ASM97018v1_genomic.gff', 'GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff', 'GCA_000220645.1/GCF_000220645.1_ASM22064v1_genomic.gff', 'GCA_030168595.1/GCA_030168595.1_ASM3016859v1_genomic.gff', 'GCA_002973515.1/GCF_002973515.1_ASM297351v1_genomic.gff', 'GCA_001560915.1/GCA_001560915.1_ASM156091v1_genomic.gff', 'GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff', 'GCA_000179575.2/GCF_000179575.2_ASM17957v2_genomic.gff', 'GCA_002509645.1/GCA_002509645.1.gff', 'GCA_001889405.1/GCF_001889405.1_ASM188940v1_genomic.gff', 'GCA_017549625.1/GCA_017549625.1_ASM1754962v1_genomic.gff', 'GCA_029854155.1/GCF_029854155.1_ASM2985415v1_genomic.gff', 'GCA_000235565.1/GCF_000235565.1_ASM23556v1_genomic.gff', 'GCA_002813695.1/GCF_002813695.1_ASM281369v1_genomic.gff', 'GCA_030247815.1/GCA_030247815.1_ASM3024781v1_genomic.gff', 'GCA_003052125.1/GCF_003052125.1_ASM305212v1_genomic.gff', 'GCA_905171685.1/GCA_905171685.1_GoM-Arc1_E50_genomic.gff', 'GCA_900177455.1/GCF_900177455.1_IMG-taxon_2706794887_annotated_assembly_genomic.gff', 'GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff', 'GCA_000828535.1/GCF_000828535.1_ASM82853v1_genomic.gff', 'GCA_017461225.1/GCA_017461225.1_ASM1746122v1_genomic.gff', 'GCA_017524805.1/GCA_017524805.1_ASM1752480v1_genomic.gff', 'GCA_017508065.1/GCA_017508065.1_ASM1750806v1_genomic.gff', 'GCA_017408885.1/GCA_017408885.1_ASM1740888v1_genomic.gff', 'GCA_018141185.1/GCF_018141185.1_ASM1814118v1_genomic.gff', 'GCA_002509655.1/GCA_002509655.1.gff', 'GCA_023016325.1/GCF_023016325.1_ASM2301632v1_genomic.gff', 'GCA_000021965.1/GCF_000021965.1_ASM2196v1_genomic.gff', 'GCA_002505245.1/GCA_002505245.1.gff', 'GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff', 'GCA_009917665.1/GCF_009917665.1_ASM991766v1_genomic.gff', 'GCA_000404225.1/GCF_000404225.1_ASM40422v1_genomic.gff', 'GCA_002508935.1/GCA_002508935.1.gff', 'GCA_002813085.1/GCF_002813085.1_ASM281308v1_genomic.gff', 'GCA_000969885.1/GCF_000969885.1_ASM96988v1_genomic.gff', 'GCA_000828575.1/GCF_000828575.1_ASM82857v1_genomic.gff', 'GCA_002498705.1/GCA_002498705.1.gff', 'GCA_900095295.1/GCF_900095295.1_MCBB_genomic.gff', 'GCA_900100715.1/GCF_900100715.1_IMG-taxon_2642422561_annotated_assembly_genomic.gff', 'GCA_017434555.1/GCA_017434555.1_ASM1743455v1_genomic.gff', 'GCA_002498605.1/GCA_002498605.1.gff', 'GCA_900101055.1/GCF_900101055.1_IMG-taxon_2634166298_annotated_assembly_genomic.gff', 'GCA_026625165.1/GCA_026625165.1_ASM2662516v1_genomic.gff', 'GCA_002504405.1/GCA_002504405.1.gff', 'GCA_002506565.1/GCA_002506565.1.gff', 'GCA_002505345.1/GCA_002505345.1.gff', 'GCA_016907005.1/GCF_016907005.1_ASM1690700v1_genomic.gff', 'GCA_900114835.1/GCF_900114835.1_IMG-taxon_2642422540_annotated_assembly_genomic.gff', 'GCA_900116525.1/GCF_900116525.1_IMG-taxon_2595698250_annotated_assembly_genomic.gff', 'GCA_002945325.1/GCF_002945325.1_ASM294532v1_genomic.gff', 'GCA_002509425.1/GCA_002509425.1.gff', 'GCA_000011585.1/GCF_000011585.1_ASM1158v1_genomic.gff', 'GCA_902782945.1/GCA_902782945.1.gff', 'GCA_017537385.1/GCA_017537385.1_ASM1753738v1_genomic.gff', 'GCA_023167465.1/GCF_023167465.1_ASM2316746v1_genomic.gff', 'GCA_000017625.1/GCF_000017625.1_ASM1762v1_genomic.gff', 'GCA_002494805.1/GCA_002494805.1.gff', 'GCA_019669925.1/GCF_019669925.1_ASM1966992v1_genomic.gff', 'GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff', 'GCA_017622775.1/GCA_017622775.1_ASM1762277v1_genomic.gff', 'GCA_019263745.1/GCF_019263745.1_ASM1926374v1_genomic.gff', 'GCA_902775005.1/GCA_902775005.1.gff', 'GCA_002498365.1/GCA_002498365.1.gff', 'GCA_017434525.1/GCA_017434525.1_ASM1743452v1_genomic.gff', 'GCA_000151245.1/GCA_000151245.1_ASM15124v1_genomic.gff', 'GCA_002508845.1/GCA_002508845.1.gff', 'GCA_002505275.1/GCA_002505275.1.gff', 'GCA_002508905.1/GCA_002508905.1.gff', 'GCA_000328665.1/GCF_000328665.1_ASM32866v1_genomic.gff', 'GCA_017553445.1/GCA_017553445.1_ASM1755344v1_genomic.gff', 'GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff', 'GCA_019669965.1/GCF_019669965.1_ASM1966996v1_genomic.gff', 'GCA_002498805.1/GCA_002498805.1.gff', 'GCA_002503785.1/GCA_002503785.1.gff', 'GCA_000016125.1/GCF_000016125.1_ASM1612v1_genomic.gff', 'GCA_017875815.1/GCF_017875815.1_ASM1787581v1_genomic.gff', 'GCA_003584605.1/GCA_003584605.1_ASM358460v1_genomic.gff', 'GCA_019262145.1/GCA_019262145.1_ASM1926214v1_genomic.gff', 'GCA_900103415.1/GCF_900103415.1_IMG-taxon_2593339167_annotated_assembly_genomic.gff', 'GCA_017875625.1/GCF_017875625.1_ASM1787562v1_genomic.gff', 'GCA_017875565.1/GCF_017875565.1_ASM1787556v1_genomic.gff', 'GCA_027555065.1/GCF_027555065.1_ASM2755506v1_genomic.gff', 'GCA_017357525.1/GCF_017357525.1_ASM1735752v1_genomic.gff', 'GCA_000350305.1/GCA_000350305.1_ASM35030v1_genomic.gff', 'GCA_902827205.1/GCA_902827205.1_Methanocaldococcus_lauensis_SG1_genomic.gff', 'GCA_030247745.1/GCA_030247745.1_ASM3024774v1_genomic.gff', 'GCA_002508625.1/GCA_002508625.1.gff', 'GCA_017468685.1/GCA_017468685.1_ASM1746868v1_genomic.gff', 'GCA_000828555.1/GCF_000828555.1_ASM82855v1_genomic.gff', 'GCA_000275865.1/GCA_000275865.1_ASM27586v1_genomic.gff', 'GCA_000015825.1/GCF_000015825.1_ASM1582v1_genomic.gff', 'GCA_027554905.1/GCF_027554905.1_ASM2755490v1_genomic.gff', 'GCA_017875325.1/GCF_017875325.1_ASM1787532v1_genomic.gff', 'GCA_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff', 'GCA_024662215.1/GCF_024662215.1_ASM2466221v1_genomic.gff', 'GCA_000016525.1/GCF_000016525.1_ASM1652v1_genomic.gff', 'GCA_026684035.1/GCF_026684035.1_ASM2668403v1_genomic.gff', 'GCA_017553245.1/GCA_017553245.1_ASM1755324v1_genomic.gff', 'GCA_000063445.1/GCF_000063445.1_ASM6344v1_genomic.gff', 'GCA_002498765.1/GCA_002498765.1.gff', 'GCA_017874375.1/GCF_017874375.1_ASM1787437v1_genomic.gff', 'GCA_002495325.1/GCA_002495325.1.gff', 'GCA_000739065.1/GCF_000739065.1_ASM73906v1_genomic.gff', 'GCA_002498425.1/GCA_002498425.1.gff', 'GCA_022846155.1/GCF_022846155.1_ASM2284615v1_genomic.gff', 'GCA_017561385.1/GCA_017561385.1_ASM1756138v1_genomic.gff', 'GCA_002498375.1/GCA_002498375.1.gff', 'GCA_000007185.1/GCF_000007185.1_ASM718v1_genomic.gff', 'GCA_000017165.1/GCF_000017165.1_ASM1716v1_genomic.gff', 'GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', 'GCA_003711245.1/GCF_003711245.1_ASM371124v1_genomic.gff', 'GCA_000147875.1/GCF_000147875.1_ASM14787v1_genomic.gff', 'GCA_000745485.1/GCF_000745485.1_ASM74548v1_genomic.gff', 'GCA_002508855.1/GCA_002508855.1.gff', 'GCA_000214415.1/GCF_000214415.1_ASM21441v1_genomic.gff', 'GCA_013388255.1/GCF_013388255.1_ASM1338825v1_genomic.gff', 'GCA_000376965.1/GCF_000376965.1_ASM37696v1_genomic.gff', 'GCA_902383905.1/GCF_902383905.1_UHGG_MGYG-HGUT-02160_genomic.gff', 'GCA_002495685.1/GCA_002495685.1.gff', 'GCA_002503135.1/GCA_002503135.1.gff', 'GCA_000970245.1/GCF_000970245.1_ASM97024v1_genomic.gff', 'GCA_000304355.2/GCF_000304355.2_Mb_MS2_genomic.gff', 'GCA_000013445.1/GCF_000013445.1_ASM1344v1_genomic.gff', 'GCA_017411925.1/GCA_017411925.1_ASM1741192v1_genomic.gff', 'GCA_017873625.1/GCF_017873625.1_ASM1787362v1_genomic.gff', 'GCA_000711215.1/GCF_000711215.1_ASM71121v1_genomic.gff', 'GCA_003491305.1/GCA_003491305.1_ASM349130v1_genomic.gff', 'GCA_000320505.1/GCF_000320505.1_ASM32050v2_genomic.gff', 'GCA_017407825.1/GCA_017407825.1_ASM1740782v1_genomic.gff', 'GCA_000970005.1/GCF_000970005.1_ASM97000v1_genomic.gff', 'GCA_013761105.1/GCF_013761105.1_ASM1376110v1_genomic.gff', 'GCA_000970205.1/GCF_000970205.1_ASM97020v1_genomic.gff', 'GCA_004363215.1/GCF_004363215.1_ASM436321v1_genomic.gff', 'GCA_024807035.1/GCF_024807035.1_ASM2480703v1_genomic.gff', 'GCA_017519045.1/GCA_017519045.1_ASM1751904v1_genomic.gff', 'GCA_013761045.1/GCF_013761045.1_ASM1376104v1_genomic.gff', 'GCA_017873855.1/GCF_017873855.1_ASM1787385v1_genomic.gff', 'GCA_001304615.2/GCF_001304615.2_ASM130461v2_genomic.gff', 'GCA_017412625.1/GCA_017412625.1_ASM1741262v1_genomic.gff', 'GCA_023617305.1/GCF_023617305.1_ASM2361730v1_genomic.gff', 'GCA_000970045.1/GCF_000970045.1_ASM97004v1_genomic.gff', 'GCA_009911715.1/GCA_009911715.1_ASM991171v1_genomic.gff', 'GCA_025449255.1/GCF_025449255.1_ASM2544925v1_genomic.gff', 'GCA_017875395.1/GCF_017875395.1_ASM1787539v1_genomic.gff', 'GCA_025914035.1/GCF_025914035.1_ASM2591403v1_genomic.gff', 'GCA_003385755.1/GCF_003385755.1_ASM338575v1_genomic.gff', 'GCA_017633855.1/GCA_017633855.1_ASM1763385v1_genomic.gff', 'GCA_000191585.1/GCF_000191585.1_ASM19158v1_genomic.gff', 'GCA_013761055.1/GCF_013761055.1_ASM1376105v1_genomic.gff', 'GCA_000007065.1/GCF_000007065.1_ASM706v1_genomic.gff', 'GCA_002498785.1/GCA_002498785.1.gff', 'GCA_900079125.1/GCA_900079125.1_MBBA_genomic.gff', 'GCA_002813655.1/GCF_002813655.1_ASM281365v1_genomic.gff', 'GCA_017940815.1/GCA_017940815.1_ASM1794081v1_genomic.gff', 'GCA_017387505.1/GCA_017387505.1_ASM1738750v1_genomic.gff', 'GCA_000204415.1/GCF_000204415.1_ASM20441v1_genomic.gff', 'GCA_002506995.1/GCA_002506995.1.gff', 'GCA_000800805.1/GCF_000800805.1_ASM80080v1_genomic.gff', 'GCA_003351865.1/GCF_003351865.1_ASM335186v1_genomic.gff', 'GCA_017410345.1/GCA_017410345.1_ASM1741034v1_genomic.gff', 'GCA_002494585.1/GCA_002494585.1.gff', 'GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff', 'GCA_000017185.1/GCF_000017185.1_ASM1718v1_genomic.gff', 'GCA_017473945.1/GCA_017473945.1_ASM1747394v1_genomic.gff', 'GCA_002502465.1/GCA_002502465.1.gff', 'GCA_024170505.1/GCF_024170505.1_ASM2417050v1_genomic.gff', 'GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', 'GCA_017409525.1/GCA_017409525.1_ASM1740952v1_genomic.gff', 'GCA_001458655.1/GCF_001458655.1_Mb9_genomic.gff', 'GCA_017406825.1/GCA_017406825.1_ASM1740682v1_genomic.gff', 'GCA_000008645.1/GCF_000008645.1_ASM864v1_genomic.gff', 'GCA_000145295.1/GCF_000145295.1_ASM14529v1_genomic.gff', 'GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', 'GCA_900036045.1/GCF_900036045.1_Methanoculleus_sp_MAB1_genomic.gff', 'GCA_900106905.1/GCF_900106905.1_IMG-taxon_2642422606_annotated_assembly_genomic.gff', 'GCA_000023985.1/GCF_000023985.1_ASM2398v1_genomic.gff', 'GCA_014203295.1/GCF_014203295.1_ASM1420329v1_genomic.gff', 'GCA_000341715.1/GCF_000341715.1_ASM34171v1_genomic.gff', 'GCA_018502485.1/GCF_018502485.1_ASM1850248v1_genomic.gff', 'GCA_029633915.1/GCF_029633915.1_ASM2963391v1_genomic.gff', 'GCA_008033705.1/GCF_008033705.1_ASM803370v1_genomic.gff', 'GCA_014207315.1/GCF_014207315.1_ASM1420731v1_genomic.gff', 'GCA_902787415.1/GCA_902787415.1.gff', 'GCA_000025865.1/GCF_000025865.1_ASM2586v1_genomic.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "275"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "faa_paths = []\n",
    "for file in faa_files:\n",
    "    faa_paths.append(file[40:])\n",
    "print(faa_paths)\n",
    "len(faa_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1316e212",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdC', 'fmdC', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdG', 'fwdH', 'ftr', 'mch', 'mtd', 'hmd', 'mer;', 'mtrA', 'mtrB', 'mtrC', 'mtrD', 'mtrE', 'mtrF', 'mtrG', 'mtrH', 'mcrA', 'mcrB', 'mcrG', 'hdrA1', 'hdrB1', 'hdrC1', 'hdrA2', 'hdrB2', 'hdrC2', 'hdrD', 'hdrE', 'mvhA', 'mvhG', 'mvhD', 'fdhA', 'fdhB']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b63f759",
   "metadata": {},
   "source": [
    "GFF File Format: \n",
    "- Fields must be tab-separated. Also, all but the final field in each feature line must contain a value; \"empty\" columns should be denoted with a '.'\n",
    "[0] seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.\n",
    "[1] source - name of the program that generated this feature, or the data source (database or project name)\n",
    "[2] feature - feature type name, e.g. Gene, Variation, Similarity\n",
    "[3] start - Start position* of the feature, with sequence numbering starting at 1.\n",
    "[4] end - End position* of the feature, with sequence numbering starting at 1.\n",
    "[5] score - A floating point value.\n",
    "[6] strand - defined as + (forward) or - (reverse).\n",
    "[7] frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..\n",
    "[8] attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "e3288df2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create an empty dictionary to store the gene counts for each species\n",
    "import pandas as pd\n",
    "gene_counts = {}\n",
    "\n",
    "for path_name in faa_paths:\n",
    "    species = path_name[:15]  #keep species name from the path_name\n",
    "    gene_counts[species] = {}\n",
    "\n",
    "    #read the GFF file\n",
    "    with open(path_name, 'r') as file:\n",
    "        for line in file:\n",
    "            if line.startswith('>'):\n",
    "                if \"mrtA\" in line or \"MrtA\" in line:\n",
    "                    if 'mrtA' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtA'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtA'] += 1\n",
    "                if \"mrtB\" in line or \"MrtB\" in line:\n",
    "                    if 'mrtB' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtB'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtB'] += 1\n",
    "                if \"mrtG\" in line or \"MrtG\" in line:\n",
    "                    if 'mrtG' not in gene_counts[species]:\n",
    "                        gene_counts[species]['mrtG'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['mrtG'] += 1\n",
    "\n",
    "\n",
    "# Create a dataframe from the gene counts\n",
    "df = pd.DataFrame.from_dict(gene_counts, orient='index')\n",
    "df.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d146647d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename_axis('assembly_accession')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8df86a25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape[0] #missing 5 rows (species) \n",
    "df.shape[1] #missing 5 columns (genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4132f9d3",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[43mto_search\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m      2\u001b[0m df\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/frame.py:3813\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3811\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_iterator(key):\n\u001b[1;32m   3812\u001b[0m         key \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(key)\n\u001b[0;32m-> 3813\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_indexer_strict\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcolumns\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m[\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m   3815\u001b[0m \u001b[38;5;66;03m# take() does not accept boolean indexers\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(indexer, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdtype\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mbool\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6070\u001b[0m, in \u001b[0;36mIndex._get_indexer_strict\u001b[0;34m(self, key, axis_name)\u001b[0m\n\u001b[1;32m   6067\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   6068\u001b[0m     keyarr, indexer, new_indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex_non_unique(keyarr)\n\u001b[0;32m-> 6070\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_raise_if_missing\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkeyarr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis_name\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   6072\u001b[0m keyarr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtake(indexer)\n\u001b[1;32m   6073\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(key, Index):\n\u001b[1;32m   6074\u001b[0m     \u001b[38;5;66;03m# GH 42790 - Preserve name from an Index\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6133\u001b[0m, in \u001b[0;36mIndex._raise_if_missing\u001b[0;34m(self, key, indexer, axis_name)\u001b[0m\n\u001b[1;32m   6130\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNone of [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mkey\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m] are in the [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00maxis_name\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m]\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   6132\u001b[0m not_found \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(ensure_index(key)[missing_mask\u001b[38;5;241m.\u001b[39mnonzero()[\u001b[38;5;241m0\u001b[39m]]\u001b[38;5;241m.\u001b[39munique())\n\u001b[0;32m-> 6133\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mnot_found\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not in index\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\""
     ]
    }
   ],
   "source": [
    "df = df[to_search]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "ec15cc8d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mcrA</th>\n",
       "      <th>mcrG</th>\n",
       "      <th>mcrB</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>mtbC</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>mtbA</th>\n",
       "      <th>...</th>\n",
       "      <th>mtmB</th>\n",
       "      <th>mtbB</th>\n",
       "      <th>mttC</th>\n",
       "      <th>mttB</th>\n",
       "      <th>hdrA1</th>\n",
       "      <th>mtmC</th>\n",
       "      <th>hdrB1</th>\n",
       "      <th>hdrC1</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_001458655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000008645.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000145295.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008033705.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002813655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001477655.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_028743435.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000970325.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001027005.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_000306725.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    mvhA  mcrA  mcrG  mcrB  mvhG  hdrA2  mtbC  hdrE  hdrD  \\\n",
       "assembly_accession                                                          \n",
       "GCF_001458655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000008645.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000145295.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_008033705.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_002813655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "...                  ...   ...   ...   ...   ...    ...   ...   ...   ...   \n",
       "GCF_001477655.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_028743435.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000970325.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   0.0   \n",
       "GCF_001027005.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   1.0   \n",
       "GCA_000306725.1      0.0   0.0   0.0   0.0   0.0    0.0   0.0   0.0   0.0   \n",
       "\n",
       "                    mtbA  ...  mtmB  mtbB  mttC  mttB  hdrA1  mtmC  hdrB1  \\\n",
       "assembly_accession        ...                                               \n",
       "GCF_001458655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000008645.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000145295.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_008033705.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_002813655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "...                  ...  ...   ...   ...   ...   ...    ...   ...    ...   \n",
       "GCF_001477655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_028743435.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000970325.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_001027005.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCA_000306725.1      1.0  ...   4.0   3.0   1.0   2.0    0.0   0.0    0.0   \n",
       "\n",
       "                    hdrC1  hdrB2  hdrC2  \n",
       "assembly_accession                       \n",
       "GCF_001458655.1       0.0    0.0    0.0  \n",
       "GCF_000008645.1       0.0    0.0    0.0  \n",
       "GCF_000145295.1       0.0    0.0    0.0  \n",
       "GCF_008033705.1       0.0    0.0    0.0  \n",
       "GCF_002813655.1       0.0    0.0    0.0  \n",
       "...                   ...    ...    ...  \n",
       "GCF_001477655.1       0.0    0.0    0.0  \n",
       "GCF_028743435.1       0.0    0.0    0.0  \n",
       "GCF_000970325.1       0.0    0.0    0.0  \n",
       "GCF_001027005.1       0.0    0.0    0.0  \n",
       "GCA_000306725.1       0.0    0.0    0.0  \n",
       "\n",
       "[199 rows x 23 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add columns that are missing, fill in with 0\n",
    "missing = ['mtmC','hdrB1', 'hdrC1', 'hdrB2', 'hdrC2']\n",
    "for i in missing: \n",
    "    df[i] = 0.0\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ef5985d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002495685.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Accession                  organism_name   Isolate\n",
       "0  GCA_902788255.1  uncultured Methanobrevibacter  RUG13354\n",
       "1  GCA_002494785.1   Methanobacteriaceae archaeon    UBA117\n",
       "2  GCA_902784195.1  uncultured Methanobrevibacter  RUG12946\n",
       "3  GCA_002495685.1             Methanobrevibacter    UBA412\n",
       "4  GCA_902771225.1  uncultured Methanobrevibacter  RUG11657"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#match species to the assembly acession numbers \n",
    "#first import csv with species and their accession numbers \n",
    "species = pd.read_csv('/Users/gracedai/Documents/MAGS.csv')\n",
    "species = species[['Accession', 'organism_name', 'Isolate']]\n",
    "species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4c6d603f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mrtB</th>\n",
       "      <th>mrtA</th>\n",
       "      <th>mrtG</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   mrtB  mrtA  mrtG Accession organism_name Isolate\n",
       "0     2   0.0   0.0       NaN           NaN     NaN\n",
       "1     1   1.0   1.0       NaN           NaN     NaN\n",
       "2     1   1.0   1.0       NaN           NaN     NaN\n",
       "3     1   1.0   1.0       NaN           NaN     NaN\n",
       "4     1   1.0   1.0       NaN           NaN     NaN"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='Accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "950cd6a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('Mags1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "649191c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "      <td>1.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "      <td>2.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>GCA_902764455.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>GCA_902768415.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11363</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                      organism_name   Isolate  16S  tRNA\n",
       "0    GCA_902788045.1      uncultured Methanobrevibacter  RUG13330  0.0    32\n",
       "1    GCA_902774685.1      uncultured Methanobrevibacter  RUG11990  0.0    28\n",
       "2    GCA_002497785.1                       Methanolinea    UBA144  1.0    37\n",
       "3    GCA_002502925.1  Methanomassiliicoccaceae archaeon    UBA414  2.0    41\n",
       "4    GCA_002506795.1                   Methanobacterium    UBA397  0.0    38\n",
       "..               ...                                ...       ...  ...   ...\n",
       "218  GCA_902764455.1      uncultured Methanobrevibacter  RUG10962  0.0    35\n",
       "219  GCA_902802065.1      uncultured Methanobrevibacter  RUG14729  2.0    35\n",
       "220  GCA_902787415.1              methanogenic archaeon  RUG13270  0.0    35\n",
       "221  GCA_902768415.1      uncultured Methanobrevibacter  RUG11363  0.0    27\n",
       "222  GCA_002502855.1                   Methanobacterium    UBA279  0.0    37\n",
       "\n",
       "[223 rows x 5 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "33c18b50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_902770715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11588</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902801725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14697</td>\n",
       "      <td>1.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GCA_902758945.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10414</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_902769175.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11428</td>\n",
       "      <td>0.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>GCA_902777475.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_902759195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10437</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>GCA_902784295.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12954</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>GCA_902762245.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>GCA_902770085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11531</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>GCA_902774705.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11996</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                  organism_name   Isolate  16S  tRNA\n",
       "38   GCA_902770715.1  uncultured Methanobrevibacter  RUG11588  0.0    30\n",
       "37   GCA_902801725.1  uncultured Methanobrevibacter  RUG14697  1.0    30\n",
       "166  GCA_902798335.1  uncultured Methanobrevibacter  RUG14349  0.0    31\n",
       "167  GCA_902757275.1  uncultured Methanobrevibacter  RUG10244  0.0    36\n",
       "75   GCA_902758945.1  uncultured Methanobrevibacter  RUG10414  0.0    33\n",
       "34   GCA_902769175.1  uncultured Methanobrevibacter  RUG11428  0.0    41\n",
       "125  GCA_902777475.1  uncultured Methanobrevibacter  RUG12275  0.0    34\n",
       "32   GCA_902759195.1  uncultured Methanobrevibacter  RUG10437  0.0    33\n",
       "174  GCA_902784295.1  uncultured Methanobrevibacter  RUG12954  0.0    31\n",
       "31   GCA_902788255.1  uncultured Methanobrevibacter  RUG13354  0.0    31\n",
       "177  GCA_902762245.1  uncultured Methanobrevibacter  RUG10743  0.0    30\n",
       "178  GCA_902770085.1  uncultured Methanobrevibacter  RUG11531  0.0    34\n",
       "123  GCA_902775005.1  uncultured Methanobrevibacter  RUG12025  0.0    30\n",
       "25   GCA_902762085.1  uncultured Methanobrevibacter  RUG10738  0.0    23\n",
       "140  GCA_902774705.1  uncultured Methanobrevibacter  RUG11996  0.0    82\n",
       "184  GCA_902765105.1  uncultured Methanobrevibacter  RUG11035  0.0    30\n",
       "185  GCA_902785855.1  uncultured Methanobrevibacter  RUG13115  0.0    28\n",
       "120  GCA_902757365.1  uncultured Methanobrevibacter  RUG10252  0.0    36\n",
       "21   GCA_902784715.1  uncultured Methanobrevibacter  RUG12995  0.0    27\n",
       "0    GCA_902788045.1  uncultured Methanobrevibacter  RUG13330  0.0    32"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_name'] = df['organism_name'].astype(str)\n",
    "df_sorted = df.sort_values('organism_name', ascending=True)\n",
    "df_sorted.tail(n=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "cefb5125",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sorted.to_csv('prokka_16s.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "14289b16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "c18cc498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism_name</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>strain</th>\n",
       "      <th>fwdA</th>\n",
       "      <th>fmdA</th>\n",
       "      <th>fwdB</th>\n",
       "      <th>fmdB</th>\n",
       "      <th>fwdC</th>\n",
       "      <th>fmdC</th>\n",
       "      <th>fwdD</th>\n",
       "      <th>...</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>mvhD</th>\n",
       "      <th>fdhA</th>\n",
       "      <th>fdhB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#CO2 -&gt; methane</td>\n",
       "      <td># KEGG M00567</td>\n",
       "      <td>#</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00203</td>\n",
       "      <td>...</td>\n",
       "      <td>K03388</td>\n",
       "      <td>K03389</td>\n",
       "      <td>K03390</td>\n",
       "      <td>K08264</td>\n",
       "      <td>K08265</td>\n",
       "      <td>K14126</td>\n",
       "      <td>K14128</td>\n",
       "      <td>K14127</td>\n",
       "      <td>K22516</td>\n",
       "      <td>K00125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit D [EC...</td>\n",
       "      <td>...</td>\n",
       "      <td>heterodisulfide reductase subunit A2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit B2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit C2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit D [EC:1.8.98.1]</td>\n",
       "      <td>heterodisulfide reductase subunit E [EC:1.8.98.1]</td>\n",
       "      <td>F420-non-reducing hydrogenase large subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase small subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase iron-sulfur subu...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) alpha su...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) beta sub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Candidatus Methanogranum sp. U3.2.1</td>\n",
       "      <td>GCA_019262145.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Candidatus Methanoliparum sp. LAM-1</td>\n",
       "      <td>GCF_020885915.1</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis</td>\n",
       "      <td>GCF_902383905.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis ...</td>\n",
       "      <td>GCF_000404225.1</td>\n",
       "      <td>Issoire-Mx1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_902387285.1</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_003711245.1</td>\n",
       "      <td>Mx-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus Mx1201</td>\n",
       "      <td>GCF_000300255.2</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Candidatus Methanomethylophilus sp. 1R26 (arch...</td>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>1R26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       organism_name assembly_accession  \\\n",
       "0                                    #CO2 -> methane      # KEGG M00567   \n",
       "1                                                  #                  #   \n",
       "2                Candidatus Methanogranum sp. U3.2.1    GCA_019262145.1   \n",
       "3                Candidatus Methanoliparum sp. LAM-1    GCF_020885915.1   \n",
       "4      Candidatus Methanomassiliicoccus intestinalis    GCF_902383905.1   \n",
       "5  Candidatus Methanomassiliicoccus intestinalis ...    GCF_000404225.1   \n",
       "6              Candidatus Methanomethylophilus alvus    GCF_902387285.1   \n",
       "7              Candidatus Methanomethylophilus alvus    GCF_003711245.1   \n",
       "8       Candidatus Methanomethylophilus alvus Mx1201    GCF_000300255.2   \n",
       "9  Candidatus Methanomethylophilus sp. 1R26 (arch...    GCF_001481295.1   \n",
       "\n",
       "        strain                                               fwdA  \\\n",
       "0            #                                             K00200   \n",
       "1            #  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2           na                                                  0   \n",
       "3        LAM-1                                                  0   \n",
       "4           na                                                  0   \n",
       "5  Issoire-Mx1                                                  0   \n",
       "6       Mx1201                                                  0   \n",
       "7        Mx-05                                                  0   \n",
       "8       Mx1201                                                  0   \n",
       "9         1R26                                                  0   \n",
       "\n",
       "                                                fmdA  \\\n",
       "0                                             K00200   \n",
       "1  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdD  ...  \\\n",
       "0                                             K00203  ...   \n",
       "1  formylmethanofuran dehydrogenase subunit D [EC...  ...   \n",
       "2                                                  0  ...   \n",
       "3                                                  0  ...   \n",
       "4                                                  0  ...   \n",
       "5                                                  0  ...   \n",
       "6                                                  0  ...   \n",
       "7                                                  0  ...   \n",
       "8                                                  0  ...   \n",
       "9                                                  0  ...   \n",
       "\n",
       "                                               hdrA2  \\\n",
       "0                                             K03388   \n",
       "1  heterodisulfide reductase subunit A2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  2   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrB2  \\\n",
       "0                                             K03389   \n",
       "1  heterodisulfide reductase subunit B2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrC2  \\\n",
       "0                                             K03390   \n",
       "1  heterodisulfide reductase subunit C2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrD  \\\n",
       "0                                             K08264   \n",
       "1  heterodisulfide reductase subunit D [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrE  \\\n",
       "0                                             K08265   \n",
       "1  heterodisulfide reductase subunit E [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhA  \\\n",
       "0                                             K14126   \n",
       "1  F420-non-reducing hydrogenase large subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhG  \\\n",
       "0                                             K14128   \n",
       "1  F420-non-reducing hydrogenase small subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhD  \\\n",
       "0                                             K14127   \n",
       "1  F420-non-reducing hydrogenase iron-sulfur subu...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhA  \\\n",
       "0                                             K22516   \n",
       "1  formate dehydrogenase (coenzyme F420) alpha su...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhB  \n",
       "0                                             K00125  \n",
       "1  formate dehydrogenase (coenzyme F420) beta sub...  \n",
       "2                                                  0  \n",
       "3                                                  0  \n",
       "4                                                  0  \n",
       "5                                                  0  \n",
       "6                                                  0  \n",
       "7                                                  0  \n",
       "8                                                  0  \n",
       "9                                                  0  \n",
       "\n",
       "[10 rows x 46 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs = pd.read_csv('/Users/gracedai/Downloads/co2.csv')\n",
    "prokka = pd.read_csv('/Users/gracedai/Downloads/prokkas.csv')\n",
    "rumen_only = pd.read_csv('/Users/gracedai/Downloads/rumen.csv')\n",
    "wgs.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "620a03c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = pd.merge(rumen_only, wgs, on='assembly_accession', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "d7c75738",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>strain_x</th>\n",
       "      <th>host</th>\n",
       "      <th>location</th>\n",
       "      <th>mcrA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000024185.1</td>\n",
       "      <td>Methanobrevibacter ruminantium</td>\n",
       "      <td>M1</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>Germany</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001027005.1</td>\n",
       "      <td>Methanosarcina barkeri</td>\n",
       "      <td>CM1</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001563245.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>Lamb</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000711215.1</td>\n",
       "      <td>Methanomicrobium mobile</td>\n",
       "      <td>BP</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>1R26</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Denmark</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_003111625.1</td>\n",
       "      <td>Methanobrevibacter thaueri</td>\n",
       "      <td>DSM 11995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCF_000320505.1</td>\n",
       "      <td>Methanobrevibacter boviskoreani</td>\n",
       "      <td>JH1</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Korea</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCF_001729965.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>WGK6</td>\n",
       "      <td>Kangaroo</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCF_003268005.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>BMS</td>\n",
       "      <td>Steer</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCF_001548675.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>YE315</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCF_001477655.1</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_030168595.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-DZ1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCF_000404165.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>AbM4</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCF_000621965.1</td>\n",
       "      <td>Methanobrevibacter wolinii</td>\n",
       "      <td>SH</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCF_900116525.1</td>\n",
       "      <td>Methanosarcina</td>\n",
       "      <td>DSM 11855</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCF_900114585.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>DSM 16632</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_030168575.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-RCC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCF_000762265.1</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902789165.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13441</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_902799105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14438</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_902800515.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14577</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_902765875.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11103</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_902781855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12711</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902757195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10238</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GCA_017540705.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6915</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GCA_002509015.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA188</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GCA_902775725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12100</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GCA_902763115.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10830</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GCA_902758425.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10359</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GCA_902786095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13139</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GCA_902759335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10447</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>GCA_017535165.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6752</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GCA_902756745.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10188</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GCA_017524805.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6768</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>GCA_902800035.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14542</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GCA_902800615.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14594</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>GCA_002505605.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA187</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>GCA_902774605.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11987</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>GCA_902775675.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12086</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GCA_902768855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11403</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>GCA_902798395.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14364</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession                              organism_name_x   strain_x  \\\n",
       "0     GCF_000024185.1              Methanobrevibacter ruminantium          M1   \n",
       "1     GCF_001027005.1                       Methanosarcina barkeri        CM1   \n",
       "2     GCF_001563245.1                   Methanobrevibacter olleyae       YLM1   \n",
       "4     GCF_000711215.1                      Methanomicrobium mobile         BP   \n",
       "5     GCF_001481295.1              Candidatus Methanomethylophilus       1R26   \n",
       "6     GCF_003111625.1                   Methanobrevibacter thaueri  DSM 11995   \n",
       "9     GCF_000320505.1              Methanobrevibacter boviskoreani        JH1   \n",
       "10    GCF_001729965.1                               Methanosphaera       WGK6   \n",
       "11    GCF_003268005.1                               Methanosphaera        BMS   \n",
       "12    GCF_001548675.1                           Methanobrevibacter      YE315   \n",
       "13    GCF_001477655.1                  Methanobrevibacter millerae        SM9   \n",
       "14    GCA_030168595.1             Methanomassiliicoccales archaeon    LGM-DZ1   \n",
       "15    GCF_000404165.1                           Methanobrevibacter       AbM4   \n",
       "16    GCF_000621965.1                   Methanobrevibacter wolinii         SH   \n",
       "17    GCF_900116525.1                               Methanosarcina  DSM 11855   \n",
       "18    GCF_900114585.1                  Methanobrevibacter olleyae   DSM 16632   \n",
       "19    GCA_030168575.1             Methanomassiliicoccales archaeon   LGM-RCC1   \n",
       "20    GCF_000762265.1                  Methanobacterium formicicum       BRM9   \n",
       "21    GCA_902789165.1                uncultured Methanobrevibacter   RUG13441   \n",
       "22    GCA_902762085.1                uncultured Methanobrevibacter   RUG10738   \n",
       "23    GCA_902799105.1                uncultured Methanobrevibacter   RUG14438   \n",
       "24    GCA_902774685.1                uncultured Methanobrevibacter   RUG11990   \n",
       "28    GCA_902775005.1                uncultured Methanobrevibacter   RUG12025   \n",
       "30    GCA_902762095.1                uncultured Methanobrevibacter   RUG10736   \n",
       "33    GCA_902800515.1                uncultured Methanobrevibacter   RUG14577   \n",
       "35    GCA_902787415.1                        methanogenic archaeon   RUG13270   \n",
       "36    GCA_902765105.1                uncultured Methanobrevibacter   RUG11035   \n",
       "37    GCA_902763685.1                        methanogenic archaeon   RUG10898   \n",
       "39    GCA_902765875.1                uncultured Methanobrevibacter   RUG11103   \n",
       "41    GCA_902784715.1                uncultured Methanobrevibacter   RUG12995   \n",
       "43    GCA_902781855.1                uncultured Methanobrevibacter   RUG12711   \n",
       "44    GCA_902757195.1                uncultured Methanobrevibacter   RUG10238   \n",
       "47    GCA_017540705.1  Candidatus Methanomethylophilaceae archaeon   RGIG6915   \n",
       "48    GCA_002509015.1                           Methanobrevibacter     UBA188   \n",
       "49    GCA_902775725.1                uncultured Methanobrevibacter   RUG12100   \n",
       "58    GCA_902763115.1                uncultured Methanobrevibacter   RUG10830   \n",
       "59    GCA_902758425.1                uncultured Methanobrevibacter   RUG10359   \n",
       "62    GCA_902786095.1                uncultured Methanobrevibacter   RUG13139   \n",
       "63    GCA_902759335.1                uncultured Methanobrevibacter   RUG10447   \n",
       "65    GCA_017535165.1  Candidatus Methanomethylophilaceae archaeon   RGIG6752   \n",
       "66    GCA_902756745.1                uncultured Methanobrevibacter   RUG10188   \n",
       "67    GCA_902757275.1                uncultured Methanobrevibacter   RUG10244   \n",
       "69    GCA_902785855.1                uncultured Methanobrevibacter   RUG13115   \n",
       "71    GCA_017524805.1  Candidatus Methanomethylophilaceae archaeon   RGIG6768   \n",
       "72    GCA_902802065.1                uncultured Methanobrevibacter   RUG14729   \n",
       "73    GCA_902800035.1                uncultured Methanobrevibacter   RUG14542   \n",
       "74    GCA_902798335.1                uncultured Methanobrevibacter   RUG14349   \n",
       "77    GCA_902800615.1                uncultured Methanobrevibacter   RUG14594   \n",
       "78    GCA_002505605.1                           Methanobrevibacter     UBA187   \n",
       "79    GCA_902757365.1                uncultured Methanobrevibacter   RUG10252   \n",
       "83    GCA_902774605.1                uncultured Methanobrevibacter   RUG11987   \n",
       "84    GCA_902775675.1                uncultured Methanobrevibacter   RUG12086   \n",
       "85    GCA_902768855.1                uncultured Methanobrevibacter   RUG11403   \n",
       "86    GCA_902798395.1                uncultured Methanobrevibacter   RUG14364   \n",
       "88    GCA_902784195.1                uncultured Methanobrevibacter   RUG12946   \n",
       "89    GCA_902771225.1                uncultured Methanobrevibacter   RUG11657   \n",
       "\n",
       "                                                 host     location  mcrA  \n",
       "0                                              Bovine      Germany     1  \n",
       "1                                                 Cow  New Zealand     1  \n",
       "2                                                Lamb  New Zealand     2  \n",
       "4                                                 Cow    Australia     1  \n",
       "5                                                 Cow      Denmark     1  \n",
       "6                                                 Cow          NaN     1  \n",
       "9                                              Cattle        Korea     1  \n",
       "10                                           Kangaroo    Australia     1  \n",
       "11                                              Steer    Australia     1  \n",
       "12                                             Cattle    Australia     1  \n",
       "13                                             Sheep   New Zealand     2  \n",
       "14                                                NaN        China     1  \n",
       "15                                             Sheep   New Zealand     1  \n",
       "16                                             Bovine          USA     1  \n",
       "17                                             Bovine          USA     1  \n",
       "18                                            Bovine           USA     2  \n",
       "19                                                NaN        China     1  \n",
       "20                                                Cow  New Zealand     2  \n",
       "21                                                Cow     Scotland     1  \n",
       "22                                                Cow     Scotland     2  \n",
       "23                                                Cow     Scotland     1  \n",
       "24                                                Cow     Scotland     2  \n",
       "28                                                Cow     Scotland     1  \n",
       "30                                                Cow     Scotland     1  \n",
       "33                                                Cow     Scotland     2  \n",
       "35                                                Cow     Scotland     1  \n",
       "36                                                Cow     Scotland     1  \n",
       "37                                                Cow     Scotland     1  \n",
       "39                                                Cow     Scotland     1  \n",
       "41                                                Cow     Scotland     1  \n",
       "43                                                Cow     Scotland     1  \n",
       "44                                                Cow     Scotland     1  \n",
       "47  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "48                               Cattle; high methane          NaN     1  \n",
       "49                                                Cow     Scotland     1  \n",
       "58                                                Cow     Scotland     2  \n",
       "59                                                Cow     Scotland     1  \n",
       "62                                                Cow     Scotland     1  \n",
       "63                                                Cow     Scotland     1  \n",
       "65  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "66                                                Cow     Scotland     1  \n",
       "67                                                Cow     Scotland     1  \n",
       "69                                                Cow     Scotland     2  \n",
       "71  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "72                                                Cow     Scotland     1  \n",
       "73                                                Cow     Scotland     1  \n",
       "74                                                Cow     Scotland     1  \n",
       "77                                                Cow     Scotland     1  \n",
       "78                               Cattle; high methane          NaN     1  \n",
       "79                                                Cow     Scotland     1  \n",
       "83                                                Cow     Scotland     1  \n",
       "84                                                Cow     Scotland     1  \n",
       "85                                                Cow     Scotland     1  \n",
       "86                                                Cow     Scotland     1  \n",
       "88                                                Cow     Scotland     1  \n",
       "89                                                Cow     Scotland     1  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out['mcrA'] = out['mcrA'].fillna(0).astype(int)\n",
    "out_mcrA = out[out['mcrA'] >= 1]\n",
    "out_mcrA = out_mcrA[['assembly_accession', 'organism_name_x', 'strain_x','host', 'location', 'mcrA']]\n",
    "out_mcrA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "52f52ca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_mcrA.to_csv('out_mcrA.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6cc8db58",
   "metadata": {},
   "source": [
    "# Begin MRT Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "01dd3668",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCA_001477655.1', 'GCA_001563245.1', 'GCA_000762265.1', 'GCA_000012545.1', 'GCA_002502965.1', 'GCA_902762095.1', 'GCA_000953115.1', 'GCA_900095815.1', 'GCA_902763685.1', 'GCA_902795935.1', 'GCA_002506325.1', 'GCA_002506905.1', 'GCA_002509405.1', 'GCA_002496945.1', 'GCA_002505245.1', 'GCA_002498705.1', 'GCA_002498605.1', 'GCA_002504405.1', 'GCA_002506565.1', 'GCA_002505345.1', 'GCA_002494805.1', 'GCA_002498365.1', 'GCA_002505275.1', 'GCA_002498805.1', 'GCA_002503785.1', 'GCA_002508625.1', 'GCA_002498765.1', 'GCA_002495325.1', 'GCA_002498425.1', 'GCA_002498375.1', 'GCA_002498785.1', 'GCA_002506995.1', 'GCA_002494585.1', 'GCA_002502465.1', 'GCA_000499765.1', 'GCA_902787415.1']\n"
     ]
    }
   ],
   "source": [
    "#the below section is to extract just genes that are mcr + mrt from species that have at least 1 mrt gene \n",
    "# later, we will do an analysis of species that only have mcr vs have mrt & mcr. also are there species that have 2 mcr and no mrt? need to see if it is rly mrt or not!\n",
    "# expect that there are a lot more genes that are actually mcrA but aren't marked so\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "#read the text file of gene names into a list\n",
    "gene_file = open(\"jgi_ncbi_downloads.txt\", \"r\")\n",
    "data = gene_file.read()\n",
    "target_strings = data.split(\"\\n\")\n",
    "print(target_strings)\n",
    "gene_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6b098c6c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n",
      "./GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff\n",
      "./GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff\n",
      "./GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff\n",
      "./GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff\n",
      "./GCA_902762095.1/GCA_902762095_genomic.gff\n",
      "./GCA_002502965.1/GCA_002502965.1.gff\n",
      "./GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff\n",
      "./GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff\n",
      "./GCA_902763685.1/GCA_902763685.1.gff\n",
      "./GCA_902795935.1/GCA_902795935.1.gff\n",
      "./GCA_002506325.1/GCA_002506325.1.gff\n",
      "./GCA_002506905.1/GCA_002506905.1.gff\n",
      "./GCA_002509405.1/GCA_002509405.1.gff\n",
      "./GCA_002496945.1/GCA_002496945.1.gff\n",
      "./GCA_002505245.1/GCA_002505245.1.gff\n",
      "./GCA_002498705.1/GCA_002498705.1.gff\n",
      "./GCA_002498605.1/GCA_002498605.1.gff\n",
      "./GCA_002504405.1/GCA_002504405.1.gff\n",
      "./GCA_002506565.1/GCA_002506565.1.gff\n",
      "./GCA_002505345.1/GCA_002505345.1.gff\n",
      "./GCA_002494805.1/GCA_002494805.1.gff\n",
      "./GCA_002498365.1/GCA_002498365.1.gff\n",
      "./GCA_002505275.1/GCA_002505275.1.gff\n",
      "./GCA_002498805.1/GCA_002498805.1.gff\n",
      "./GCA_002503785.1/GCA_002503785.1.gff\n",
      "./GCA_002508625.1/GCA_002508625.1.gff\n",
      "./GCA_002498765.1/GCA_002498765.1.gff\n",
      "./GCA_002495325.1/GCA_002495325.1.gff\n",
      "./GCA_002498425.1/GCA_002498425.1.gff\n",
      "./GCA_002498375.1/GCA_002498375.1.gff\n",
      "./GCA_002498785.1/GCA_002498785.1.gff\n",
      "./GCA_002506995.1/GCA_002506995.1.gff\n",
      "./GCA_002494585.1/GCA_002494585.1.gff\n",
      "./GCA_002502465.1/GCA_002502465.1.gff\n",
      "./GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff\n",
      "./GCA_902787415.1/GCA_902787415.1.gff\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Initialize a list to store the paths of matching GFF files\n",
    "gff_paths = ['./GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', './GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', \n",
    "             './GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', './GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', './GCA_902762095.1/GCA_902762095_genomic.gff']\n",
    "\n",
    "# Walk through the directory and its subdirectories\n",
    "for root, _, files in os.walk(\".\"):\n",
    "    for filename in files:\n",
    "        for target_string in target_strings:\n",
    "            # Check if the target string is in the filename\n",
    "            if target_string in filename:\n",
    "                # Check if the file is a GFF file (you can adjust this condition as needed)\n",
    "                if filename.endswith(\".gff\"):\n",
    "                    # Construct the full path to the GFF file\n",
    "                    gff_path = os.path.join(root, filename)\n",
    "                    gff_paths.append(gff_path)\n",
    "print(len(gff_paths))\n",
    "# Print the list of matching GFF file paths\n",
    "for path in gff_paths:\n",
    "    print(path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a49f5a6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1505565</td>\n",
       "      <td>1506359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>2219996</td>\n",
       "      <td>2220754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1341629</td>\n",
       "      <td>1342426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1666316</td>\n",
       "      <td>1667068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>934501</td>\n",
       "      <td>935250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>2213588</td>\n",
       "      <td>2214388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000012545.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>378927</td>\n",
       "      <td>379724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12323</td>\n",
       "      <td>13117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>889</td>\n",
       "      <td>1647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_002502965.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>129896</td>\n",
       "      <td>130669</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession index  gene    start     stop\n",
       "0    GCA_001477655.1     1  mcrG  1505565  1506359\n",
       "1    GCA_001477655.1     2  mcrG  2219996  2220754\n",
       "2    GCA_001563245.1     1  mcrG  1341629  1342426\n",
       "3    GCA_001563245.1     2  mcrG  1666316  1667068\n",
       "4    GCA_000762265.1     1  mcrG   934501   935250\n",
       "5    GCA_000762265.1     2  mcrG  2213588  2214388\n",
       "6    GCA_000012545.1     1  mcrG   378927   379724\n",
       "7    GCA_902762095.1     1  mrtG    12323    13117\n",
       "8    GCA_902762095.1     2  mcrG      889     1647\n",
       "9    GCA_002502965.1     1  mrtG   129896   130669"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create dataframe to store the output \n",
    "df = pd.DataFrame(columns=[\"assembly_accession\", \"index\", \"gene\", \"start\", \"stop\"])\n",
    "\n",
    "# iterate through the GFF files\n",
    "for path_name in gff_paths:\n",
    "    species = path_name[2:17]  #keep species name from the path_name\n",
    "    gene_count = 0            #index to keep count of number of times gene shows up in one species\n",
    "    \n",
    "    with open(path_name, \"r\") as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                fields = line.strip().split('\\t')\n",
    "                if \"Name=mrtG\" in line or \"Name=MrtG\" in line:   #specify desired gene\n",
    "                    start = int(fields[3])\n",
    "                    stop = int(fields[4])\n",
    "                    gene_count += 1\n",
    "                    new_row = pd.DataFrame({\"assembly_accession\": [species],\n",
    "                                            \"index\": [gene_count],\n",
    "                                            \"gene\": ['mrtG'],\n",
    "                                            \"start\": [start],\n",
    "                                            \"stop\": [stop]})\n",
    "                    df = pd.concat([df, new_row], ignore_index=True)\n",
    "                if \"Name=mcrG\" in line or \"Name=McrG\" in line:   #specify desired gene\n",
    "                    start = int(fields[3])\n",
    "                    stop = int(fields[4])\n",
    "                    gene_count += 1\n",
    "                    new_row = pd.DataFrame({\"assembly_accession\": [species],\n",
    "                                            \"index\": [gene_count],\n",
    "                                            \"gene\": ['mcrG'],\n",
    "                                            \"start\": [start],\n",
    "                                            \"stop\": [stop]})\n",
    "\n",
    "\n",
    "                    df = pd.concat([df, new_row], ignore_index=True)\n",
    "\n",
    "df.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7aaecdae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1505565</td>\n",
       "      <td>1506359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>2219996</td>\n",
       "      <td>2220754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1341629</td>\n",
       "      <td>1342426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1666316</td>\n",
       "      <td>1667068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>934501</td>\n",
       "      <td>935250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2213588</td>\n",
       "      <td>2214388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000012545.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>378927</td>\n",
       "      <td>379724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12323</td>\n",
       "      <td>13117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>889</td>\n",
       "      <td>1647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_002502965.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>129896</td>\n",
       "      <td>130669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>900346</td>\n",
       "      <td>901095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1073013</td>\n",
       "      <td>1073813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1413253</td>\n",
       "      <td>1414050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1444106</td>\n",
       "      <td>1444855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12303</td>\n",
       "      <td>13079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_902795935.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>4531</td>\n",
       "      <td>5328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002506325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29656</td>\n",
       "      <td>30429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002506905.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>54903</td>\n",
       "      <td>55679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002509405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>49427</td>\n",
       "      <td>50200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_002496945.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48198</td>\n",
       "      <td>48974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_002505245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>24589</td>\n",
       "      <td>25362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>22459</td>\n",
       "      <td>23223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>225</td>\n",
       "      <td>989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_002498605.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>5893</td>\n",
       "      <td>6666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_002504405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29702</td>\n",
       "      <td>30478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_002506565.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>33009</td>\n",
       "      <td>33782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_002505345.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>78795</td>\n",
       "      <td>79568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_002494805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>37236</td>\n",
       "      <td>38009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_002498365.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>248040</td>\n",
       "      <td>248813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_002505275.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>31552</td>\n",
       "      <td>32325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_002498805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>76047</td>\n",
       "      <td>76820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_002503785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>57520</td>\n",
       "      <td>58293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_002508625.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>58150</td>\n",
       "      <td>58923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_002498765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>254078</td>\n",
       "      <td>254851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_002495325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>56118</td>\n",
       "      <td>56897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_002498425.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48345</td>\n",
       "      <td>49118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>25637</td>\n",
       "      <td>26401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>83239</td>\n",
       "      <td>84003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_002498785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>51721</td>\n",
       "      <td>52176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_002506995.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48364</td>\n",
       "      <td>49137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCA_002494585.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>3798</td>\n",
       "      <td>4571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_002502465.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>47657</td>\n",
       "      <td>48430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1664303</td>\n",
       "      <td>1665052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1785266</td>\n",
       "      <td>1786066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2086</td>\n",
       "      <td>2862</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession index  gene    start     stop\n",
       "0     GCA_001477655.1     1  mrtG  1505565  1506359\n",
       "1     GCA_001477655.1     2  mcrG  2219996  2220754\n",
       "2     GCA_001563245.1     1  mrtG  1341629  1342426\n",
       "3     GCA_001563245.1     2  mcrG  1666316  1667068\n",
       "4     GCA_000762265.1     1  mcrG   934501   935250\n",
       "5     GCA_000762265.1     2  mrtG  2213588  2214388\n",
       "6     GCA_000012545.1     1  mrtG   378927   379724\n",
       "7     GCA_902762095.1     1  mrtG    12323    13117\n",
       "8     GCA_902762095.1     2  mcrG      889     1647\n",
       "9     GCA_002502965.1     1  mrtG   129896   130669\n",
       "10    GCA_000953115.1     1  mcrG   900346   901095\n",
       "11    GCA_000953115.1     2  mrtG  1073013  1073813\n",
       "12    GCA_900095815.1     1  mrtG  1413253  1414050\n",
       "13    GCA_900095815.1     2  mcrG  1444106  1444855\n",
       "14    GCA_902763685.1     1  mrtG    12303    13079\n",
       "15    GCA_902795935.1     1  mrtG     4531     5328\n",
       "16    GCA_002506325.1     1  mrtG    29656    30429\n",
       "17    GCA_002506905.1     1  mrtG    54903    55679\n",
       "18    GCA_002509405.1     1  mrtG    49427    50200\n",
       "19    GCA_002496945.1     1  mrtG    48198    48974\n",
       "20    GCA_002505245.1     1  mrtG    24589    25362\n",
       "21    GCA_002498705.1     1  mcrG    22459    23223\n",
       "22    GCA_002498705.1     2  mrtG      225      989\n",
       "23    GCA_002498605.1     1  mrtG     5893     6666\n",
       "24    GCA_002504405.1     1  mrtG    29702    30478\n",
       "25    GCA_002506565.1     1  mrtG    33009    33782\n",
       "26    GCA_002505345.1     1  mrtG    78795    79568\n",
       "27    GCA_002494805.1     1  mrtG    37236    38009\n",
       "28    GCA_002498365.1     1  mrtG   248040   248813\n",
       "29    GCA_002505275.1     1  mrtG    31552    32325\n",
       "30    GCA_002498805.1     1  mrtG    76047    76820\n",
       "31    GCA_002503785.1     1  mrtG    57520    58293\n",
       "32    GCA_002508625.1     1  mrtG    58150    58923\n",
       "33    GCA_002498765.1     1  mrtG   254078   254851\n",
       "34    GCA_002495325.1     1  mrtG    56118    56897\n",
       "35    GCA_002498425.1     1  mrtG    48345    49118\n",
       "36    GCA_002498375.1     1  mrtG    25637    26401\n",
       "37    GCA_002498375.1     2  mcrG    83239    84003\n",
       "38    GCA_002498785.1     1  mrtG    51721    52176\n",
       "39    GCA_002506995.1     1  mrtG    48364    49137\n",
       "40    GCA_002494585.1     1  mrtG     3798     4571\n",
       "41    GCA_002502465.1     1  mrtG    47657    48430\n",
       "42    GCA_000499765.1     1  mcrG  1664303  1665052\n",
       "43    GCA_000499765.1     2  mrtG  1785266  1786066\n",
       "44    GCA_902787415.1     1  mrtG     2086     2862"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Update the dataframe based on specific mrt genes found from protein.faa files \n",
    "df.loc[0, 'gene'] = 'mrtG'\n",
    "df.loc[2, 'gene'] = 'mrtG'\n",
    "df.loc[5, 'gene'] = 'mrtG'\n",
    "df.loc[6, 'gene'] = 'mrtG'\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ffdea7d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1505565</td>\n",
       "      <td>1506359</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>2219996</td>\n",
       "      <td>2220754</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1341629</td>\n",
       "      <td>1342426</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1666316</td>\n",
       "      <td>1667068</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>934501</td>\n",
       "      <td>935250</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2213588</td>\n",
       "      <td>2214388</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000012545.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>378927</td>\n",
       "      <td>379724</td>\n",
       "      <td>Methanosphaera stadtmanae</td>\n",
       "      <td>DSM 3091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12323</td>\n",
       "      <td>13117</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>889</td>\n",
       "      <td>1647</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_002502965.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>129896</td>\n",
       "      <td>130669</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>900346</td>\n",
       "      <td>901095</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>DSM1535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1073013</td>\n",
       "      <td>1073813</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>DSM1535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1413253</td>\n",
       "      <td>1414050</td>\n",
       "      <td>Methanothermobacter wolfeii</td>\n",
       "      <td>SIV6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1444106</td>\n",
       "      <td>1444855</td>\n",
       "      <td>Methanothermobacter wolfeii</td>\n",
       "      <td>SIV6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12303</td>\n",
       "      <td>13079</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_902795935.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>4531</td>\n",
       "      <td>5328</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002506325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29656</td>\n",
       "      <td>30429</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002506905.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>54903</td>\n",
       "      <td>55679</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002509405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>49427</td>\n",
       "      <td>50200</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_002496945.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48198</td>\n",
       "      <td>48974</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_002505245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>24589</td>\n",
       "      <td>25362</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>22459</td>\n",
       "      <td>23223</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>225</td>\n",
       "      <td>989</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_002498605.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>5893</td>\n",
       "      <td>6666</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_002504405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29702</td>\n",
       "      <td>30478</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_002506565.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>33009</td>\n",
       "      <td>33782</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_002505345.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>78795</td>\n",
       "      <td>79568</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_002494805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>37236</td>\n",
       "      <td>38009</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_002498365.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>248040</td>\n",
       "      <td>248813</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_002505275.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>31552</td>\n",
       "      <td>32325</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_002498805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>76047</td>\n",
       "      <td>76820</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_002503785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>57520</td>\n",
       "      <td>58293</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_002508625.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>58150</td>\n",
       "      <td>58923</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_002498765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>254078</td>\n",
       "      <td>254851</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_002495325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>56118</td>\n",
       "      <td>56897</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_002498425.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48345</td>\n",
       "      <td>49118</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>25637</td>\n",
       "      <td>26401</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>83239</td>\n",
       "      <td>84003</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_002498785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>51721</td>\n",
       "      <td>52176</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_002506995.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48364</td>\n",
       "      <td>49137</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCA_002494585.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>3798</td>\n",
       "      <td>4571</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_002502465.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>47657</td>\n",
       "      <td>48430</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1664303</td>\n",
       "      <td>1665052</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>MB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1785266</td>\n",
       "      <td>1786066</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>MB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2086</td>\n",
       "      <td>2862</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession index  gene    start     stop  \\\n",
       "0     GCA_001477655.1     1  mrtG  1505565  1506359   \n",
       "1     GCA_001477655.1     2  mcrG  2219996  2220754   \n",
       "2     GCA_001563245.1     1  mrtG  1341629  1342426   \n",
       "3     GCA_001563245.1     2  mcrG  1666316  1667068   \n",
       "4     GCA_000762265.1     1  mcrG   934501   935250   \n",
       "5     GCA_000762265.1     2  mrtG  2213588  2214388   \n",
       "6     GCA_000012545.1     1  mrtG   378927   379724   \n",
       "7     GCA_902762095.1     1  mrtG    12323    13117   \n",
       "8     GCA_902762095.1     2  mcrG      889     1647   \n",
       "9     GCA_002502965.1     1  mrtG   129896   130669   \n",
       "10    GCA_000953115.1     1  mcrG   900346   901095   \n",
       "11    GCA_000953115.1     2  mrtG  1073013  1073813   \n",
       "12    GCA_900095815.1     1  mrtG  1413253  1414050   \n",
       "13    GCA_900095815.1     2  mcrG  1444106  1444855   \n",
       "14    GCA_902763685.1     1  mrtG    12303    13079   \n",
       "15    GCA_902795935.1     1  mrtG     4531     5328   \n",
       "16    GCA_002506325.1     1  mrtG    29656    30429   \n",
       "17    GCA_002506905.1     1  mrtG    54903    55679   \n",
       "18    GCA_002509405.1     1  mrtG    49427    50200   \n",
       "19    GCA_002496945.1     1  mrtG    48198    48974   \n",
       "20    GCA_002505245.1     1  mrtG    24589    25362   \n",
       "21    GCA_002498705.1     1  mcrG    22459    23223   \n",
       "22    GCA_002498705.1     2  mrtG      225      989   \n",
       "23    GCA_002498605.1     1  mrtG     5893     6666   \n",
       "24    GCA_002504405.1     1  mrtG    29702    30478   \n",
       "25    GCA_002506565.1     1  mrtG    33009    33782   \n",
       "26    GCA_002505345.1     1  mrtG    78795    79568   \n",
       "27    GCA_002494805.1     1  mrtG    37236    38009   \n",
       "28    GCA_002498365.1     1  mrtG   248040   248813   \n",
       "29    GCA_002505275.1     1  mrtG    31552    32325   \n",
       "30    GCA_002498805.1     1  mrtG    76047    76820   \n",
       "31    GCA_002503785.1     1  mrtG    57520    58293   \n",
       "32    GCA_002508625.1     1  mrtG    58150    58923   \n",
       "33    GCA_002498765.1     1  mrtG   254078   254851   \n",
       "34    GCA_002495325.1     1  mrtG    56118    56897   \n",
       "35    GCA_002498425.1     1  mrtG    48345    49118   \n",
       "36    GCA_002498375.1     1  mrtG    25637    26401   \n",
       "37    GCA_002498375.1     2  mcrG    83239    84003   \n",
       "38    GCA_002498785.1     1  mrtG    51721    52176   \n",
       "39    GCA_002506995.1     1  mrtG    48364    49137   \n",
       "40    GCA_002494585.1     1  mrtG     3798     4571   \n",
       "41    GCA_002502465.1     1  mrtG    47657    48430   \n",
       "42    GCA_000499765.1     1  mcrG  1664303  1665052   \n",
       "43    GCA_000499765.1     2  mrtG  1785266  1786066   \n",
       "44    GCA_902787415.1     1  mrtG     2086     2862   \n",
       "\n",
       "                        organism_name    strain  \n",
       "0         Methanobrevibacter millerae       SM9  \n",
       "1         Methanobrevibacter millerae       SM9  \n",
       "2          Methanobrevibacter olleyae      YLM1  \n",
       "3          Methanobrevibacter olleyae      YLM1  \n",
       "4         Methanobacterium formicicum      BRM9  \n",
       "5         Methanobacterium formicicum      BRM9  \n",
       "6          Methanosphaera stadtmanae   DSM 3091  \n",
       "7       uncultured Methanobrevibacter  RUG10736  \n",
       "8       uncultured Methanobrevibacter  RUG10736  \n",
       "9   Methanomassiliicoccaceae archaeon    UBA381  \n",
       "10        Methanobacterium formicicum   DSM1535  \n",
       "11        Methanobacterium formicicum   DSM1535  \n",
       "12        Methanothermobacter wolfeii      SIV6  \n",
       "13        Methanothermobacter wolfeii      SIV6  \n",
       "14              methanogenic archaeon  RUG10898  \n",
       "15      uncultured Methanobrevibacter  RUG14110  \n",
       "16  Methanomassiliicoccaceae archaeon    UBA314  \n",
       "17  Methanomassiliicoccaceae archaeon    UBA328  \n",
       "18  Methanomassiliicoccaceae archaeon     UBA72  \n",
       "19  Methanomassiliicoccaceae archaeon    UBA408  \n",
       "20  Methanomassiliicoccaceae archaeon    UBA422  \n",
       "21                 Methanocorpusculum    UBA362  \n",
       "22                 Methanocorpusculum    UBA362  \n",
       "23  Methanomassiliicoccaceae archaeon    UBA404  \n",
       "24  Methanomassiliicoccaceae archaeon     UBA71  \n",
       "25  Methanomassiliicoccaceae archaeon    UBA333  \n",
       "26  Methanomassiliicoccaceae archaeon    UBA370  \n",
       "27  Methanomassiliicoccaceae archaeon    UBA482  \n",
       "28  Methanomassiliicoccaceae archaeon    UBA421  \n",
       "29  Methanomassiliicoccaceae archaeon    UBA371  \n",
       "30  Methanomassiliicoccaceae archaeon    UBA399  \n",
       "31  Methanomassiliicoccaceae archaeon     UBA75  \n",
       "32  Methanomassiliicoccaceae archaeon    UBA344  \n",
       "33  Methanomassiliicoccaceae archaeon    UBA343  \n",
       "34  Methanomassiliicoccaceae archaeon    UBA593  \n",
       "35  Methanomassiliicoccaceae archaeon    UBA317  \n",
       "36                 Methanocorpusculum    UBA424  \n",
       "37                 Methanocorpusculum    UBA424  \n",
       "38  Methanomassiliicoccaceae archaeon    UBA400  \n",
       "39  Methanomassiliicoccaceae archaeon    UBA427  \n",
       "40  Methanomassiliicoccaceae archaeon    UBA409  \n",
       "41  Methanomassiliicoccaceae archaeon    UBA296  \n",
       "42                   Methanobacterium       MB1  \n",
       "43                   Methanobacterium       MB1  \n",
       "44              methanogenic archaeon  RUG13270  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species = pd.read_csv('/Users/gracedai/Downloads/both.csv')\n",
    "species = species[['assembly_accession', 'organism_name', 'strain']]\n",
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='assembly_accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f5797e26",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to find fasta file in a folder and extract the gene sequence\n",
    "def get_sequence(folder_name, start, stop):\n",
    "    fasta_file = [file for file in os.listdir(folder_name) if file.endswith(\".fna\")]\n",
    "    print(fasta_file)\n",
    "    if not fasta_file:\n",
    "        return None\n",
    "    fasta_path = os.path.join(folder_name, fasta_file[0])\n",
    "    with open(fasta_path, \"r\") as file:\n",
    "        lines = file.readlines()\n",
    "        sequence = \"\".join(line.strip() for line in lines[1:] if not line.startswith(\">\"))\n",
    "    return sequence[start-1:stop]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "30c0470e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCF_001477655.1_ASM147765v1_genomic.fna']\n",
      "['GCF_001477655.1_ASM147765v1_genomic.fna']\n",
      "['GCF_001563245.1_ASM156324v1_genomic.fna']\n",
      "['GCF_001563245.1_ASM156324v1_genomic.fna']\n",
      "['GCF_000762265.1_ASM76226v1_genomic.fna']\n",
      "['GCF_000762265.1_ASM76226v1_genomic.fna']\n",
      "['GCF_000012545.1_ASM1254v1_genomic.fna']\n",
      "['GCA_902762095.1_Rumen_uncultured_genome_RUG10736_genomic.fna']\n",
      "['GCA_902762095.1_Rumen_uncultured_genome_RUG10736_genomic.fna']\n",
      "['GCA_002502965.1_ASM250296v1_genomic.fna']\n",
      "['GCA_000953115.1_DSM1535_genomic.fna']\n",
      "['GCA_000953115.1_DSM1535_genomic.fna']\n",
      "['GCA_900095815.1_SIV6_genomic.fna']\n",
      "['GCA_900095815.1_SIV6_genomic.fna']\n",
      "['GCA_902763685.1_Rumen_uncultured_genome_RUG10898_genomic.fna']\n",
      "['GCA_902795935.1_Rumen_uncultured_genome_RUG14110_genomic.fna']\n",
      "['GCA_002506325.1_ASM250632v1_genomic.fna']\n",
      "['GCA_002506905.1_ASM250690v1_genomic.fna']\n",
      "['GCA_002509405.1_ASM250940v1_genomic.fna']\n",
      "['GCA_002496945.1_ASM249694v1_genomic.fna']\n",
      "['GCA_002505245.1_ASM250524v1_genomic.fna']\n",
      "['GCA_002498705.1_ASM249870v1_genomic.fna']\n",
      "['GCA_002498705.1_ASM249870v1_genomic.fna']\n",
      "['GCA_002498605.1_ASM249860v1_genomic.fna']\n",
      "['GCA_002504405.1_ASM250440v1_genomic.fna']\n",
      "['GCA_002506565.1_ASM250656v1_genomic.fna']\n",
      "['GCA_002505345.1_ASM250534v1_genomic.fna']\n",
      "['GCA_002494805.1_ASM249480v1_genomic.fna']\n",
      "['GCA_002498365.1_ASM249836v1_genomic.fna']\n",
      "['GCA_002505275.1_ASM250527v1_genomic.fna']\n",
      "['GCA_002498805.1_ASM249880v1_genomic.fna']\n",
      "['GCA_002503785.1_ASM250378v1_genomic.fna']\n",
      "['GCA_002508625.1_ASM250862v1_genomic.fna']\n",
      "['GCA_002498765.1_ASM249876v1_genomic.fna']\n",
      "['GCA_002495325.1_ASM249532v1_genomic.fna']\n",
      "['GCA_002498425.1_ASM249842v1_genomic.fna']\n",
      "['GCA_002498375.1_ASM249837v1_genomic.fna']\n",
      "['GCA_002498375.1_ASM249837v1_genomic.fna']\n",
      "['GCA_002498785.1_ASM249878v1_genomic.fna']\n",
      "['GCA_002506995.1_ASM250699v1_genomic.fna']\n",
      "['GCA_002494585.1_ASM249458v1_genomic.fna']\n",
      "['GCA_002502465.1_ASM250246v1_genomic.fna']\n",
      "['GCA_000499765.1_Methanobacterium_genomic.fna']\n",
      "['GCA_000499765.1_Methanobacterium_genomic.fna']\n",
      "['GCA_902787415.1_Rumen_uncultured_genome_RUG13270_genomic.fna']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1505565</td>\n",
       "      <td>1506359</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>CTATTCACCTAATTTCTTTTTTAAATCATTAGAAAATACTCCCATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>2219996</td>\n",
       "      <td>2220754</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>TTACCATCCTGCGTTTGGATCAAATGCACCAAAGGATCTGGAAACG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1341629</td>\n",
       "      <td>1342426</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>ATGACATATAAAGCTCAATTTGCTCCAGGAGATACTAAAATTGCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1666316</td>\n",
       "      <td>1667068</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>CTATAATTCTGGGTTGTAACCAAAATTAGATCTGGTGACGTGTATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>934501</td>\n",
       "      <td>935250</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "      <td>TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACGTGTATCCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000762265.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2213588</td>\n",
       "      <td>2214388</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "      <td>ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000012545.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>378927</td>\n",
       "      <td>379724</td>\n",
       "      <td>Methanosphaera stadtmanae</td>\n",
       "      <td>DSM 3091</td>\n",
       "      <td>ATGGCATATGAAGTACAATATTATCCTGGAGAAAGTGTTATTGCAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12323</td>\n",
       "      <td>13117</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "      <td>TTCATTGTCAAATTCAAAAGCAATTTAACAGCAGTGGGAGCTGTCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>889</td>\n",
       "      <td>1647</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "      <td>GCCGAATCATATATCTAAGGTTCTTCGTGAACTAAAGGAAACTGGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_002502965.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>129896</td>\n",
       "      <td>130669</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA381</td>\n",
       "      <td>AATCCGTGCATCCGCCCTCGAAGAATAGTCGTTCCATCCTGTCGTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>900346</td>\n",
       "      <td>901095</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>DSM1535</td>\n",
       "      <td>TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACGTGTATCCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000953115.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1073013</td>\n",
       "      <td>1073813</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>DSM1535</td>\n",
       "      <td>ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1413253</td>\n",
       "      <td>1414050</td>\n",
       "      <td>Methanothermobacter wolfeii</td>\n",
       "      <td>SIV6</td>\n",
       "      <td>TTAATCATCACCTAGCCTTTTTCTTAGGTCTTCCTTGAATACATCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_900095815.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1444106</td>\n",
       "      <td>1444855</td>\n",
       "      <td>Methanothermobacter wolfeii</td>\n",
       "      <td>SIV6</td>\n",
       "      <td>TTATTCTGGGTTGTACCCTCCTTGTGATCTTAAAACGTGGATTCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>12303</td>\n",
       "      <td>13079</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "      <td>TTCCGGGGGTGAAGTGGATCAGTTCCATATCCAGGTCGCCGACGTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_902795935.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>4531</td>\n",
       "      <td>5328</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14110</td>\n",
       "      <td>GTATGCATTTCATGGATCCTTTTAATTTTGATACAGATAGGAGTAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002506325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29656</td>\n",
       "      <td>30429</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA314</td>\n",
       "      <td>GAGGTCCATGCCCACGATCATGGGCAGGATGTTCAGCTTGGGCAGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002506905.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>54903</td>\n",
       "      <td>55679</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA328</td>\n",
       "      <td>GTGCCTGGAAATGCTCGACCGTCTTCCTGTAATTCTCCATCGCCAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002509405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>49427</td>\n",
       "      <td>50200</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA72</td>\n",
       "      <td>TCTTGCGCTCAGAGGCATAGAGTCCGACGAGCTCGACAGGGGCTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_002496945.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48198</td>\n",
       "      <td>48974</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA408</td>\n",
       "      <td>CGATCTTCTTCACGGCGTCCCTCAGTATAATGGCGGATTTCGGGTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_002505245.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>24589</td>\n",
       "      <td>25362</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA422</td>\n",
       "      <td>TTCGCCTCGATGGCCTTCGTTACGACGGGTTCCGAATAGTGGGTCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>22459</td>\n",
       "      <td>23223</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA362</td>\n",
       "      <td>TGAAGGAGTTGGGGGTCGATGTATGCGGGGAGAACGGCGAGTACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_002498705.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>225</td>\n",
       "      <td>989</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA362</td>\n",
       "      <td>TATTATTCCACGTGAGAACCACGATAACAAATGGATTTCCTGTAGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_002498605.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>5893</td>\n",
       "      <td>6666</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA404</td>\n",
       "      <td>AGATAACCGTAATAGACCCCGGAGCAAAAAACAATCTTTCTCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_002504405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>29702</td>\n",
       "      <td>30478</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA71</td>\n",
       "      <td>CGGTTCCGTCCATCCGGATGCGCTCGGCCACTGGCCCCGCGGAGTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_002506565.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>33009</td>\n",
       "      <td>33782</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA333</td>\n",
       "      <td>CGCGAAGGAAGCATCGGCATCCCGTTCCCCGACACTCTTTACAAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_002505345.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>78795</td>\n",
       "      <td>79568</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA370</td>\n",
       "      <td>GAGCTGGCGGAGTCTTTCGAAGTGATAAAGGAACTTGCAGAATCCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_002494805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>37236</td>\n",
       "      <td>38009</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA482</td>\n",
       "      <td>GCATTAAGCCTTATCCCCTGAAGGAGCTCCTGACGATATTCCTTTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_002498365.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>248040</td>\n",
       "      <td>248813</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA421</td>\n",
       "      <td>TCACACCCCCTTCACTTTGAAAGGCTGGTATCCGCCCAGTGTCCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_002505275.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>31552</td>\n",
       "      <td>32325</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA371</td>\n",
       "      <td>GCCTCTGGATCAGTGTTTTGAATTCCTCGTCATCGCGAACACCCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_002498805.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>76047</td>\n",
       "      <td>76820</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA399</td>\n",
       "      <td>AGCCCCGTAGGCAGCATATCTGAAAGGAGGATCCTCGGACTGATCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_002503785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>57520</td>\n",
       "      <td>58293</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA75</td>\n",
       "      <td>CGTATATGAGAAGGAAAATAAAAGCGAGGATTATAGAATTCCCGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_002508625.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>58150</td>\n",
       "      <td>58923</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA344</td>\n",
       "      <td>ACATCCGCCGGGCCGAATATGTTCCGGTGACGGACGAGGAGGCGGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_002498765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>254078</td>\n",
       "      <td>254851</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA343</td>\n",
       "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTCGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_002495325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>56118</td>\n",
       "      <td>56897</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA593</td>\n",
       "      <td>AGGAGCCGCTGACCGATCTGACGGCCCTCCGCTACGGGGTGCGGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_002498425.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48345</td>\n",
       "      <td>49118</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA317</td>\n",
       "      <td>AGTTCTATGGAACTCTGTATGTGGATGTTTACGACGATGAAGGAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>25637</td>\n",
       "      <td>26401</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA424</td>\n",
       "      <td>TGATCGCGATCGAACTGGATCCGGGACGTTTTGCGGCACTAAAACA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_002498375.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>83239</td>\n",
       "      <td>84003</td>\n",
       "      <td>Methanocorpusculum</td>\n",
       "      <td>UBA424</td>\n",
       "      <td>TTCTGAATATATTCTCGATACATGTTACATCCCCGTTGGATACCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_002498785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>51721</td>\n",
       "      <td>52176</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA400</td>\n",
       "      <td>ACCCGTCGGATTGGACGGGTTGTTCACATACATGATCCTGGTCTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_002506995.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>48364</td>\n",
       "      <td>49137</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA427</td>\n",
       "      <td>CCCCCTGGAGACCGATAATGACACGAAAAACGATGCAGAAGACGTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCA_002494585.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>3798</td>\n",
       "      <td>4571</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA409</td>\n",
       "      <td>CCTTTCCGCATCCACGGACATGGGCGACACGAACAACAGCAAAGTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_002502465.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>47657</td>\n",
       "      <td>48430</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA296</td>\n",
       "      <td>GGAGATGCGAGGTGGACCGTGACAAGTGCATCAAGTGCTACACATG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1664303</td>\n",
       "      <td>1665052</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>MB1</td>\n",
       "      <td>TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACATGTATTCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_000499765.1</td>\n",
       "      <td>2</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>1785266</td>\n",
       "      <td>1786066</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>MB1</td>\n",
       "      <td>ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>1</td>\n",
       "      <td>mrtG</td>\n",
       "      <td>2086</td>\n",
       "      <td>2862</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CAGCAGCAGCCAAGGAGGCAGAGGAGAAAGCCGCCGCAGAGGAGAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession index  gene    start     stop  \\\n",
       "0     GCA_001477655.1     1  mrtG  1505565  1506359   \n",
       "1     GCA_001477655.1     2  mcrG  2219996  2220754   \n",
       "2     GCA_001563245.1     1  mrtG  1341629  1342426   \n",
       "3     GCA_001563245.1     2  mcrG  1666316  1667068   \n",
       "4     GCA_000762265.1     1  mcrG   934501   935250   \n",
       "5     GCA_000762265.1     2  mrtG  2213588  2214388   \n",
       "6     GCA_000012545.1     1  mrtG   378927   379724   \n",
       "7     GCA_902762095.1     1  mrtG    12323    13117   \n",
       "8     GCA_902762095.1     2  mcrG      889     1647   \n",
       "9     GCA_002502965.1     1  mrtG   129896   130669   \n",
       "10    GCA_000953115.1     1  mcrG   900346   901095   \n",
       "11    GCA_000953115.1     2  mrtG  1073013  1073813   \n",
       "12    GCA_900095815.1     1  mrtG  1413253  1414050   \n",
       "13    GCA_900095815.1     2  mcrG  1444106  1444855   \n",
       "14    GCA_902763685.1     1  mrtG    12303    13079   \n",
       "15    GCA_902795935.1     1  mrtG     4531     5328   \n",
       "16    GCA_002506325.1     1  mrtG    29656    30429   \n",
       "17    GCA_002506905.1     1  mrtG    54903    55679   \n",
       "18    GCA_002509405.1     1  mrtG    49427    50200   \n",
       "19    GCA_002496945.1     1  mrtG    48198    48974   \n",
       "20    GCA_002505245.1     1  mrtG    24589    25362   \n",
       "21    GCA_002498705.1     1  mcrG    22459    23223   \n",
       "22    GCA_002498705.1     2  mrtG      225      989   \n",
       "23    GCA_002498605.1     1  mrtG     5893     6666   \n",
       "24    GCA_002504405.1     1  mrtG    29702    30478   \n",
       "25    GCA_002506565.1     1  mrtG    33009    33782   \n",
       "26    GCA_002505345.1     1  mrtG    78795    79568   \n",
       "27    GCA_002494805.1     1  mrtG    37236    38009   \n",
       "28    GCA_002498365.1     1  mrtG   248040   248813   \n",
       "29    GCA_002505275.1     1  mrtG    31552    32325   \n",
       "30    GCA_002498805.1     1  mrtG    76047    76820   \n",
       "31    GCA_002503785.1     1  mrtG    57520    58293   \n",
       "32    GCA_002508625.1     1  mrtG    58150    58923   \n",
       "33    GCA_002498765.1     1  mrtG   254078   254851   \n",
       "34    GCA_002495325.1     1  mrtG    56118    56897   \n",
       "35    GCA_002498425.1     1  mrtG    48345    49118   \n",
       "36    GCA_002498375.1     1  mrtG    25637    26401   \n",
       "37    GCA_002498375.1     2  mcrG    83239    84003   \n",
       "38    GCA_002498785.1     1  mrtG    51721    52176   \n",
       "39    GCA_002506995.1     1  mrtG    48364    49137   \n",
       "40    GCA_002494585.1     1  mrtG     3798     4571   \n",
       "41    GCA_002502465.1     1  mrtG    47657    48430   \n",
       "42    GCA_000499765.1     1  mcrG  1664303  1665052   \n",
       "43    GCA_000499765.1     2  mrtG  1785266  1786066   \n",
       "44    GCA_902787415.1     1  mrtG     2086     2862   \n",
       "\n",
       "                        organism_name    strain  \\\n",
       "0         Methanobrevibacter millerae       SM9   \n",
       "1         Methanobrevibacter millerae       SM9   \n",
       "2          Methanobrevibacter olleyae      YLM1   \n",
       "3          Methanobrevibacter olleyae      YLM1   \n",
       "4         Methanobacterium formicicum      BRM9   \n",
       "5         Methanobacterium formicicum      BRM9   \n",
       "6          Methanosphaera stadtmanae   DSM 3091   \n",
       "7       uncultured Methanobrevibacter  RUG10736   \n",
       "8       uncultured Methanobrevibacter  RUG10736   \n",
       "9   Methanomassiliicoccaceae archaeon    UBA381   \n",
       "10        Methanobacterium formicicum   DSM1535   \n",
       "11        Methanobacterium formicicum   DSM1535   \n",
       "12        Methanothermobacter wolfeii      SIV6   \n",
       "13        Methanothermobacter wolfeii      SIV6   \n",
       "14              methanogenic archaeon  RUG10898   \n",
       "15      uncultured Methanobrevibacter  RUG14110   \n",
       "16  Methanomassiliicoccaceae archaeon    UBA314   \n",
       "17  Methanomassiliicoccaceae archaeon    UBA328   \n",
       "18  Methanomassiliicoccaceae archaeon     UBA72   \n",
       "19  Methanomassiliicoccaceae archaeon    UBA408   \n",
       "20  Methanomassiliicoccaceae archaeon    UBA422   \n",
       "21                 Methanocorpusculum    UBA362   \n",
       "22                 Methanocorpusculum    UBA362   \n",
       "23  Methanomassiliicoccaceae archaeon    UBA404   \n",
       "24  Methanomassiliicoccaceae archaeon     UBA71   \n",
       "25  Methanomassiliicoccaceae archaeon    UBA333   \n",
       "26  Methanomassiliicoccaceae archaeon    UBA370   \n",
       "27  Methanomassiliicoccaceae archaeon    UBA482   \n",
       "28  Methanomassiliicoccaceae archaeon    UBA421   \n",
       "29  Methanomassiliicoccaceae archaeon    UBA371   \n",
       "30  Methanomassiliicoccaceae archaeon    UBA399   \n",
       "31  Methanomassiliicoccaceae archaeon     UBA75   \n",
       "32  Methanomassiliicoccaceae archaeon    UBA344   \n",
       "33  Methanomassiliicoccaceae archaeon    UBA343   \n",
       "34  Methanomassiliicoccaceae archaeon    UBA593   \n",
       "35  Methanomassiliicoccaceae archaeon    UBA317   \n",
       "36                 Methanocorpusculum    UBA424   \n",
       "37                 Methanocorpusculum    UBA424   \n",
       "38  Methanomassiliicoccaceae archaeon    UBA400   \n",
       "39  Methanomassiliicoccaceae archaeon    UBA427   \n",
       "40  Methanomassiliicoccaceae archaeon    UBA409   \n",
       "41  Methanomassiliicoccaceae archaeon    UBA296   \n",
       "42                   Methanobacterium       MB1   \n",
       "43                   Methanobacterium       MB1   \n",
       "44              methanogenic archaeon  RUG13270   \n",
       "\n",
       "                                             Sequence  \n",
       "0   CTATTCACCTAATTTCTTTTTTAAATCATTAGAAAATACTCCCATA...  \n",
       "1   TTACCATCCTGCGTTTGGATCAAATGCACCAAAGGATCTGGAAACG...  \n",
       "2   ATGACATATAAAGCTCAATTTGCTCCAGGAGATACTAAAATTGCTG...  \n",
       "3   CTATAATTCTGGGTTGTAACCAAAATTAGATCTGGTGACGTGTATT...  \n",
       "4   TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACGTGTATCCTG...  \n",
       "5   ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...  \n",
       "6   ATGGCATATGAAGTACAATATTATCCTGGAGAAAGTGTTATTGCAG...  \n",
       "7   TTCATTGTCAAATTCAAAAGCAATTTAACAGCAGTGGGAGCTGTCA...  \n",
       "8   GCCGAATCATATATCTAAGGTTCTTCGTGAACTAAAGGAAACTGGC...  \n",
       "9   AATCCGTGCATCCGCCCTCGAAGAATAGTCGTTCCATCCTGTCGTA...  \n",
       "10  TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACGTGTATCCTG...  \n",
       "11  ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...  \n",
       "12  TTAATCATCACCTAGCCTTTTTCTTAGGTCTTCCTTGAATACATCC...  \n",
       "13  TTATTCTGGGTTGTACCCTCCTTGTGATCTTAAAACGTGGATTCTC...  \n",
       "14  TTCCGGGGGTGAAGTGGATCAGTTCCATATCCAGGTCGCCGACGTT...  \n",
       "15  GTATGCATTTCATGGATCCTTTTAATTTTGATACAGATAGGAGTAA...  \n",
       "16  GAGGTCCATGCCCACGATCATGGGCAGGATGTTCAGCTTGGGCAGC...  \n",
       "17  GTGCCTGGAAATGCTCGACCGTCTTCCTGTAATTCTCCATCGCCAT...  \n",
       "18  TCTTGCGCTCAGAGGCATAGAGTCCGACGAGCTCGACAGGGGCTTC...  \n",
       "19  CGATCTTCTTCACGGCGTCCCTCAGTATAATGGCGGATTTCGGGTC...  \n",
       "20  TTCGCCTCGATGGCCTTCGTTACGACGGGTTCCGAATAGTGGGTCA...  \n",
       "21  TGAAGGAGTTGGGGGTCGATGTATGCGGGGAGAACGGCGAGTACCA...  \n",
       "22  TATTATTCCACGTGAGAACCACGATAACAAATGGATTTCCTGTAGT...  \n",
       "23  AGATAACCGTAATAGACCCCGGAGCAAAAAACAATCTTTCTCTTGC...  \n",
       "24  CGGTTCCGTCCATCCGGATGCGCTCGGCCACTGGCCCCGCGGAGTC...  \n",
       "25  CGCGAAGGAAGCATCGGCATCCCGTTCCCCGACACTCTTTACAAGA...  \n",
       "26  GAGCTGGCGGAGTCTTTCGAAGTGATAAAGGAACTTGCAGAATCCG...  \n",
       "27  GCATTAAGCCTTATCCCCTGAAGGAGCTCCTGACGATATTCCTTTA...  \n",
       "28  TCACACCCCCTTCACTTTGAAAGGCTGGTATCCGCCCAGTGTCCTG...  \n",
       "29  GCCTCTGGATCAGTGTTTTGAATTCCTCGTCATCGCGAACACCCGA...  \n",
       "30  AGCCCCGTAGGCAGCATATCTGAAAGGAGGATCCTCGGACTGATCG...  \n",
       "31  CGTATATGAGAAGGAAAATAAAAGCGAGGATTATAGAATTCCCGCG...  \n",
       "32  ACATCCGCCGGGCCGAATATGTTCCGGTGACGGACGAGGAGGCGGT...  \n",
       "33  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTCGT...  \n",
       "34  AGGAGCCGCTGACCGATCTGACGGCCCTCCGCTACGGGGTGCGGAT...  \n",
       "35  AGTTCTATGGAACTCTGTATGTGGATGTTTACGACGATGAAGGAAA...  \n",
       "36  TGATCGCGATCGAACTGGATCCGGGACGTTTTGCGGCACTAAAACA...  \n",
       "37  TTCTGAATATATTCTCGATACATGTTACATCCCCGTTGGATACCGA...  \n",
       "38  ACCCGTCGGATTGGACGGGTTGTTCACATACATGATCCTGGTCTTT...  \n",
       "39  CCCCCTGGAGACCGATAATGACACGAAAAACGATGCAGAAGACGTC...  \n",
       "40  CCTTTCCGCATCCACGGACATGGGCGACACGAACAACAGCAAAGTT...  \n",
       "41  GGAGATGCGAGGTGGACCGTGACAAGTGCATCAAGTGCTACACATG...  \n",
       "42  TTATTCTGGACAGAATCCACCTTTAGATCGGCTCACATGTATTCTG...  \n",
       "43  ATGTCATACAAACCCCAGTACACCCCTGGTGAAACAAAAATAGCTC...  \n",
       "44  CAGCAGCAGCCAAGGAGGCAGAGGAGAAAGCCGCCGCAGAGGAGAA...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences = []\n",
    "\n",
    "for _, row in df.iterrows():\n",
    "    folder_name = row[\"assembly_accession\"]\n",
    "    start, stop = row[\"start\"], row[\"stop\"]\n",
    "    sequence = get_sequence(folder_name, start, stop) #extract the gene sequence using function\n",
    "    sequences.append(sequence)\n",
    "\n",
    "# add column of sequences to existing dataframe\n",
    "df[\"Sequence\"] = sequences\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4df05235",
   "metadata": {},
   "outputs": [],
   "source": [
    "#out_fasta should be \"name.fasta\" to ensure the output file is in fasta format\n",
    "def to_fasta(df, out_fasta):\n",
    "    df[\"organism_name\"] = df[\"organism_name\"].str.replace(\" \",\"-\")\n",
    "    df[\"strain\"] = df[\"strain\"].str.replace(\" \",\"-\")\n",
    "    with open(out_fasta, 'w') as file:\n",
    "        for _, row in df.iterrows():\n",
    "            header = f'>{row[\"assembly_accession\"]}|{row[\"organism_name\"]}|{row[\"strain\"]}|{row[\"gene\"]}|{row[\"start\"]}|{row[\"stop\"]}'\n",
    "            sequence = row[\"Sequence\"]\n",
    "            file.write(f'{header}\\n{sequence}\\n')\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "13bb1b36",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export to local fasta file\n",
    "to_fasta(df, \"mcrG_mrtG.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6688d6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
