{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8270eb45",
   "metadata": {},
   "source": [
    "A Python script for retrieving given gene sequences. The output is a fasta file (name=gene name), with headers that denote the species name start and end base pair, and the accession number of the species. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aa0ed9",
   "metadata": {},
   "source": [
    "How it works:\n",
    "A. Create a dataframe of metadata regarding the genes \n",
    "1. iterate through all gff files and find the start and stop base pair number of the desired gene \n",
    "2. input start and stop in different columns in a dataframe, on the corresponding row (with species name, strain, accession number) \n",
    "\n",
    "B. Gather the sequences and input into dataframe\n",
    "1. unzip all fasta files in the directory\n",
    "2. for each fasta file, take the string using the start and stop index (dont include header)\n",
    "3. input that string into corresponding row of dataframe, using the accession number as a key  \n",
    "\n",
    "C. Output dataframe into a formatted fasta file \n",
    "1. create a header based on row in dataframe \n",
    "2. append the gene sequence \n",
    "3. create new line for new row in df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a84427e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/gracedai/Documents/MethanogeneDB\n"
     ]
    }
   ],
   "source": [
    "#Check that we are in the correct working directory\n",
    "import os \n",
    "print(os.getcwd())\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1032c329",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/gracedai/Documents/MethanogeneDB/prokka_out\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f642bd3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902788045.1/GCA_902788045.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774685.1/GCA_902774685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497785.1/GCA_002497785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502925.1/GCA_002502925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506795.1/GCA_002506795.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508885.1/GCA_002508885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497395.1/GCA_002497395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501765.1/GCA_002501765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759285.1/GCA_902759285.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902764425.1/GCA_902764425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502965.1/GCA_002502965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800375.1/GCA_902800375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496085.1/GCA_002496085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800035.1/GCA_902800035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494885.1/GCA_002494885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762095.1/GCA_902762095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505895.1/GCA_002505895.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506085.1/GCA_002506085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496495.1/GCA_002496495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902768855.1/GCA_902768855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784455.1/GCA_902784455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784715.1/GCA_902784715.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902789505.1/GCA_902789505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502975.1/GCA_002502975.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503105.1/GCA_002503105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762085.1/GCA_902762085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506585.1/GCA_002506585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506095.1/GCA_002506095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495885.1/GCA_002495885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502405.1/GCA_002502405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503005.1/GCA_002503005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902788255.1/GCA_902788255.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759195.1/GCA_902759195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503455.1/GCA_002503455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902769175.1/GCA_902769175.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496395.1/GCA_002496395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503825.1/GCA_002503825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902801725.1/GCA_902801725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902770715.1/GCA_902770715.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506785.1/GCA_002506785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902763685.1/GCA_902763685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496295.1/GCA_002496295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504965.1/GCA_002504965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508925.1/GCA_002508925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496325.1/GCA_002496325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498825.1/GCA_002498825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496805.1/GCA_002496805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759475.1/GCA_902759475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902795935.1/GCA_902795935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496165.1/GCA_002496165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499445.1/GCA_002499445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497075.1/GCA_002497075.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902781855.1/GCA_902781855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506325.1/GCA_002506325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508825.1/GCA_002508825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506905.1/GCA_002506905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509545.1/GCA_002509545.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506845.1/GCA_002506845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509405.1/GCA_002509405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902758425.1/GCA_902758425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496945.1/GCA_002496945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902756745.1/GCA_902756745.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775675.1/GCA_902775675.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495445.1/GCA_002495445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509505.1/GCA_002509505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506535.1/GCA_002506535.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509015.1/GCA_002509015.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505605.1/GCA_002505605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509645.1/GCA_002509645.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504605.1/GCA_002504605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508505.1/GCA_002508505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902792995.1/GCA_902792995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507025.1/GCA_002507025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501695.1/GCA_002501695.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494705.1/GCA_002494705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902758945.1/GCA_902758945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498505.1/GCA_002498505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902786095.1/GCA_902786095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497365.1/GCA_002497365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499755.1/GCA_002499755.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505925.1/GCA_002505925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508745.1/GCA_002508745.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509655.1/GCA_002509655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505245.1/GCA_002505245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507065.1/GCA_002507065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508545.1/GCA_002508545.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504505.1/GCA_002504505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508935.1/GCA_002508935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496565.1/GCA_002496565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498445.1/GCA_002498445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759335.1/GCA_902759335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902799105.1/GCA_902799105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503885.1/GCA_002503885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498705.1/GCA_002498705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775725.1/GCA_902775725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499825.1/GCA_002499825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498315.1/GCA_002498315.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498605.1/GCA_002498605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902755425.1/GCA_902755425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499145.1/GCA_002499145.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496465.1/GCA_002496465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499455.1/GCA_002499455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506335.1/GCA_002506335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508705.1/GCA_002508705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506915.1/GCA_002506915.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504405.1/GCA_002504405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509045.1/GCA_002509045.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506565.1/GCA_002506565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505345.1/GCA_002505345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508675.1/GCA_002508675.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506415.1/GCA_002506415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509425.1/GCA_002509425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506105.1/GCA_002506105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505725.1/GCA_002505725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902782945.1/GCA_902782945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495565.1/GCA_002495565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496345.1/GCA_002496345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495625.1/GCA_002495625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494805.1/GCA_002494805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499035.1/GCA_002499035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757365.1/GCA_902757365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495725.1/GCA_002495725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503585.1/GCA_002503585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775005.1/GCA_902775005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498365.1/GCA_002498365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902777475.1/GCA_902777475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508845.1/GCA_002508845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508775.1/GCA_002508775.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506455.1/GCA_002506455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505275.1/GCA_002505275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505765.1/GCA_002505765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508905.1/GCA_002508905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494845.1/GCA_002494845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496555.1/GCA_002496555.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774605.1/GCA_902774605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498805.1/GCA_002498805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503785.1/GCA_002503785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798025.1/GCA_902798025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498225.1/GCA_002498225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498905.1/GCA_002498905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774705.1/GCA_902774705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498065.1/GCA_002498065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902778875.1/GCA_902778875.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508805.1/GCA_002508805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509335.1/GCA_002509335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509565.1/GCA_002509565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508475.1/GCA_002508475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508625.1/GCA_002508625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504525.1/GCA_002504525.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495365.1/GCA_002495365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499905.1/GCA_002499905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502785.1/GCA_002502785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497555.1/GCA_002497555.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784195.1/GCA_902784195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505025.1/GCA_002505025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505535.1/GCA_002505535.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509325.1/GCA_002509325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508725.1/GCA_002508725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507205.1/GCA_002507205.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504765.1/GCA_002504765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509575.1/GCA_002509575.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508665.1/GCA_002508665.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902763115.1/GCA_902763115.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505225.1/GCA_002505225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498765.1/GCA_002498765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495325.1/GCA_002495325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798335.1/GCA_902798335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757275.1/GCA_902757275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497105.1/GCA_002497105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498425.1/GCA_002498425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499025.1/GCA_002499025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495805.1/GCA_002495805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503085.1/GCA_002503085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498375.1/GCA_002498375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784295.1/GCA_902784295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497655.1/GCA_002497655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508425.1/GCA_002508425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762245.1/GCA_902762245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902770085.1/GCA_902770085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508855.1/GCA_002508855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508765.1/GCA_002508765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495685.1/GCA_002495685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503135.1/GCA_002503135.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502845.1/GCA_002502845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902765105.1/GCA_902765105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902785855.1/GCA_902785855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495585.1/GCA_002495585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502635.1/GCA_002502635.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494495.1/GCA_002494495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505485.1/GCA_002505485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902771225.1/GCA_902771225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502735.1/GCA_002502735.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497995.1/GCA_002497995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902802035.1/GCA_902802035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498385.1/GCA_002498385.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502805.1/GCA_002502805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503565.1/GCA_002503565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501655.1/GCA_002501655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800615.1/GCA_902800615.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498785.1/GCA_002498785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902789165.1/GCA_902789165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902765875.1/GCA_902765875.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504585.1/GCA_002504585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509385.1/GCA_002509385.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902790605.1/GCA_902790605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506995.1/GCA_002506995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504485.1/GCA_002504485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494585.1/GCA_002494585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502465.1/GCA_002502465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800455.1/GCA_902800455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757195.1/GCA_902757195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798395.1/GCA_902798395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800515.1/GCA_902800515.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902785165.1/GCA_902785165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498485.1/GCA_002498485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494785.1/GCA_002494785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503125.1/GCA_002503125.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509495.1/GCA_002509495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902764455.1/GCA_902764455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902802065.1/GCA_902802065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902787415.1/GCA_902787415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902768415.1/GCA_902768415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502855.1/GCA_002502855.1.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "223"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# list to store paths to gff files\n",
    "gff_files = []\n",
    "# iterate through current working directory and for each file \n",
    "# append the gff path to a list \n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gff\"):\n",
    "            gff_files.append(os.path.join(root, file))\n",
    "print(gff_files)\n",
    "len(gff_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a4818c09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCA_902788045.1/GCA_902788045.1.gff', 'GCA_902774685.1/GCA_902774685.1.gff', 'GCA_002497785.1/GCA_002497785.1.gff', 'GCA_002502925.1/GCA_002502925.1.gff', 'GCA_002506795.1/GCA_002506795.1.gff', 'GCA_002508885.1/GCA_002508885.1.gff', 'GCA_002497395.1/GCA_002497395.1.gff', 'GCA_002501765.1/GCA_002501765.1.gff', 'GCA_902759285.1/GCA_902759285.1.gff', 'GCA_902764425.1/GCA_902764425.1.gff', 'GCA_002502965.1/GCA_002502965.1.gff', 'GCA_902800375.1/GCA_902800375.1.gff', 'GCA_002496085.1/GCA_002496085.1.gff', 'GCA_902800035.1/GCA_902800035.1.gff', 'GCA_002494885.1/GCA_002494885.1.gff', 'GCA_902762095.1/GCA_902762095.1.gff', 'GCA_002505895.1/GCA_002505895.1.gff', 'GCA_002506085.1/GCA_002506085.1.gff', 'GCA_002496495.1/GCA_002496495.1.gff', 'GCA_902768855.1/GCA_902768855.1.gff', 'GCA_902784455.1/GCA_902784455.1.gff', 'GCA_902784715.1/GCA_902784715.1.gff', 'GCA_902789505.1/GCA_902789505.1.gff', 'GCA_002502975.1/GCA_002502975.1.gff', 'GCA_002503105.1/GCA_002503105.1.gff', 'GCA_902762085.1/GCA_902762085.1.gff', 'GCA_002506585.1/GCA_002506585.1.gff', 'GCA_002506095.1/GCA_002506095.1.gff', 'GCA_002495885.1/GCA_002495885.1.gff', 'GCA_002502405.1/GCA_002502405.1.gff', 'GCA_002503005.1/GCA_002503005.1.gff', 'GCA_902788255.1/GCA_902788255.1.gff', 'GCA_902759195.1/GCA_902759195.1.gff', 'GCA_002503455.1/GCA_002503455.1.gff', 'GCA_902769175.1/GCA_902769175.1.gff', 'GCA_002496395.1/GCA_002496395.1.gff', 'GCA_002503825.1/GCA_002503825.1.gff', 'GCA_902801725.1/GCA_902801725.1.gff', 'GCA_902770715.1/GCA_902770715.1.gff', 'GCA_002506785.1/GCA_002506785.1.gff', 'GCA_902763685.1/GCA_902763685.1.gff', 'GCA_002496295.1/GCA_002496295.1.gff', 'GCA_002504965.1/GCA_002504965.1.gff', 'GCA_002508925.1/GCA_002508925.1.gff', 'GCA_002496325.1/GCA_002496325.1.gff', 'GCA_002498825.1/GCA_002498825.1.gff', 'GCA_002496805.1/GCA_002496805.1.gff', 'GCA_902759475.1/GCA_902759475.1.gff', 'GCA_902795935.1/GCA_902795935.1.gff', 'GCA_002496165.1/GCA_002496165.1.gff', 'GCA_002499445.1/GCA_002499445.1.gff', 'GCA_002497075.1/GCA_002497075.1.gff', 'GCA_902781855.1/GCA_902781855.1.gff', 'GCA_002506325.1/GCA_002506325.1.gff', 'GCA_002508825.1/GCA_002508825.1.gff', 'GCA_002506905.1/GCA_002506905.1.gff', 'GCA_002509545.1/GCA_002509545.1.gff', 'GCA_002506845.1/GCA_002506845.1.gff', 'GCA_002509405.1/GCA_002509405.1.gff', 'GCA_902758425.1/GCA_902758425.1.gff', 'GCA_002496945.1/GCA_002496945.1.gff', 'GCA_902756745.1/GCA_902756745.1.gff', 'GCA_902775675.1/GCA_902775675.1.gff', 'GCA_002495445.1/GCA_002495445.1.gff', 'GCA_002509505.1/GCA_002509505.1.gff', 'GCA_002506535.1/GCA_002506535.1.gff', 'GCA_002509015.1/GCA_002509015.1.gff', 'GCA_002505605.1/GCA_002505605.1.gff', 'GCA_002509645.1/GCA_002509645.1.gff', 'GCA_002504605.1/GCA_002504605.1.gff', 'GCA_002508505.1/GCA_002508505.1.gff', 'GCA_902792995.1/GCA_902792995.1.gff', 'GCA_002507025.1/GCA_002507025.1.gff', 'GCA_002501695.1/GCA_002501695.1.gff', 'GCA_002494705.1/GCA_002494705.1.gff', 'GCA_902758945.1/GCA_902758945.1.gff', 'GCA_002498505.1/GCA_002498505.1.gff', 'GCA_902786095.1/GCA_902786095.1.gff', 'GCA_002497365.1/GCA_002497365.1.gff', 'GCA_002499755.1/GCA_002499755.1.gff', 'GCA_002505925.1/GCA_002505925.1.gff', 'GCA_002508745.1/GCA_002508745.1.gff', 'GCA_002509655.1/GCA_002509655.1.gff', 'GCA_002505245.1/GCA_002505245.1.gff', 'GCA_002507065.1/GCA_002507065.1.gff', 'GCA_002508545.1/GCA_002508545.1.gff', 'GCA_002504505.1/GCA_002504505.1.gff', 'GCA_002508935.1/GCA_002508935.1.gff', 'GCA_002496565.1/GCA_002496565.1.gff', 'GCA_002498445.1/GCA_002498445.1.gff', 'GCA_902759335.1/GCA_902759335.1.gff', 'GCA_902799105.1/GCA_902799105.1.gff', 'GCA_002503885.1/GCA_002503885.1.gff', 'GCA_002498705.1/GCA_002498705.1.gff', 'GCA_902775725.1/GCA_902775725.1.gff', 'GCA_002499825.1/GCA_002499825.1.gff', 'GCA_002498315.1/GCA_002498315.1.gff', 'GCA_002498605.1/GCA_002498605.1.gff', 'GCA_902755425.1/GCA_902755425.1.gff', 'GCA_002499145.1/GCA_002499145.1.gff', 'GCA_002496465.1/GCA_002496465.1.gff', 'GCA_002499455.1/GCA_002499455.1.gff', 'GCA_002506335.1/GCA_002506335.1.gff', 'GCA_002508705.1/GCA_002508705.1.gff', 'GCA_002506915.1/GCA_002506915.1.gff', 'GCA_002504405.1/GCA_002504405.1.gff', 'GCA_002509045.1/GCA_002509045.1.gff', 'GCA_002506565.1/GCA_002506565.1.gff', 'GCA_002505345.1/GCA_002505345.1.gff', 'GCA_002508675.1/GCA_002508675.1.gff', 'GCA_002506415.1/GCA_002506415.1.gff', 'GCA_002509425.1/GCA_002509425.1.gff', 'GCA_002506105.1/GCA_002506105.1.gff', 'GCA_002505725.1/GCA_002505725.1.gff', 'GCA_902782945.1/GCA_902782945.1.gff', 'GCA_002495565.1/GCA_002495565.1.gff', 'GCA_002496345.1/GCA_002496345.1.gff', 'GCA_002495625.1/GCA_002495625.1.gff', 'GCA_002494805.1/GCA_002494805.1.gff', 'GCA_002499035.1/GCA_002499035.1.gff', 'GCA_902757365.1/GCA_902757365.1.gff', 'GCA_002495725.1/GCA_002495725.1.gff', 'GCA_002503585.1/GCA_002503585.1.gff', 'GCA_902775005.1/GCA_902775005.1.gff', 'GCA_002498365.1/GCA_002498365.1.gff', 'GCA_902777475.1/GCA_902777475.1.gff', 'GCA_002508845.1/GCA_002508845.1.gff', 'GCA_002508775.1/GCA_002508775.1.gff', 'GCA_002506455.1/GCA_002506455.1.gff', 'GCA_002505275.1/GCA_002505275.1.gff', 'GCA_002505765.1/GCA_002505765.1.gff', 'GCA_002508905.1/GCA_002508905.1.gff', 'GCA_002494845.1/GCA_002494845.1.gff', 'GCA_002496555.1/GCA_002496555.1.gff', 'GCA_902774605.1/GCA_902774605.1.gff', 'GCA_002498805.1/GCA_002498805.1.gff', 'GCA_002503785.1/GCA_002503785.1.gff', 'GCA_902798025.1/GCA_902798025.1.gff', 'GCA_002498225.1/GCA_002498225.1.gff', 'GCA_002498905.1/GCA_002498905.1.gff', 'GCA_902774705.1/GCA_902774705.1.gff', 'GCA_002498065.1/GCA_002498065.1.gff', 'GCA_902778875.1/GCA_902778875.1.gff', 'GCA_002508805.1/GCA_002508805.1.gff', 'GCA_002509335.1/GCA_002509335.1.gff', 'GCA_002509565.1/GCA_002509565.1.gff', 'GCA_002508475.1/GCA_002508475.1.gff', 'GCA_002508625.1/GCA_002508625.1.gff', 'GCA_002504525.1/GCA_002504525.1.gff', 'GCA_002495365.1/GCA_002495365.1.gff', 'GCA_002499905.1/GCA_002499905.1.gff', 'GCA_002502785.1/GCA_002502785.1.gff', 'GCA_002497555.1/GCA_002497555.1.gff', 'GCA_902784195.1/GCA_902784195.1.gff', 'GCA_002505025.1/GCA_002505025.1.gff', 'GCA_002505535.1/GCA_002505535.1.gff', 'GCA_002509325.1/GCA_002509325.1.gff', 'GCA_002508725.1/GCA_002508725.1.gff', 'GCA_002507205.1/GCA_002507205.1.gff', 'GCA_002504765.1/GCA_002504765.1.gff', 'GCA_002509575.1/GCA_002509575.1.gff', 'GCA_002508665.1/GCA_002508665.1.gff', 'GCA_902763115.1/GCA_902763115.1.gff', 'GCA_002505225.1/GCA_002505225.1.gff', 'GCA_002498765.1/GCA_002498765.1.gff', 'GCA_002495325.1/GCA_002495325.1.gff', 'GCA_902798335.1/GCA_902798335.1.gff', 'GCA_902757275.1/GCA_902757275.1.gff', 'GCA_002497105.1/GCA_002497105.1.gff', 'GCA_002498425.1/GCA_002498425.1.gff', 'GCA_002499025.1/GCA_002499025.1.gff', 'GCA_002495805.1/GCA_002495805.1.gff', 'GCA_002503085.1/GCA_002503085.1.gff', 'GCA_002498375.1/GCA_002498375.1.gff', 'GCA_902784295.1/GCA_902784295.1.gff', 'GCA_002497655.1/GCA_002497655.1.gff', 'GCA_002508425.1/GCA_002508425.1.gff', 'GCA_902762245.1/GCA_902762245.1.gff', 'GCA_902770085.1/GCA_902770085.1.gff', 'GCA_002508855.1/GCA_002508855.1.gff', 'GCA_002508765.1/GCA_002508765.1.gff', 'GCA_002495685.1/GCA_002495685.1.gff', 'GCA_002503135.1/GCA_002503135.1.gff', 'GCA_002502845.1/GCA_002502845.1.gff', 'GCA_902765105.1/GCA_902765105.1.gff', 'GCA_902785855.1/GCA_902785855.1.gff', 'GCA_002495585.1/GCA_002495585.1.gff', 'GCA_002502635.1/GCA_002502635.1.gff', 'GCA_002494495.1/GCA_002494495.1.gff', 'GCA_002505485.1/GCA_002505485.1.gff', 'GCA_902771225.1/GCA_902771225.1.gff', 'GCA_002502735.1/GCA_002502735.1.gff', 'GCA_002497995.1/GCA_002497995.1.gff', 'GCA_902802035.1/GCA_902802035.1.gff', 'GCA_002498385.1/GCA_002498385.1.gff', 'GCA_002502805.1/GCA_002502805.1.gff', 'GCA_002503565.1/GCA_002503565.1.gff', 'GCA_002501655.1/GCA_002501655.1.gff', 'GCA_902800615.1/GCA_902800615.1.gff', 'GCA_002498785.1/GCA_002498785.1.gff', 'GCA_902789165.1/GCA_902789165.1.gff', 'GCA_902765875.1/GCA_902765875.1.gff', 'GCA_002504585.1/GCA_002504585.1.gff', 'GCA_002509385.1/GCA_002509385.1.gff', 'GCA_902790605.1/GCA_902790605.1.gff', 'GCA_002506995.1/GCA_002506995.1.gff', 'GCA_002504485.1/GCA_002504485.1.gff', 'GCA_002494585.1/GCA_002494585.1.gff', 'GCA_002502465.1/GCA_002502465.1.gff', 'GCA_902800455.1/GCA_902800455.1.gff', 'GCA_902757195.1/GCA_902757195.1.gff', 'GCA_902798395.1/GCA_902798395.1.gff', 'GCA_902800515.1/GCA_902800515.1.gff', 'GCA_902785165.1/GCA_902785165.1.gff', 'GCA_002498485.1/GCA_002498485.1.gff', 'GCA_002494785.1/GCA_002494785.1.gff', 'GCA_002503125.1/GCA_002503125.1.gff', 'GCA_002509495.1/GCA_002509495.1.gff', 'GCA_902764455.1/GCA_902764455.1.gff', 'GCA_902802065.1/GCA_902802065.1.gff', 'GCA_902787415.1/GCA_902787415.1.gff', 'GCA_902768415.1/GCA_902768415.1.gff', 'GCA_002502855.1/GCA_002502855.1.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "223"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_paths = []\n",
    "for file in gff_files:\n",
    "    gff_paths.append(file[51:])\n",
    "print(gff_paths)\n",
    "len(gff_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1316e212",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sh import gunzip"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b63f759",
   "metadata": {},
   "source": [
    "# Create a database of metadata regarding the genes\n",
    "GFF File Format: \n",
    "- Fields must be tab-separated. Also, all but the final field in each feature line must contain a value; \"empty\" columns should be denoted with a '.'\n",
    "[0] seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.\n",
    "[1] source - name of the program that generated this feature, or the data source (database or project name)\n",
    "[2] feature - feature type name, e.g. Gene, Variation, Similarity\n",
    "[3] start - Start position* of the feature, with sequence numbering starting at 1.\n",
    "[4] end - End position* of the feature, with sequence numbering starting at 1.\n",
    "[5] score - A floating point value.\n",
    "[6] strand - defined as + (forward) or - (reverse).\n",
    "[7] frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..\n",
    "[8] attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e3288df2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>13849</td>\n",
       "      <td>14601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107694</td>\n",
       "      <td>108455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>1</td>\n",
       "      <td>53242</td>\n",
       "      <td>53448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107399</td>\n",
       "      <td>108148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002508885.1</td>\n",
       "      <td>1</td>\n",
       "      <td>44628</td>\n",
       "      <td>45401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>GCA_002498485.1</td>\n",
       "      <td>1</td>\n",
       "      <td>80363</td>\n",
       "      <td>81112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>181973</td>\n",
       "      <td>182722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>GCA_002503125.1</td>\n",
       "      <td>1</td>\n",
       "      <td>7083</td>\n",
       "      <td>7832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>1</td>\n",
       "      <td>12667</td>\n",
       "      <td>13419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>1</td>\n",
       "      <td>4551</td>\n",
       "      <td>5300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession index   start    stop\n",
       "0      GCA_902774685.1     1   13849   14601\n",
       "1      GCA_002497785.1     1  107694  108455\n",
       "2      GCA_002502925.1     1   53242   53448\n",
       "3      GCA_002506795.1     1  107399  108148\n",
       "4      GCA_002508885.1     1   44628   45401\n",
       "..                 ...   ...     ...     ...\n",
       "146    GCA_002498485.1     1   80363   81112\n",
       "147    GCA_002494785.1     1  181973  182722\n",
       "148    GCA_002503125.1     1    7083    7832\n",
       "149    GCA_902802065.1     1   12667   13419\n",
       "150    GCA_002502855.1     1    4551    5300\n",
       "\n",
       "[151 rows x 4 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "#get the start and stop of the given gene \n",
    "\n",
    "#create dataframe to store the output \n",
    "df = pd.DataFrame(columns=[\"assembly_accession\", \"index\", \"start\", \"stop\"])\n",
    "\n",
    "# iterate through the GFF files\n",
    "for path_name in gff_paths:\n",
    "    species = path_name[:15]  #keep species name from the path_name\n",
    "    gene_count = 0            #index to keep count of number of times gene shows up in one species\n",
    "    \n",
    "    with open(path_name, \"r\") as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                fields = line.strip().split('\\t')\n",
    "#                 if fields[2] == \"gene\":\n",
    "                if \"gene=mcrG\" in line:   #specify desired gene\n",
    "                    start = int(fields[3])\n",
    "                    stop = int(fields[4])\n",
    "                    gene_count += 1\n",
    "\n",
    "                    new_row = pd.DataFrame({\"assembly_accession\": [species],\n",
    "                                            \"index\": [gene_count],\n",
    "                                            \"start\": [start],\n",
    "                                            \"stop\": [stop]})\n",
    "\n",
    "                    df = pd.concat([df, new_row], ignore_index=True)\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "24924002",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>13849</td>\n",
       "      <td>14601</td>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107694</td>\n",
       "      <td>108455</td>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>1</td>\n",
       "      <td>53242</td>\n",
       "      <td>53448</td>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107399</td>\n",
       "      <td>108148</td>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002508885.1</td>\n",
       "      <td>1</td>\n",
       "      <td>44628</td>\n",
       "      <td>45401</td>\n",
       "      <td>GCA_002508885.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>GCA_002498485.1</td>\n",
       "      <td>1</td>\n",
       "      <td>80363</td>\n",
       "      <td>81112</td>\n",
       "      <td>GCA_002498485.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>181973</td>\n",
       "      <td>182722</td>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>GCA_002503125.1</td>\n",
       "      <td>1</td>\n",
       "      <td>7083</td>\n",
       "      <td>7832</td>\n",
       "      <td>GCA_002503125.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>1</td>\n",
       "      <td>12667</td>\n",
       "      <td>13419</td>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>1</td>\n",
       "      <td>4551</td>\n",
       "      <td>5300</td>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession index   start    stop        Accession  \\\n",
       "0      GCA_902774685.1     1   13849   14601  GCA_902774685.1   \n",
       "1      GCA_002497785.1     1  107694  108455  GCA_002497785.1   \n",
       "2      GCA_002502925.1     1   53242   53448  GCA_002502925.1   \n",
       "3      GCA_002506795.1     1  107399  108148  GCA_002506795.1   \n",
       "4      GCA_002508885.1     1   44628   45401  GCA_002508885.1   \n",
       "..                 ...   ...     ...     ...              ...   \n",
       "146    GCA_002498485.1     1   80363   81112  GCA_002498485.1   \n",
       "147    GCA_002494785.1     1  181973  182722  GCA_002494785.1   \n",
       "148    GCA_002503125.1     1    7083    7832  GCA_002503125.1   \n",
       "149    GCA_902802065.1     1   12667   13419  GCA_902802065.1   \n",
       "150    GCA_002502855.1     1    4551    5300  GCA_002502855.1   \n",
       "\n",
       "                         organism_name   Isolate  \n",
       "0        uncultured Methanobrevibacter  RUG11990  \n",
       "1                         Methanolinea    UBA144  \n",
       "2    Methanomassiliicoccaceae archaeon    UBA414  \n",
       "3                     Methanobacterium    UBA397  \n",
       "4                   Methanobrevibacter    UBA395  \n",
       "..                                 ...       ...  \n",
       "146                   Methanobacterium    UBA302  \n",
       "147       Methanobacteriaceae archaeon    UBA117  \n",
       "148                   Methanobacterium    UBA418  \n",
       "149      uncultured Methanobrevibacter  RUG14729  \n",
       "150                   Methanobacterium    UBA279  \n",
       "\n",
       "[151 rows x 7 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#insert columns with organism and strain name\n",
    "species = pd.read_csv('/Users/gracedai/Documents/MAGS.csv')\n",
    "species = species[['Accession', 'organism_name', 'Isolate']]\n",
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='Accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb8bf9a0",
   "metadata": {},
   "source": [
    "# Gather gene sequences and put into database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f040986e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#os.chdir(\"/Users/gracedai/Documents/asmg labs files/fna sequences\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "94e23059",
   "metadata": {},
   "outputs": [],
   "source": [
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gz\"):\n",
    "            #os.remove(os.path.join(root, file))\n",
    "            gunzip(os.path.join(root, file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3568c7a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/gracedai/Documents/MethanogeneDB/MAGS'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b0cc37d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to find fasta file in a folder and extract the gene sequence\n",
    "def get_sequence(folder_name, start, stop):\n",
    "    fasta_file = [file for file in os.listdir(folder_name) if file.endswith(\".fna\")]\n",
    "    fasta_path = os.path.join(folder_name, fasta_file[0])\n",
    "    with open(fasta_path, \"r\") as file:\n",
    "        lines = file.readlines()\n",
    "        sequence = \"\".join(line.strip() for line in lines[1:] if not line.startswith(\">\"))\n",
    "    return sequence[start-1:stop]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "2ab5d043",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACCAACATTTT\n"
     ]
    }
   ],
   "source": [
    "#test function\n",
    "# out = get_sequence(\"GCA_001563305.1\", 10, 20)\n",
    "# print(out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bea2e142",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>1</td>\n",
       "      <td>13849</td>\n",
       "      <td>14601</td>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>AATTTAAAATAAACTAAAATATTGATTGAAATTATAAAATTATAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107694</td>\n",
       "      <td>108455</td>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "      <td>ACAGGACCGCCCCGGTAAGATCAAACCCATGGTTCCCCGTTTCCGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>1</td>\n",
       "      <td>53242</td>\n",
       "      <td>53448</td>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "      <td>GTGACCATCACCGTAACCGACGGAACGGGGAAGATGCCGTCGTTAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107399</td>\n",
       "      <td>108148</td>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "      <td>CTAGGAATGCAGTTAGTATGATGGTTCTTTTGGCTGATTCAGGCGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002508885.1</td>\n",
       "      <td>1</td>\n",
       "      <td>44628</td>\n",
       "      <td>45401</td>\n",
       "      <td>GCA_002508885.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA395</td>\n",
       "      <td>TAATATTTAATTAATAATAAATTTTTAACATTAATAAATTATAATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>GCA_002498485.1</td>\n",
       "      <td>1</td>\n",
       "      <td>80363</td>\n",
       "      <td>81112</td>\n",
       "      <td>GCA_002498485.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA302</td>\n",
       "      <td>TTTGGTCATTATCTCATCTAATTTGTGGATGTTTAAAGCTACAGAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>181973</td>\n",
       "      <td>182722</td>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "      <td>CTTAAAACTAGTTTAATTGCTACAGGGGATTTCTCTAAACCTAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>GCA_002503125.1</td>\n",
       "      <td>1</td>\n",
       "      <td>7083</td>\n",
       "      <td>7832</td>\n",
       "      <td>GCA_002503125.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA418</td>\n",
       "      <td>AATAAGTGATATAAAAGGTTATGTTCTTAAAAATAAGGATATAATG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>1</td>\n",
       "      <td>12667</td>\n",
       "      <td>13419</td>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>GGTTTTAAGAGTTTCCTTGGAGTGGATAAAATAAACAAATTTTGGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>1</td>\n",
       "      <td>4551</td>\n",
       "      <td>5300</td>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "      <td>AGGGCTTTCTTTTCCTCTTCTTTAAGGGCTTTTTCTTCTTCTTCAG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession index   start    stop        Accession  \\\n",
       "0      GCA_902774685.1     1   13849   14601  GCA_902774685.1   \n",
       "1      GCA_002497785.1     1  107694  108455  GCA_002497785.1   \n",
       "2      GCA_002502925.1     1   53242   53448  GCA_002502925.1   \n",
       "3      GCA_002506795.1     1  107399  108148  GCA_002506795.1   \n",
       "4      GCA_002508885.1     1   44628   45401  GCA_002508885.1   \n",
       "..                 ...   ...     ...     ...              ...   \n",
       "146    GCA_002498485.1     1   80363   81112  GCA_002498485.1   \n",
       "147    GCA_002494785.1     1  181973  182722  GCA_002494785.1   \n",
       "148    GCA_002503125.1     1    7083    7832  GCA_002503125.1   \n",
       "149    GCA_902802065.1     1   12667   13419  GCA_902802065.1   \n",
       "150    GCA_002502855.1     1    4551    5300  GCA_002502855.1   \n",
       "\n",
       "                         organism_name   Isolate  \\\n",
       "0        uncultured Methanobrevibacter  RUG11990   \n",
       "1                         Methanolinea    UBA144   \n",
       "2    Methanomassiliicoccaceae archaeon    UBA414   \n",
       "3                     Methanobacterium    UBA397   \n",
       "4                   Methanobrevibacter    UBA395   \n",
       "..                                 ...       ...   \n",
       "146                   Methanobacterium    UBA302   \n",
       "147       Methanobacteriaceae archaeon    UBA117   \n",
       "148                   Methanobacterium    UBA418   \n",
       "149      uncultured Methanobrevibacter  RUG14729   \n",
       "150                   Methanobacterium    UBA279   \n",
       "\n",
       "                                              Sequence  \n",
       "0    AATTTAAAATAAACTAAAATATTGATTGAAATTATAAAATTATAAT...  \n",
       "1    ACAGGACCGCCCCGGTAAGATCAAACCCATGGTTCCCCGTTTCCGC...  \n",
       "2    GTGACCATCACCGTAACCGACGGAACGGGGAAGATGCCGTCGTTAC...  \n",
       "3    CTAGGAATGCAGTTAGTATGATGGTTCTTTTGGCTGATTCAGGCGT...  \n",
       "4    TAATATTTAATTAATAATAAATTTTTAACATTAATAAATTATAATT...  \n",
       "..                                                 ...  \n",
       "146  TTTGGTCATTATCTCATCTAATTTGTGGATGTTTAAAGCTACAGAC...  \n",
       "147  CTTAAAACTAGTTTAATTGCTACAGGGGATTTCTCTAAACCTAAAT...  \n",
       "148  AATAAGTGATATAAAAGGTTATGTTCTTAAAAATAAGGATATAATG...  \n",
       "149  GGTTTTAAGAGTTTCCTTGGAGTGGATAAAATAAACAAATTTTGGC...  \n",
       "150  AGGGCTTTCTTTTCCTCTTCTTTAAGGGCTTTTTCTTCTTCTTCAG...  \n",
       "\n",
       "[151 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get the fasta files and sequences, put them in a column and merge with previous df\n",
    "# store the gene sequences\n",
    "sequences = []\n",
    "\n",
    "for _, row in df.iterrows():\n",
    "    folder_name = row[\"assembly_accession\"]\n",
    "    #print(folder_name)\n",
    "    start, stop = row[\"start\"], row[\"stop\"]\n",
    "    sequence = get_sequence(folder_name, start, stop) #extract the gene sequence using function\n",
    "    sequences.append(sequence)\n",
    "\n",
    "# add column of sequences to existing dataframe\n",
    "df[\"Sequence\"] = sequences\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "13c61a33",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>index</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_002496325.1</td>\n",
       "      <td>1</td>\n",
       "      <td>29558</td>\n",
       "      <td>30307</td>\n",
       "      <td>GCA_002496325.1</td>\n",
       "      <td>Candidatus Methanoperedens</td>\n",
       "      <td>UBA453</td>\n",
       "      <td>CAAGGAGATCGAACTTGATGTGGGCTCTGTGATCCTGGCTACTGGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>1</td>\n",
       "      <td>181973</td>\n",
       "      <td>182722</td>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "      <td>CTTAAAACTAGTTTAATTGCTACAGGGGATTTCTCTAAACCTAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCA_002507025.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107329</td>\n",
       "      <td>108078</td>\n",
       "      <td>GCA_002507025.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA373</td>\n",
       "      <td>GCCCAGGACTTTCAGCCATGGGTTCCGGAGATATAACCACGCTATC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>GCA_002504485.1</td>\n",
       "      <td>1</td>\n",
       "      <td>107514</td>\n",
       "      <td>108263</td>\n",
       "      <td>GCA_002504485.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA426</td>\n",
       "      <td>GTTATAACTGCGATCATAATTGTATCACCAATAATTGGAGCTTCAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_002505925.1</td>\n",
       "      <td>1</td>\n",
       "      <td>7086</td>\n",
       "      <td>7835</td>\n",
       "      <td>GCA_002505925.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA351</td>\n",
       "      <td>CTATGTATTTGACTTTAAGTTCCTTGAAGTTCCAGTGAATTTTACA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>GCA_902763115.1</td>\n",
       "      <td>1</td>\n",
       "      <td>68491</td>\n",
       "      <td>69249</td>\n",
       "      <td>GCA_902763115.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10830</td>\n",
       "      <td>GCTGATTATGTACTGGCGAATCCTCCATTCAATCTTAAAAAATGGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>1</td>\n",
       "      <td>88088</td>\n",
       "      <td>88846</td>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>TTTTTTAGGTTGAAAAATGAAATTCCAAGTAATGTATGATAACGGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>1</td>\n",
       "      <td>50887</td>\n",
       "      <td>51636</td>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>TAGCTTTAGCTCTATTAATATTCTTAGTTCCAAAATCCAAGTTTAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GCA_902775725.1</td>\n",
       "      <td>1</td>\n",
       "      <td>78019</td>\n",
       "      <td>78771</td>\n",
       "      <td>GCA_902775725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12100</td>\n",
       "      <td>TTTCAAAGTCTAATGCTCCAAATGGACAAGCAGATGCACATAAACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>1</td>\n",
       "      <td>58375</td>\n",
       "      <td>59127</td>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>TTCTCATCCAGCCAGTCATCGGCATCGAGAAATGCTATGTATTCTC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession index   start    stop        Accession  \\\n",
       "24     GCA_002496325.1     1   29558   30307  GCA_002496325.1   \n",
       "147    GCA_002494785.1     1  181973  182722  GCA_002494785.1   \n",
       "40     GCA_002507025.1     1  107329  108078  GCA_002507025.1   \n",
       "142    GCA_002504485.1     1  107514  108263  GCA_002504485.1   \n",
       "43     GCA_002505925.1     1    7086    7835  GCA_002505925.1   \n",
       "..                 ...   ...     ...     ...              ...   \n",
       "114    GCA_902763115.1     1   68491   69249  GCA_902763115.1   \n",
       "115    GCA_902798335.1     1   88088   88846  GCA_902798335.1   \n",
       "116    GCA_902757275.1     1   50887   51636  GCA_902757275.1   \n",
       "55     GCA_902775725.1     1   78019   78771  GCA_902775725.1   \n",
       "75     GCA_902757365.1     1   58375   59127  GCA_902757365.1   \n",
       "\n",
       "                     organism_name    strain  \\\n",
       "24      Candidatus Methanoperedens    UBA453   \n",
       "147   Methanobacteriaceae archaeon    UBA117   \n",
       "40                Methanobacterium    UBA373   \n",
       "142               Methanobacterium    UBA426   \n",
       "43                Methanobacterium    UBA351   \n",
       "..                             ...       ...   \n",
       "114  uncultured Methanobrevibacter  RUG10830   \n",
       "115  uncultured Methanobrevibacter  RUG14349   \n",
       "116  uncultured Methanobrevibacter  RUG10244   \n",
       "55   uncultured Methanobrevibacter  RUG12100   \n",
       "75   uncultured Methanobrevibacter  RUG10252   \n",
       "\n",
       "                                              Sequence  \n",
       "24   CAAGGAGATCGAACTTGATGTGGGCTCTGTGATCCTGGCTACTGGC...  \n",
       "147  CTTAAAACTAGTTTAATTGCTACAGGGGATTTCTCTAAACCTAAAT...  \n",
       "40   GCCCAGGACTTTCAGCCATGGGTTCCGGAGATATAACCACGCTATC...  \n",
       "142  GTTATAACTGCGATCATAATTGTATCACCAATAATTGGAGCTTCAT...  \n",
       "43   CTATGTATTTGACTTTAAGTTCCTTGAAGTTCCAGTGAATTTTACA...  \n",
       "..                                                 ...  \n",
       "114  GCTGATTATGTACTGGCGAATCCTCCATTCAATCTTAAAAAATGGG...  \n",
       "115  TTTTTTAGGTTGAAAAATGAAATTCCAAGTAATGTATGATAACGGA...  \n",
       "116  TAGCTTTAGCTCTATTAATATTCTTAGTTCCAAAATCCAAGTTTAA...  \n",
       "55   TTTCAAAGTCTAATGCTCCAAATGGACAAGCAGATGCACATAAACC...  \n",
       "75   TTCTCATCCAGCCAGTCATCGGCATCGAGAAATGCTATGTATTCTC...  \n",
       "\n",
       "[151 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_name'] = df['organism_name'].astype(str)\n",
    "df = df.rename(columns={'Isolate': 'strain'})\n",
    "df_sorted = df.sort_values('organism_name', ascending=True)\n",
    "df_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "cefb5125",
   "metadata": {},
   "outputs": [],
   "source": [
    "# export locally to a csv\n",
    "df_sorted.to_csv('mcrG_prokka_seqs.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d74169f9",
   "metadata": {},
   "source": [
    "# Output dataframes to a formatted fasta file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "13586802",
   "metadata": {},
   "outputs": [],
   "source": [
    "#out_fasta should be \"name.fasta\" to ensure the output file is in fasta format\n",
    "def to_fasta(df, out_fasta):\n",
    "    df[\"organism_name\"] = df[\"organism_name\"].str.replace(\" \",\"-\")\n",
    "    df[\"strain\"] = df[\"strain\"].str.replace(\" \",\"-\")\n",
    "    with open(out_fasta, 'w') as file:\n",
    "        for _, row in df.iterrows():\n",
    "            header = f'>{row[\"assembly_accession\"]}|{row[\"organism_name\"]}|{row[\"strain\"]}|{row[\"index\"]}|{row[\"start\"]}|{row[\"stop\"]}'\n",
    "            sequence = row[\"Sequence\"]\n",
    "            file.write(f'{header}\\n{sequence}\\n')\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7c79cf0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export to local fasta file\n",
    "to_fasta(df_sorted, \"mcrG_prokka_seqs.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "334a7b05",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
