{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8270eb45",
   "metadata": {},
   "source": [
    "A Python script for searching through all gff files in a directory. The script searches through a given gff file, notes the count of a given gene in a pathway, and then outputs that number into the correct cell associated with that gene and pathway in the excel spreadsheet."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aa0ed9",
   "metadata": {},
   "source": [
    "How it works:\n",
    "A. Flip the csv vertical genes to be names of columns. \n",
    "B. Write all species ids to the given csv (should be names of rows). \n",
    "C. Create a list of all gene names to search for called to_search.txt\n",
    "\n",
    "1. Search through all files in a directory and find all gff files(even if they are in a sub-folder)\n",
    "- unzip the file in the same folder\n",
    "- write the path of the file to a list\n",
    "- trim path name to part that is necessary for operating \n",
    "\n",
    "2. for each file in the list, perform the following:\n",
    "- for each line in to_search.txt, find number of lines in which that word shows up at least once (counts of lines, not words!)\n",
    "- create a dataframe with the columns as gene names from to_search.txt \n",
    "- add row to dataframe for each species searched, that row has all the counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "a84427e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/gracedai/Downloads\n"
     ]
    }
   ],
   "source": [
    "#Check that we are in the correct working directory\n",
    "import os \n",
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d4ad666f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving notices: ...working... done\n",
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.3.1\n",
      "  latest version: 23.5.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.5.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - sh\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    certifi-2023.5.7           |  py310hca03da5_0         153 KB\n",
      "    openssl-1.1.1u             |       h1a28f6b_0         2.6 MB\n",
      "    sh-1.14.2                  |  py310hca03da5_0          67 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         2.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  sh                 pkgs/main/osx-arm64::sh-1.14.2-py310hca03da5_0 \n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    anaconda::ca-certificates-2022.4.26-h~ --> pkgs/main::ca-certificates-2023.05.30-hca03da5_0 \n",
      "  certifi            anaconda::certifi-2022.6.15-py310hca0~ --> pkgs/main::certifi-2023.5.7-py310hca03da5_0 \n",
      "  openssl                                 1.1.1t-h1a28f6b_0 --> 1.1.1u-h1a28f6b_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "certifi-2023.5.7     | 153 KB    |                                       |   0% \n",
      "openssl-1.1.1u       | 2.6 MB    |                                       |   0% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     |                                       |   0% \u001b[A\u001b[A\n",
      "certifi-2023.5.7     | 153 KB    | ##################################### | 100% \u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "sh-1.14.2            | 67 KB     | ########8                             |  24% \u001b[A\u001b[A\n",
      "openssl-1.1.1u       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: \\ WARNING conda.core.path_actions:verify(1093): Unable to create environments file. Path not writable.\n",
      "  environment location: /Users/gracedai/.conda/environments.txt\n",
      "\n",
      "done\n",
      "Executing transaction: / WARNING conda.core.envs_manager:register_env(49): Unable to register environment. Path not writable or missing.\n",
      "  environment location: /Users/gracedai/anaconda3\n",
      "  registry file: /Users/gracedai/.conda/environments.txt\n",
      "done\n"
     ]
    }
   ],
   "source": [
    "# Install a conda package in the current Jupyter kernel\n",
    "import sys\n",
    "!conda install --yes --prefix {sys.prefix} sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "df450b61",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/gracedai/Documents/MethanogeneDB/prokka_out\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0c2c402a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#unzips a file name, given the path, and deletes the zipped version\n",
    "from sh import gunzip\n",
    "#this line is for testing purposes\n",
    "#gunzip('GCF_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01379dfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#unzip whatever files needed\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gz\"):\n",
    "            #os.remove(os.path.join(root, file))\n",
    "            gunzip(os.path.join(root, file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f642bd3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902788045.1/GCA_902788045.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774685.1/GCA_902774685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497785.1/GCA_002497785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502925.1/GCA_002502925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506795.1/GCA_002506795.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508885.1/GCA_002508885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497395.1/GCA_002497395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501765.1/GCA_002501765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759285.1/GCA_902759285.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902764425.1/GCA_902764425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502965.1/GCA_002502965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800375.1/GCA_902800375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496085.1/GCA_002496085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800035.1/GCA_902800035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494885.1/GCA_002494885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762095.1/GCA_902762095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505895.1/GCA_002505895.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506085.1/GCA_002506085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496495.1/GCA_002496495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902768855.1/GCA_902768855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784455.1/GCA_902784455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784715.1/GCA_902784715.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902789505.1/GCA_902789505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502975.1/GCA_002502975.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503105.1/GCA_002503105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762085.1/GCA_902762085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506585.1/GCA_002506585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506095.1/GCA_002506095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495885.1/GCA_002495885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502405.1/GCA_002502405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503005.1/GCA_002503005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902788255.1/GCA_902788255.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759195.1/GCA_902759195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503455.1/GCA_002503455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902769175.1/GCA_902769175.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496395.1/GCA_002496395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503825.1/GCA_002503825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902801725.1/GCA_902801725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902770715.1/GCA_902770715.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506785.1/GCA_002506785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902763685.1/GCA_902763685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496295.1/GCA_002496295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504965.1/GCA_002504965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508925.1/GCA_002508925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496325.1/GCA_002496325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498825.1/GCA_002498825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496805.1/GCA_002496805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759475.1/GCA_902759475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902795935.1/GCA_902795935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496165.1/GCA_002496165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499445.1/GCA_002499445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497075.1/GCA_002497075.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902781855.1/GCA_902781855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506325.1/GCA_002506325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508825.1/GCA_002508825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506905.1/GCA_002506905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509545.1/GCA_002509545.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506845.1/GCA_002506845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509405.1/GCA_002509405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902758425.1/GCA_902758425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496945.1/GCA_002496945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902756745.1/GCA_902756745.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775675.1/GCA_902775675.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495445.1/GCA_002495445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509505.1/GCA_002509505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506535.1/GCA_002506535.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509015.1/GCA_002509015.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505605.1/GCA_002505605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509645.1/GCA_002509645.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504605.1/GCA_002504605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508505.1/GCA_002508505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902792995.1/GCA_902792995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507025.1/GCA_002507025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501695.1/GCA_002501695.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494705.1/GCA_002494705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902758945.1/GCA_902758945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498505.1/GCA_002498505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902786095.1/GCA_902786095.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497365.1/GCA_002497365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499755.1/GCA_002499755.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505925.1/GCA_002505925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508745.1/GCA_002508745.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509655.1/GCA_002509655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505245.1/GCA_002505245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507065.1/GCA_002507065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508545.1/GCA_002508545.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504505.1/GCA_002504505.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508935.1/GCA_002508935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496565.1/GCA_002496565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498445.1/GCA_002498445.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902759335.1/GCA_902759335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902799105.1/GCA_902799105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503885.1/GCA_002503885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498705.1/GCA_002498705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775725.1/GCA_902775725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499825.1/GCA_002499825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498315.1/GCA_002498315.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498605.1/GCA_002498605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902755425.1/GCA_902755425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499145.1/GCA_002499145.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496465.1/GCA_002496465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499455.1/GCA_002499455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506335.1/GCA_002506335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508705.1/GCA_002508705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506915.1/GCA_002506915.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504405.1/GCA_002504405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509045.1/GCA_002509045.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506565.1/GCA_002506565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505345.1/GCA_002505345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508675.1/GCA_002508675.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506415.1/GCA_002506415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509425.1/GCA_002509425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506105.1/GCA_002506105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505725.1/GCA_002505725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902782945.1/GCA_902782945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495565.1/GCA_002495565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496345.1/GCA_002496345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495625.1/GCA_002495625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494805.1/GCA_002494805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499035.1/GCA_002499035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757365.1/GCA_902757365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495725.1/GCA_002495725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503585.1/GCA_002503585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902775005.1/GCA_902775005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498365.1/GCA_002498365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902777475.1/GCA_902777475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508845.1/GCA_002508845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508775.1/GCA_002508775.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506455.1/GCA_002506455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505275.1/GCA_002505275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505765.1/GCA_002505765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508905.1/GCA_002508905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494845.1/GCA_002494845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002496555.1/GCA_002496555.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774605.1/GCA_902774605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498805.1/GCA_002498805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503785.1/GCA_002503785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798025.1/GCA_902798025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498225.1/GCA_002498225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498905.1/GCA_002498905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902774705.1/GCA_902774705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498065.1/GCA_002498065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902778875.1/GCA_902778875.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508805.1/GCA_002508805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509335.1/GCA_002509335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509565.1/GCA_002509565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508475.1/GCA_002508475.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508625.1/GCA_002508625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504525.1/GCA_002504525.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495365.1/GCA_002495365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499905.1/GCA_002499905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502785.1/GCA_002502785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497555.1/GCA_002497555.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784195.1/GCA_902784195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505025.1/GCA_002505025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505535.1/GCA_002505535.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509325.1/GCA_002509325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508725.1/GCA_002508725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002507205.1/GCA_002507205.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504765.1/GCA_002504765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509575.1/GCA_002509575.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508665.1/GCA_002508665.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902763115.1/GCA_902763115.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505225.1/GCA_002505225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498765.1/GCA_002498765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495325.1/GCA_002495325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798335.1/GCA_902798335.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757275.1/GCA_902757275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497105.1/GCA_002497105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498425.1/GCA_002498425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002499025.1/GCA_002499025.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495805.1/GCA_002495805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503085.1/GCA_002503085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498375.1/GCA_002498375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902784295.1/GCA_902784295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497655.1/GCA_002497655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508425.1/GCA_002508425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902762245.1/GCA_902762245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902770085.1/GCA_902770085.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508855.1/GCA_002508855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002508765.1/GCA_002508765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495685.1/GCA_002495685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503135.1/GCA_002503135.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502845.1/GCA_002502845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902765105.1/GCA_902765105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902785855.1/GCA_902785855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002495585.1/GCA_002495585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502635.1/GCA_002502635.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494495.1/GCA_002494495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002505485.1/GCA_002505485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902771225.1/GCA_902771225.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502735.1/GCA_002502735.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002497995.1/GCA_002497995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902802035.1/GCA_902802035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498385.1/GCA_002498385.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502805.1/GCA_002502805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503565.1/GCA_002503565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002501655.1/GCA_002501655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800615.1/GCA_902800615.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498785.1/GCA_002498785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902789165.1/GCA_902789165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902765875.1/GCA_902765875.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504585.1/GCA_002504585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509385.1/GCA_002509385.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902790605.1/GCA_902790605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002506995.1/GCA_002506995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002504485.1/GCA_002504485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494585.1/GCA_002494585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502465.1/GCA_002502465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800455.1/GCA_902800455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902757195.1/GCA_902757195.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902798395.1/GCA_902798395.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902800515.1/GCA_902800515.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902785165.1/GCA_902785165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002498485.1/GCA_002498485.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002494785.1/GCA_002494785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002503125.1/GCA_002503125.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002509495.1/GCA_002509495.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902764455.1/GCA_902764455.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902802065.1/GCA_902802065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902787415.1/GCA_902787415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_902768415.1/GCA_902768415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/prokka_out/GCA_002502855.1/GCA_002502855.1.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "223"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# list to store paths to gff files\n",
    "gff_files = []\n",
    "# iterate through current working directory and for each file \n",
    "# unzip the file and append the path to a list \n",
    "#run this twice ?\n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gff.gz\"):\n",
    "            gunzip(os.path.join(root, file))\n",
    "        if file.endswith(\".gff\"):\n",
    "            gff_files.append(os.path.join(root, file))\n",
    "print(gff_files)\n",
    "len(gff_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a4818c09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCA_902788045.1/GCA_902788045.1.gff', 'GCA_902774685.1/GCA_902774685.1.gff', 'GCA_002497785.1/GCA_002497785.1.gff', 'GCA_002502925.1/GCA_002502925.1.gff', 'GCA_002506795.1/GCA_002506795.1.gff', 'GCA_002508885.1/GCA_002508885.1.gff', 'GCA_002497395.1/GCA_002497395.1.gff', 'GCA_002501765.1/GCA_002501765.1.gff', 'GCA_902759285.1/GCA_902759285.1.gff', 'GCA_902764425.1/GCA_902764425.1.gff', 'GCA_002502965.1/GCA_002502965.1.gff', 'GCA_902800375.1/GCA_902800375.1.gff', 'GCA_002496085.1/GCA_002496085.1.gff', 'GCA_902800035.1/GCA_902800035.1.gff', 'GCA_002494885.1/GCA_002494885.1.gff', 'GCA_902762095.1/GCA_902762095.1.gff', 'GCA_002505895.1/GCA_002505895.1.gff', 'GCA_002506085.1/GCA_002506085.1.gff', 'GCA_002496495.1/GCA_002496495.1.gff', 'GCA_902768855.1/GCA_902768855.1.gff', 'GCA_902784455.1/GCA_902784455.1.gff', 'GCA_902784715.1/GCA_902784715.1.gff', 'GCA_902789505.1/GCA_902789505.1.gff', 'GCA_002502975.1/GCA_002502975.1.gff', 'GCA_002503105.1/GCA_002503105.1.gff', 'GCA_902762085.1/GCA_902762085.1.gff', 'GCA_002506585.1/GCA_002506585.1.gff', 'GCA_002506095.1/GCA_002506095.1.gff', 'GCA_002495885.1/GCA_002495885.1.gff', 'GCA_002502405.1/GCA_002502405.1.gff', 'GCA_002503005.1/GCA_002503005.1.gff', 'GCA_902788255.1/GCA_902788255.1.gff', 'GCA_902759195.1/GCA_902759195.1.gff', 'GCA_002503455.1/GCA_002503455.1.gff', 'GCA_902769175.1/GCA_902769175.1.gff', 'GCA_002496395.1/GCA_002496395.1.gff', 'GCA_002503825.1/GCA_002503825.1.gff', 'GCA_902801725.1/GCA_902801725.1.gff', 'GCA_902770715.1/GCA_902770715.1.gff', 'GCA_002506785.1/GCA_002506785.1.gff', 'GCA_902763685.1/GCA_902763685.1.gff', 'GCA_002496295.1/GCA_002496295.1.gff', 'GCA_002504965.1/GCA_002504965.1.gff', 'GCA_002508925.1/GCA_002508925.1.gff', 'GCA_002496325.1/GCA_002496325.1.gff', 'GCA_002498825.1/GCA_002498825.1.gff', 'GCA_002496805.1/GCA_002496805.1.gff', 'GCA_902759475.1/GCA_902759475.1.gff', 'GCA_902795935.1/GCA_902795935.1.gff', 'GCA_002496165.1/GCA_002496165.1.gff', 'GCA_002499445.1/GCA_002499445.1.gff', 'GCA_002497075.1/GCA_002497075.1.gff', 'GCA_902781855.1/GCA_902781855.1.gff', 'GCA_002506325.1/GCA_002506325.1.gff', 'GCA_002508825.1/GCA_002508825.1.gff', 'GCA_002506905.1/GCA_002506905.1.gff', 'GCA_002509545.1/GCA_002509545.1.gff', 'GCA_002506845.1/GCA_002506845.1.gff', 'GCA_002509405.1/GCA_002509405.1.gff', 'GCA_902758425.1/GCA_902758425.1.gff', 'GCA_002496945.1/GCA_002496945.1.gff', 'GCA_902756745.1/GCA_902756745.1.gff', 'GCA_902775675.1/GCA_902775675.1.gff', 'GCA_002495445.1/GCA_002495445.1.gff', 'GCA_002509505.1/GCA_002509505.1.gff', 'GCA_002506535.1/GCA_002506535.1.gff', 'GCA_002509015.1/GCA_002509015.1.gff', 'GCA_002505605.1/GCA_002505605.1.gff', 'GCA_002509645.1/GCA_002509645.1.gff', 'GCA_002504605.1/GCA_002504605.1.gff', 'GCA_002508505.1/GCA_002508505.1.gff', 'GCA_902792995.1/GCA_902792995.1.gff', 'GCA_002507025.1/GCA_002507025.1.gff', 'GCA_002501695.1/GCA_002501695.1.gff', 'GCA_002494705.1/GCA_002494705.1.gff', 'GCA_902758945.1/GCA_902758945.1.gff', 'GCA_002498505.1/GCA_002498505.1.gff', 'GCA_902786095.1/GCA_902786095.1.gff', 'GCA_002497365.1/GCA_002497365.1.gff', 'GCA_002499755.1/GCA_002499755.1.gff', 'GCA_002505925.1/GCA_002505925.1.gff', 'GCA_002508745.1/GCA_002508745.1.gff', 'GCA_002509655.1/GCA_002509655.1.gff', 'GCA_002505245.1/GCA_002505245.1.gff', 'GCA_002507065.1/GCA_002507065.1.gff', 'GCA_002508545.1/GCA_002508545.1.gff', 'GCA_002504505.1/GCA_002504505.1.gff', 'GCA_002508935.1/GCA_002508935.1.gff', 'GCA_002496565.1/GCA_002496565.1.gff', 'GCA_002498445.1/GCA_002498445.1.gff', 'GCA_902759335.1/GCA_902759335.1.gff', 'GCA_902799105.1/GCA_902799105.1.gff', 'GCA_002503885.1/GCA_002503885.1.gff', 'GCA_002498705.1/GCA_002498705.1.gff', 'GCA_902775725.1/GCA_902775725.1.gff', 'GCA_002499825.1/GCA_002499825.1.gff', 'GCA_002498315.1/GCA_002498315.1.gff', 'GCA_002498605.1/GCA_002498605.1.gff', 'GCA_902755425.1/GCA_902755425.1.gff', 'GCA_002499145.1/GCA_002499145.1.gff', 'GCA_002496465.1/GCA_002496465.1.gff', 'GCA_002499455.1/GCA_002499455.1.gff', 'GCA_002506335.1/GCA_002506335.1.gff', 'GCA_002508705.1/GCA_002508705.1.gff', 'GCA_002506915.1/GCA_002506915.1.gff', 'GCA_002504405.1/GCA_002504405.1.gff', 'GCA_002509045.1/GCA_002509045.1.gff', 'GCA_002506565.1/GCA_002506565.1.gff', 'GCA_002505345.1/GCA_002505345.1.gff', 'GCA_002508675.1/GCA_002508675.1.gff', 'GCA_002506415.1/GCA_002506415.1.gff', 'GCA_002509425.1/GCA_002509425.1.gff', 'GCA_002506105.1/GCA_002506105.1.gff', 'GCA_002505725.1/GCA_002505725.1.gff', 'GCA_902782945.1/GCA_902782945.1.gff', 'GCA_002495565.1/GCA_002495565.1.gff', 'GCA_002496345.1/GCA_002496345.1.gff', 'GCA_002495625.1/GCA_002495625.1.gff', 'GCA_002494805.1/GCA_002494805.1.gff', 'GCA_002499035.1/GCA_002499035.1.gff', 'GCA_902757365.1/GCA_902757365.1.gff', 'GCA_002495725.1/GCA_002495725.1.gff', 'GCA_002503585.1/GCA_002503585.1.gff', 'GCA_902775005.1/GCA_902775005.1.gff', 'GCA_002498365.1/GCA_002498365.1.gff', 'GCA_902777475.1/GCA_902777475.1.gff', 'GCA_002508845.1/GCA_002508845.1.gff', 'GCA_002508775.1/GCA_002508775.1.gff', 'GCA_002506455.1/GCA_002506455.1.gff', 'GCA_002505275.1/GCA_002505275.1.gff', 'GCA_002505765.1/GCA_002505765.1.gff', 'GCA_002508905.1/GCA_002508905.1.gff', 'GCA_002494845.1/GCA_002494845.1.gff', 'GCA_002496555.1/GCA_002496555.1.gff', 'GCA_902774605.1/GCA_902774605.1.gff', 'GCA_002498805.1/GCA_002498805.1.gff', 'GCA_002503785.1/GCA_002503785.1.gff', 'GCA_902798025.1/GCA_902798025.1.gff', 'GCA_002498225.1/GCA_002498225.1.gff', 'GCA_002498905.1/GCA_002498905.1.gff', 'GCA_902774705.1/GCA_902774705.1.gff', 'GCA_002498065.1/GCA_002498065.1.gff', 'GCA_902778875.1/GCA_902778875.1.gff', 'GCA_002508805.1/GCA_002508805.1.gff', 'GCA_002509335.1/GCA_002509335.1.gff', 'GCA_002509565.1/GCA_002509565.1.gff', 'GCA_002508475.1/GCA_002508475.1.gff', 'GCA_002508625.1/GCA_002508625.1.gff', 'GCA_002504525.1/GCA_002504525.1.gff', 'GCA_002495365.1/GCA_002495365.1.gff', 'GCA_002499905.1/GCA_002499905.1.gff', 'GCA_002502785.1/GCA_002502785.1.gff', 'GCA_002497555.1/GCA_002497555.1.gff', 'GCA_902784195.1/GCA_902784195.1.gff', 'GCA_002505025.1/GCA_002505025.1.gff', 'GCA_002505535.1/GCA_002505535.1.gff', 'GCA_002509325.1/GCA_002509325.1.gff', 'GCA_002508725.1/GCA_002508725.1.gff', 'GCA_002507205.1/GCA_002507205.1.gff', 'GCA_002504765.1/GCA_002504765.1.gff', 'GCA_002509575.1/GCA_002509575.1.gff', 'GCA_002508665.1/GCA_002508665.1.gff', 'GCA_902763115.1/GCA_902763115.1.gff', 'GCA_002505225.1/GCA_002505225.1.gff', 'GCA_002498765.1/GCA_002498765.1.gff', 'GCA_002495325.1/GCA_002495325.1.gff', 'GCA_902798335.1/GCA_902798335.1.gff', 'GCA_902757275.1/GCA_902757275.1.gff', 'GCA_002497105.1/GCA_002497105.1.gff', 'GCA_002498425.1/GCA_002498425.1.gff', 'GCA_002499025.1/GCA_002499025.1.gff', 'GCA_002495805.1/GCA_002495805.1.gff', 'GCA_002503085.1/GCA_002503085.1.gff', 'GCA_002498375.1/GCA_002498375.1.gff', 'GCA_902784295.1/GCA_902784295.1.gff', 'GCA_002497655.1/GCA_002497655.1.gff', 'GCA_002508425.1/GCA_002508425.1.gff', 'GCA_902762245.1/GCA_902762245.1.gff', 'GCA_902770085.1/GCA_902770085.1.gff', 'GCA_002508855.1/GCA_002508855.1.gff', 'GCA_002508765.1/GCA_002508765.1.gff', 'GCA_002495685.1/GCA_002495685.1.gff', 'GCA_002503135.1/GCA_002503135.1.gff', 'GCA_002502845.1/GCA_002502845.1.gff', 'GCA_902765105.1/GCA_902765105.1.gff', 'GCA_902785855.1/GCA_902785855.1.gff', 'GCA_002495585.1/GCA_002495585.1.gff', 'GCA_002502635.1/GCA_002502635.1.gff', 'GCA_002494495.1/GCA_002494495.1.gff', 'GCA_002505485.1/GCA_002505485.1.gff', 'GCA_902771225.1/GCA_902771225.1.gff', 'GCA_002502735.1/GCA_002502735.1.gff', 'GCA_002497995.1/GCA_002497995.1.gff', 'GCA_902802035.1/GCA_902802035.1.gff', 'GCA_002498385.1/GCA_002498385.1.gff', 'GCA_002502805.1/GCA_002502805.1.gff', 'GCA_002503565.1/GCA_002503565.1.gff', 'GCA_002501655.1/GCA_002501655.1.gff', 'GCA_902800615.1/GCA_902800615.1.gff', 'GCA_002498785.1/GCA_002498785.1.gff', 'GCA_902789165.1/GCA_902789165.1.gff', 'GCA_902765875.1/GCA_902765875.1.gff', 'GCA_002504585.1/GCA_002504585.1.gff', 'GCA_002509385.1/GCA_002509385.1.gff', 'GCA_902790605.1/GCA_902790605.1.gff', 'GCA_002506995.1/GCA_002506995.1.gff', 'GCA_002504485.1/GCA_002504485.1.gff', 'GCA_002494585.1/GCA_002494585.1.gff', 'GCA_002502465.1/GCA_002502465.1.gff', 'GCA_902800455.1/GCA_902800455.1.gff', 'GCA_902757195.1/GCA_902757195.1.gff', 'GCA_902798395.1/GCA_902798395.1.gff', 'GCA_902800515.1/GCA_902800515.1.gff', 'GCA_902785165.1/GCA_902785165.1.gff', 'GCA_002498485.1/GCA_002498485.1.gff', 'GCA_002494785.1/GCA_002494785.1.gff', 'GCA_002503125.1/GCA_002503125.1.gff', 'GCA_002509495.1/GCA_002509495.1.gff', 'GCA_902764455.1/GCA_902764455.1.gff', 'GCA_902802065.1/GCA_902802065.1.gff', 'GCA_902787415.1/GCA_902787415.1.gff', 'GCA_902768415.1/GCA_902768415.1.gff', 'GCA_002502855.1/GCA_002502855.1.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "223"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_paths = []\n",
    "for file in gff_files:\n",
    "    gff_paths.append(file[51:])\n",
    "print(gff_paths)\n",
    "len(gff_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1316e212",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdC', 'fmdC', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdG', 'fwdH', 'ftr', 'mch', 'mtd', 'hmd', 'mer;', 'mtrA', 'mtrB', 'mtrC', 'mtrD', 'mtrE', 'mtrF', 'mtrG', 'mtrH', 'mcrA', 'mcrB', 'mcrG', 'hdrA1', 'hdrB1', 'hdrC1', 'hdrA2', 'hdrB2', 'hdrC2', 'hdrD', 'hdrE', 'mvhA', 'mvhG', 'mvhD', 'fdhA', 'fdhB']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "#read the text file of gene names into a list\n",
    "gene_file = open(\"/Users/gracedai/Documents/asmg labs files/optimize/to_search_co2.txt\", \"r\")\n",
    "data = gene_file.read()\n",
    "to_search = data.split(\"\\n\")\n",
    "print(to_search)\n",
    "gene_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "84a557dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(to_search)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b63f759",
   "metadata": {},
   "source": [
    "GFF File Format: \n",
    "- Fields must be tab-separated. Also, all but the final field in each feature line must contain a value; \"empty\" columns should be denoted with a '.'\n",
    "[0] seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.\n",
    "[1] source - name of the program that generated this feature, or the data source (database or project name)\n",
    "[2] feature - feature type name, e.g. Gene, Variation, Similarity\n",
    "[3] start - Start position* of the feature, with sequence numbering starting at 1.\n",
    "[4] end - End position* of the feature, with sequence numbering starting at 1.\n",
    "[5] score - A floating point value.\n",
    "[6] strand - defined as + (forward) or - (reverse).\n",
    "[7] frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..\n",
    "[8] attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e3288df2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create an empty dictionary to store the gene counts for each species\n",
    "gene_counts = {}\n",
    "\n",
    "#for gene pathways: iterate through all species gff files\n",
    "# for path_name in gff_paths:\n",
    "#     species = path_name[:15]  #keep species name from the path_name\n",
    "#     gene_counts[species] = {}\n",
    "    \n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 for gene in to_search:\n",
    "#                     if \"gene=\" + gene in line:\n",
    "#                         if gene not in gene_counts[species]:\n",
    "#                             gene_counts[species][gene] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species][gene] += 1\n",
    "    #read the GFF file\n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 fields = line.strip().split('\\t')\n",
    "#                 if (fields[2] == 'gene'):\n",
    "#                     attributes = fields[8] #field number 8 contains attributes such as gene name\n",
    "#                     for gene in to_search:\n",
    "#                         if gene in attributes:\n",
    "#                             if gene not in gene_counts[species]:\n",
    "#                                 gene_counts[species][gene] = 1\n",
    "#                             else:\n",
    "#                                 gene_counts[species][gene] += 1\n",
    "\n",
    "# for 16s and tRNA\n",
    "# for path_name in gff_paths:\n",
    "#     species = path_name[:15]  #keep species name from the path_name\n",
    "#     gene_counts[species] = {}\n",
    "\n",
    "#     #read the GFF file\n",
    "#     with open(path_name, 'r') as file:\n",
    "#         for line in file:\n",
    "#             if not line.startswith('#'):\n",
    "#                 fields = line.strip().split('\\t')\n",
    "#                 attributes = fields[8] #field number 8 contains attributes such as gene name\n",
    "#                 if fields[2] == \"gene\":\n",
    "#                     if 'tRNA' in attributes:\n",
    "#                         if 'tRNA' not in gene_counts[species]:\n",
    "#                             gene_counts[species]['tRNA'] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species]['tRNA'] += 1\n",
    "#                 if fields[2] == \"rRNA\":\n",
    "#                     if '16S' in attributes:\n",
    "#                         if '16S' not in gene_counts[species]:\n",
    "#                             gene_counts[species]['16S'] = 1\n",
    "#                         else:\n",
    "#                             gene_counts[species]['16S'] += 1\n",
    "\n",
    "for path_name in gff_paths:\n",
    "    species = path_name[:15]  #keep species name from the path_name\n",
    "    gene_counts[species] = {}\n",
    "\n",
    "    #read the GFF file\n",
    "    with open(path_name, 'r') as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                if \"rRNA\" and \"16S\" in line:\n",
    "                    if '16S' not in gene_counts[species]:\n",
    "                        gene_counts[species]['16S'] = 1\n",
    "                    else:\n",
    "                        gene_counts[species]['16S'] += 1\n",
    "                if \"tRNA\" in line: \n",
    "                    fields = line.split()\n",
    "                    #print(fields)\n",
    "                    if (fields[2] == 'tRNA'):\n",
    "                        if 'tRNA' not in gene_counts[species]:\n",
    "                            gene_counts[species]['tRNA'] = 1\n",
    "                        else:\n",
    "                            gene_counts[species]['tRNA'] += 1\n",
    "\n",
    "\n",
    "# Create a dataframe from the gene counts\n",
    "df = pd.DataFrame.from_dict(gene_counts, orient='index')\n",
    "df.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "d146647d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tRNA</th>\n",
       "      <th>16S</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCA_902788045.1</th>\n",
       "      <td>32</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902774685.1</th>\n",
       "      <td>28</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002497785.1</th>\n",
       "      <td>37</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002502925.1</th>\n",
       "      <td>41</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002506795.1</th>\n",
       "      <td>38</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902764455.1</th>\n",
       "      <td>35</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902802065.1</th>\n",
       "      <td>35</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902787415.1</th>\n",
       "      <td>35</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_902768415.1</th>\n",
       "      <td>27</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_002502855.1</th>\n",
       "      <td>37</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    tRNA  16S\n",
       "assembly_accession           \n",
       "GCA_902788045.1       32  0.0\n",
       "GCA_902774685.1       28  0.0\n",
       "GCA_002497785.1       37  1.0\n",
       "GCA_002502925.1       41  2.0\n",
       "GCA_002506795.1       38  0.0\n",
       "...                  ...  ...\n",
       "GCA_902764455.1       35  0.0\n",
       "GCA_902802065.1       35  2.0\n",
       "GCA_902787415.1       35  0.0\n",
       "GCA_902768415.1       27  0.0\n",
       "GCA_002502855.1       37  0.0\n",
       "\n",
       "[223 rows x 2 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename_axis('assembly_accession')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8df86a25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape[0] #missing 5 rows (species) \n",
    "df.shape[1] #missing 5 columns (genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4132f9d3",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[43mto_search\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m      2\u001b[0m df\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/frame.py:3813\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3811\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_iterator(key):\n\u001b[1;32m   3812\u001b[0m         key \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(key)\n\u001b[0;32m-> 3813\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_indexer_strict\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcolumns\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m[\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m   3815\u001b[0m \u001b[38;5;66;03m# take() does not accept boolean indexers\u001b[39;00m\n\u001b[1;32m   3816\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(indexer, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdtype\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mbool\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6070\u001b[0m, in \u001b[0;36mIndex._get_indexer_strict\u001b[0;34m(self, key, axis_name)\u001b[0m\n\u001b[1;32m   6067\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   6068\u001b[0m     keyarr, indexer, new_indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex_non_unique(keyarr)\n\u001b[0;32m-> 6070\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_raise_if_missing\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkeyarr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis_name\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   6072\u001b[0m keyarr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtake(indexer)\n\u001b[1;32m   6073\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(key, Index):\n\u001b[1;32m   6074\u001b[0m     \u001b[38;5;66;03m# GH 42790 - Preserve name from an Index\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/pandas/core/indexes/base.py:6133\u001b[0m, in \u001b[0;36mIndex._raise_if_missing\u001b[0;34m(self, key, indexer, axis_name)\u001b[0m\n\u001b[1;32m   6130\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNone of [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mkey\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m] are in the [\u001b[39m\u001b[38;5;132;01m{\u001b[39;00maxis_name\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m]\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   6132\u001b[0m not_found \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(ensure_index(key)[missing_mask\u001b[38;5;241m.\u001b[39mnonzero()[\u001b[38;5;241m0\u001b[39m]]\u001b[38;5;241m.\u001b[39munique())\n\u001b[0;32m-> 6133\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mnot_found\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m not in index\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['fwdA', 'fmdA', 'fwdB', 'fmdB', 'fwdD', 'fmdD', 'fwdE', 'fmdE', 'fwdF', 'fmdF', 'fwdH'] not in index\""
     ]
    }
   ],
   "source": [
    "df = df[to_search]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "ec15cc8d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mcrA</th>\n",
       "      <th>mcrG</th>\n",
       "      <th>mcrB</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>mtbC</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>mtbA</th>\n",
       "      <th>...</th>\n",
       "      <th>mtmB</th>\n",
       "      <th>mtbB</th>\n",
       "      <th>mttC</th>\n",
       "      <th>mttB</th>\n",
       "      <th>hdrA1</th>\n",
       "      <th>mtmC</th>\n",
       "      <th>hdrB1</th>\n",
       "      <th>hdrC1</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_001458655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000008645.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000145295.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008033705.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002813655.1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001477655.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_028743435.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000970325.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001027005.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCA_000306725.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    mvhA  mcrA  mcrG  mcrB  mvhG  hdrA2  mtbC  hdrE  hdrD  \\\n",
       "assembly_accession                                                          \n",
       "GCF_001458655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000008645.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000145295.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_008033705.1      1.0   2.0   2.0   2.0   1.0    0.0   0.0   0.0   0.0   \n",
       "GCF_002813655.1      1.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "...                  ...   ...   ...   ...   ...    ...   ...   ...   ...   \n",
       "GCF_001477655.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_028743435.1      0.0   2.0   2.0   2.0   0.0    0.0   0.0   0.0   0.0   \n",
       "GCF_000970325.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   0.0   \n",
       "GCF_001027005.1      0.0   1.0   1.0   1.0   0.0    1.0   0.0   0.0   1.0   \n",
       "GCA_000306725.1      0.0   0.0   0.0   0.0   0.0    0.0   0.0   0.0   0.0   \n",
       "\n",
       "                    mtbA  ...  mtmB  mtbB  mttC  mttB  hdrA1  mtmC  hdrB1  \\\n",
       "assembly_accession        ...                                               \n",
       "GCF_001458655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000008645.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000145295.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_008033705.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_002813655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "...                  ...  ...   ...   ...   ...   ...    ...   ...    ...   \n",
       "GCF_001477655.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_028743435.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_000970325.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCF_001027005.1      0.0  ...   0.0   0.0   0.0   0.0    0.0   0.0    0.0   \n",
       "GCA_000306725.1      1.0  ...   4.0   3.0   1.0   2.0    0.0   0.0    0.0   \n",
       "\n",
       "                    hdrC1  hdrB2  hdrC2  \n",
       "assembly_accession                       \n",
       "GCF_001458655.1       0.0    0.0    0.0  \n",
       "GCF_000008645.1       0.0    0.0    0.0  \n",
       "GCF_000145295.1       0.0    0.0    0.0  \n",
       "GCF_008033705.1       0.0    0.0    0.0  \n",
       "GCF_002813655.1       0.0    0.0    0.0  \n",
       "...                   ...    ...    ...  \n",
       "GCF_001477655.1       0.0    0.0    0.0  \n",
       "GCF_028743435.1       0.0    0.0    0.0  \n",
       "GCF_000970325.1       0.0    0.0    0.0  \n",
       "GCF_001027005.1       0.0    0.0    0.0  \n",
       "GCA_000306725.1       0.0    0.0    0.0  \n",
       "\n",
       "[199 rows x 23 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add columns that are missing, fill in with 0\n",
    "missing = ['mtmC','hdrB1', 'hdrC1', 'hdrB2', 'hdrC2']\n",
    "for i in missing: \n",
    "    df[i] = 0.0\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ef5985d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002494785.1</td>\n",
       "      <td>Methanobacteriaceae archaeon</td>\n",
       "      <td>UBA117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002495685.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Accession                  organism_name   Isolate\n",
       "0  GCA_902788255.1  uncultured Methanobrevibacter  RUG13354\n",
       "1  GCA_002494785.1   Methanobacteriaceae archaeon    UBA117\n",
       "2  GCA_902784195.1  uncultured Methanobrevibacter  RUG12946\n",
       "3  GCA_002495685.1             Methanobrevibacter    UBA412\n",
       "4  GCA_902771225.1  uncultured Methanobrevibacter  RUG11657"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#match species to the assembly acession numbers \n",
    "#first import csv with species and their accession numbers \n",
    "species = pd.read_csv('/Users/gracedai/Documents/MAGS.csv')\n",
    "species = species[['Accession', 'organism_name', 'Isolate']]\n",
    "species.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "4c6d603f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tRNA</th>\n",
       "      <th>16S</th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37</td>\n",
       "      <td>1.0</td>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>38</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902764455.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>35</td>\n",
       "      <td>2.0</td>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>27</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902768415.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     tRNA  16S        Accession                      organism_name   Isolate\n",
       "0      32  0.0  GCA_902788045.1      uncultured Methanobrevibacter  RUG13330\n",
       "1      28  0.0  GCA_902774685.1      uncultured Methanobrevibacter  RUG11990\n",
       "2      37  1.0  GCA_002497785.1                       Methanolinea    UBA144\n",
       "3      41  2.0  GCA_002502925.1  Methanomassiliicoccaceae archaeon    UBA414\n",
       "4      38  0.0  GCA_002506795.1                   Methanobacterium    UBA397\n",
       "..    ...  ...              ...                                ...       ...\n",
       "218    35  0.0  GCA_902764455.1      uncultured Methanobrevibacter  RUG10962\n",
       "219    35  2.0  GCA_902802065.1      uncultured Methanobrevibacter  RUG14729\n",
       "220    35  0.0  GCA_902787415.1              methanogenic archaeon  RUG13270\n",
       "221    27  0.0  GCA_902768415.1      uncultured Methanobrevibacter  RUG11363\n",
       "222    37  0.0  GCA_002502855.1                   Methanobacterium    UBA279\n",
       "\n",
       "[223 rows x 5 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='Accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "950cd6a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('Mags1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "dc9fa0b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#resort column order\n",
    "cols = ['Accession', 'organism_name', 'Isolate', \n",
    "    '16S', 'tRNA']\n",
    "df = df[cols]\n",
    "\n",
    "#add row with kegg numbers\n",
    "# kegg_row = [0,0,0,\n",
    "#     'K14082',\n",
    "#     'K16176',\n",
    "#     'K16177',\n",
    "#     'K16178',\n",
    "#     'K16179',\n",
    "#     'K14083',\n",
    "#     'K14084',\n",
    "#     'K00399',\n",
    "#     'K00401',\n",
    "#     'K00402',\n",
    "#     'K22480',\n",
    "#     'K22481',\n",
    "#     'K22482',\n",
    "#     'K03388',\n",
    "#     'K03389',\n",
    "#     'K03390',\n",
    "#     'K08264',\n",
    "#     'K08265',\n",
    "#     'K14126',\n",
    "#     'K14128',\n",
    "#     'K14127',\n",
    "#     'K22516',\n",
    "#     'K00125']\n",
    "\n",
    "# kegg_row_df = pd.DataFrame([kegg_row], columns=df.columns)\n",
    "#add row with gene description\n",
    "# descrip = [0, 0, 0, '[methyl-Co(III) methylamine-specific corrinoid protein]:coenzyme M methyltransferase [EC:2.1.1.247]',\n",
    "#     'methylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.248]',\n",
    "#     'monomethylamine corrinoid protein',\n",
    "#     'dimethylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.249]',\n",
    "#     'dimethylamine corrinoid protein',\n",
    "#     'trimethylamine---corrinoid protein Co-methyltransferase [EC:2.1.1.250]',\n",
    "#     'trimethylamine corrinoid protein',\n",
    "#     'methyl-coenzyme M reductase alpha subunit [EC:2.8.4.1]',\n",
    "#     'methyl-coenzyme M reductase beta subunit [EC:2.8.4.1]',\n",
    "#     'methyl-coenzyme M reductase gamma subunit [EC:2.8.4.1]',\n",
    "#     'heterodisulfide reductase subunit A1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit B1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit C1 [EC:1.8.7.3]',\n",
    "#     'heterodisulfide reductase subunit A2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit B2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit C2 [EC:1.8.7.3 1.8.98.4 1.8.98.5 1.8.98.6]',\n",
    "#     'heterodisulfide reductase subunit D [EC:1.8.98.1]',\n",
    "#     'heterodisulfide reductase subunit E [EC:1.8.98.1]',\n",
    "#     'F420-non-reducing hydrogenase large subunit [EC:1.12.99.- 1.8.98.5]',\n",
    "#     'F420-non-reducing hydrogenase small subunit [EC:1.12.99.- 1.8.98.5]',\n",
    "#     'F420-non-reducing hydrogenase iron-sulfur subunit [EC:1.12.99.- 1.8.98.5 1.8.98.6]',\n",
    "#     'formate dehydrogenase (coenzyme F420) alpha subunit [EC:1.17.98.3 1.8.98.6]',\n",
    "#     'formate dehydrogenase (coenzyme F420) beta subunit [EC:1.17.98.3 1.8.98.6]']\n",
    "# descrip_df = pd.DataFrame([descrip], columns=df.columns)\n",
    "# df = pd.concat([kegg_row_df, descrip_df, df], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "649191c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002497785.1</td>\n",
       "      <td>Methanolinea</td>\n",
       "      <td>UBA144</td>\n",
       "      <td>1.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_002502925.1</td>\n",
       "      <td>Methanomassiliicoccaceae archaeon</td>\n",
       "      <td>UBA414</td>\n",
       "      <td>2.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_002506795.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA397</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>GCA_902764455.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>GCA_902768415.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11363</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>GCA_002502855.1</td>\n",
       "      <td>Methanobacterium</td>\n",
       "      <td>UBA279</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                      organism_name   Isolate  16S  tRNA\n",
       "0    GCA_902788045.1      uncultured Methanobrevibacter  RUG13330  0.0    32\n",
       "1    GCA_902774685.1      uncultured Methanobrevibacter  RUG11990  0.0    28\n",
       "2    GCA_002497785.1                       Methanolinea    UBA144  1.0    37\n",
       "3    GCA_002502925.1  Methanomassiliicoccaceae archaeon    UBA414  2.0    41\n",
       "4    GCA_002506795.1                   Methanobacterium    UBA397  0.0    38\n",
       "..               ...                                ...       ...  ...   ...\n",
       "218  GCA_902764455.1      uncultured Methanobrevibacter  RUG10962  0.0    35\n",
       "219  GCA_902802065.1      uncultured Methanobrevibacter  RUG14729  2.0    35\n",
       "220  GCA_902787415.1              methanogenic archaeon  RUG13270  0.0    35\n",
       "221  GCA_902768415.1      uncultured Methanobrevibacter  RUG11363  0.0    27\n",
       "222  GCA_002502855.1                   Methanobacterium    UBA279  0.0    37\n",
       "\n",
       "[223 rows x 5 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "33c18b50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>16S</th>\n",
       "      <th>tRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_902770715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11588</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902801725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14697</td>\n",
       "      <td>1.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GCA_902758945.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10414</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_902769175.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11428</td>\n",
       "      <td>0.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>GCA_902777475.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_902759195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10437</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>GCA_902784295.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12954</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_902788255.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13354</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>GCA_902762245.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>GCA_902770085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11531</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>GCA_902774705.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11996</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902788045.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Accession                  organism_name   Isolate  16S  tRNA\n",
       "38   GCA_902770715.1  uncultured Methanobrevibacter  RUG11588  0.0    30\n",
       "37   GCA_902801725.1  uncultured Methanobrevibacter  RUG14697  1.0    30\n",
       "166  GCA_902798335.1  uncultured Methanobrevibacter  RUG14349  0.0    31\n",
       "167  GCA_902757275.1  uncultured Methanobrevibacter  RUG10244  0.0    36\n",
       "75   GCA_902758945.1  uncultured Methanobrevibacter  RUG10414  0.0    33\n",
       "34   GCA_902769175.1  uncultured Methanobrevibacter  RUG11428  0.0    41\n",
       "125  GCA_902777475.1  uncultured Methanobrevibacter  RUG12275  0.0    34\n",
       "32   GCA_902759195.1  uncultured Methanobrevibacter  RUG10437  0.0    33\n",
       "174  GCA_902784295.1  uncultured Methanobrevibacter  RUG12954  0.0    31\n",
       "31   GCA_902788255.1  uncultured Methanobrevibacter  RUG13354  0.0    31\n",
       "177  GCA_902762245.1  uncultured Methanobrevibacter  RUG10743  0.0    30\n",
       "178  GCA_902770085.1  uncultured Methanobrevibacter  RUG11531  0.0    34\n",
       "123  GCA_902775005.1  uncultured Methanobrevibacter  RUG12025  0.0    30\n",
       "25   GCA_902762085.1  uncultured Methanobrevibacter  RUG10738  0.0    23\n",
       "140  GCA_902774705.1  uncultured Methanobrevibacter  RUG11996  0.0    82\n",
       "184  GCA_902765105.1  uncultured Methanobrevibacter  RUG11035  0.0    30\n",
       "185  GCA_902785855.1  uncultured Methanobrevibacter  RUG13115  0.0    28\n",
       "120  GCA_902757365.1  uncultured Methanobrevibacter  RUG10252  0.0    36\n",
       "21   GCA_902784715.1  uncultured Methanobrevibacter  RUG12995  0.0    27\n",
       "0    GCA_902788045.1  uncultured Methanobrevibacter  RUG13330  0.0    32"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_name'] = df['organism_name'].astype(str)\n",
    "df_sorted = df.sort_values('organism_name', ascending=True)\n",
    "df_sorted.tail(n=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "cefb5125",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sorted.to_csv('prokka_16s.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "14289b16",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "os.chdir(\"/Users/gracedai/Downloads\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "c18cc498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism_name</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>strain</th>\n",
       "      <th>fwdA</th>\n",
       "      <th>fmdA</th>\n",
       "      <th>fwdB</th>\n",
       "      <th>fmdB</th>\n",
       "      <th>fwdC</th>\n",
       "      <th>fmdC</th>\n",
       "      <th>fwdD</th>\n",
       "      <th>...</th>\n",
       "      <th>hdrA2</th>\n",
       "      <th>hdrB2</th>\n",
       "      <th>hdrC2</th>\n",
       "      <th>hdrD</th>\n",
       "      <th>hdrE</th>\n",
       "      <th>mvhA</th>\n",
       "      <th>mvhG</th>\n",
       "      <th>mvhD</th>\n",
       "      <th>fdhA</th>\n",
       "      <th>fdhB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#CO2 -&gt; methane</td>\n",
       "      <td># KEGG M00567</td>\n",
       "      <td>#</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00200</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00201</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00202</td>\n",
       "      <td>K00203</td>\n",
       "      <td>...</td>\n",
       "      <td>K03388</td>\n",
       "      <td>K03389</td>\n",
       "      <td>K03390</td>\n",
       "      <td>K08264</td>\n",
       "      <td>K08265</td>\n",
       "      <td>K14126</td>\n",
       "      <td>K14128</td>\n",
       "      <td>K14127</td>\n",
       "      <td>K22516</td>\n",
       "      <td>K00125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit A [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit B [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit C [EC...</td>\n",
       "      <td>formylmethanofuran dehydrogenase subunit D [EC...</td>\n",
       "      <td>...</td>\n",
       "      <td>heterodisulfide reductase subunit A2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit B2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit C2 [EC:1.8.7...</td>\n",
       "      <td>heterodisulfide reductase subunit D [EC:1.8.98.1]</td>\n",
       "      <td>heterodisulfide reductase subunit E [EC:1.8.98.1]</td>\n",
       "      <td>F420-non-reducing hydrogenase large subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase small subunit [E...</td>\n",
       "      <td>F420-non-reducing hydrogenase iron-sulfur subu...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) alpha su...</td>\n",
       "      <td>formate dehydrogenase (coenzyme F420) beta sub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Candidatus Methanogranum sp. U3.2.1</td>\n",
       "      <td>GCA_019262145.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Candidatus Methanoliparum sp. LAM-1</td>\n",
       "      <td>GCF_020885915.1</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis</td>\n",
       "      <td>GCF_902383905.1</td>\n",
       "      <td>na</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Candidatus Methanomassiliicoccus intestinalis ...</td>\n",
       "      <td>GCF_000404225.1</td>\n",
       "      <td>Issoire-Mx1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_902387285.1</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus</td>\n",
       "      <td>GCF_003711245.1</td>\n",
       "      <td>Mx-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Candidatus Methanomethylophilus alvus Mx1201</td>\n",
       "      <td>GCF_000300255.2</td>\n",
       "      <td>Mx1201</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Candidatus Methanomethylophilus sp. 1R26 (arch...</td>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>1R26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       organism_name assembly_accession  \\\n",
       "0                                    #CO2 -> methane      # KEGG M00567   \n",
       "1                                                  #                  #   \n",
       "2                Candidatus Methanogranum sp. U3.2.1    GCA_019262145.1   \n",
       "3                Candidatus Methanoliparum sp. LAM-1    GCF_020885915.1   \n",
       "4      Candidatus Methanomassiliicoccus intestinalis    GCF_902383905.1   \n",
       "5  Candidatus Methanomassiliicoccus intestinalis ...    GCF_000404225.1   \n",
       "6              Candidatus Methanomethylophilus alvus    GCF_902387285.1   \n",
       "7              Candidatus Methanomethylophilus alvus    GCF_003711245.1   \n",
       "8       Candidatus Methanomethylophilus alvus Mx1201    GCF_000300255.2   \n",
       "9  Candidatus Methanomethylophilus sp. 1R26 (arch...    GCF_001481295.1   \n",
       "\n",
       "        strain                                               fwdA  \\\n",
       "0            #                                             K00200   \n",
       "1            #  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2           na                                                  0   \n",
       "3        LAM-1                                                  0   \n",
       "4           na                                                  0   \n",
       "5  Issoire-Mx1                                                  0   \n",
       "6       Mx1201                                                  0   \n",
       "7        Mx-05                                                  0   \n",
       "8       Mx1201                                                  0   \n",
       "9         1R26                                                  0   \n",
       "\n",
       "                                                fmdA  \\\n",
       "0                                             K00200   \n",
       "1  formylmethanofuran dehydrogenase subunit A [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdB  \\\n",
       "0                                             K00201   \n",
       "1  formylmethanofuran dehydrogenase subunit B [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fmdC  \\\n",
       "0                                             K00202   \n",
       "1  formylmethanofuran dehydrogenase subunit C [EC...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fwdD  ...  \\\n",
       "0                                             K00203  ...   \n",
       "1  formylmethanofuran dehydrogenase subunit D [EC...  ...   \n",
       "2                                                  0  ...   \n",
       "3                                                  0  ...   \n",
       "4                                                  0  ...   \n",
       "5                                                  0  ...   \n",
       "6                                                  0  ...   \n",
       "7                                                  0  ...   \n",
       "8                                                  0  ...   \n",
       "9                                                  0  ...   \n",
       "\n",
       "                                               hdrA2  \\\n",
       "0                                             K03388   \n",
       "1  heterodisulfide reductase subunit A2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  2   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrB2  \\\n",
       "0                                             K03389   \n",
       "1  heterodisulfide reductase subunit B2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                               hdrC2  \\\n",
       "0                                             K03390   \n",
       "1  heterodisulfide reductase subunit C2 [EC:1.8.7...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrD  \\\n",
       "0                                             K08264   \n",
       "1  heterodisulfide reductase subunit D [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                hdrE  \\\n",
       "0                                             K08265   \n",
       "1  heterodisulfide reductase subunit E [EC:1.8.98.1]   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhA  \\\n",
       "0                                             K14126   \n",
       "1  F420-non-reducing hydrogenase large subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhG  \\\n",
       "0                                             K14128   \n",
       "1  F420-non-reducing hydrogenase small subunit [E...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                mvhD  \\\n",
       "0                                             K14127   \n",
       "1  F420-non-reducing hydrogenase iron-sulfur subu...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhA  \\\n",
       "0                                             K22516   \n",
       "1  formate dehydrogenase (coenzyme F420) alpha su...   \n",
       "2                                                  0   \n",
       "3                                                  0   \n",
       "4                                                  0   \n",
       "5                                                  0   \n",
       "6                                                  0   \n",
       "7                                                  0   \n",
       "8                                                  0   \n",
       "9                                                  0   \n",
       "\n",
       "                                                fdhB  \n",
       "0                                             K00125  \n",
       "1  formate dehydrogenase (coenzyme F420) beta sub...  \n",
       "2                                                  0  \n",
       "3                                                  0  \n",
       "4                                                  0  \n",
       "5                                                  0  \n",
       "6                                                  0  \n",
       "7                                                  0  \n",
       "8                                                  0  \n",
       "9                                                  0  \n",
       "\n",
       "[10 rows x 46 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs = pd.read_csv('co2.csv')\n",
    "prokka = pd.read_csv('prokkas.csv')\n",
    "rumen_only = pd.read_csv('rumen.csv')\n",
    "wgs.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "620a03c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = pd.merge(rumen_only, wgs, on='assembly_accession', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "d7c75738",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>strain_x</th>\n",
       "      <th>host</th>\n",
       "      <th>location</th>\n",
       "      <th>mcrA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000024185.1</td>\n",
       "      <td>Methanobrevibacter ruminantium</td>\n",
       "      <td>M1</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>Germany</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001027005.1</td>\n",
       "      <td>Methanosarcina barkeri</td>\n",
       "      <td>CM1</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001563245.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>Lamb</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000711215.1</td>\n",
       "      <td>Methanomicrobium mobile</td>\n",
       "      <td>BP</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_001481295.1</td>\n",
       "      <td>Candidatus Methanomethylophilus</td>\n",
       "      <td>1R26</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Denmark</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_003111625.1</td>\n",
       "      <td>Methanobrevibacter thaueri</td>\n",
       "      <td>DSM 11995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCF_000320505.1</td>\n",
       "      <td>Methanobrevibacter boviskoreani</td>\n",
       "      <td>JH1</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Korea</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCF_001729965.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>WGK6</td>\n",
       "      <td>Kangaroo</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCF_003268005.1</td>\n",
       "      <td>Methanosphaera</td>\n",
       "      <td>BMS</td>\n",
       "      <td>Steer</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCF_001548675.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>YE315</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Australia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCF_001477655.1</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_030168595.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-DZ1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCF_000404165.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>AbM4</td>\n",
       "      <td>Sheep</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCF_000621965.1</td>\n",
       "      <td>Methanobrevibacter wolinii</td>\n",
       "      <td>SH</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCF_900116525.1</td>\n",
       "      <td>Methanosarcina</td>\n",
       "      <td>DSM 11855</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCF_900114585.1</td>\n",
       "      <td>Methanobrevibacter olleyae</td>\n",
       "      <td>DSM 16632</td>\n",
       "      <td>Bovine</td>\n",
       "      <td>USA</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_030168575.1</td>\n",
       "      <td>Methanomassiliicoccales archaeon</td>\n",
       "      <td>LGM-RCC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCF_000762265.1</td>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>BRM9</td>\n",
       "      <td>Cow</td>\n",
       "      <td>New Zealand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_902789165.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13441</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_902762085.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10738</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_902799105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14438</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_902762095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10736</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_902800515.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14577</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_902765105.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11035</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_902763685.1</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG10898</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_902765875.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11103</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_902784715.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12995</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_902781855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12711</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_902757195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10238</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GCA_017540705.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6915</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GCA_002509015.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA188</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GCA_902775725.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12100</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GCA_902763115.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10830</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GCA_902758425.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10359</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GCA_902786095.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13139</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GCA_902759335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10447</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>GCA_017535165.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6752</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GCA_902756745.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10188</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>GCA_902757275.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10244</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>GCA_902785855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG13115</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GCA_017524805.1</td>\n",
       "      <td>Candidatus Methanomethylophilaceae archaeon</td>\n",
       "      <td>RGIG6768</td>\n",
       "      <td>Dairy cattle; Water buffalo; Yak; Goat; Sheep;...</td>\n",
       "      <td>China</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>GCA_902800035.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14542</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GCA_902798335.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14349</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GCA_902800615.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14594</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>GCA_002505605.1</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>UBA187</td>\n",
       "      <td>Cattle; high methane</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>GCA_902774605.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11987</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>GCA_902775675.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12086</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GCA_902768855.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11403</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>GCA_902798395.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14364</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>GCA_902771225.1</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11657</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Scotland</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession                              organism_name_x   strain_x  \\\n",
       "0     GCF_000024185.1              Methanobrevibacter ruminantium          M1   \n",
       "1     GCF_001027005.1                       Methanosarcina barkeri        CM1   \n",
       "2     GCF_001563245.1                   Methanobrevibacter olleyae       YLM1   \n",
       "4     GCF_000711215.1                      Methanomicrobium mobile         BP   \n",
       "5     GCF_001481295.1              Candidatus Methanomethylophilus       1R26   \n",
       "6     GCF_003111625.1                   Methanobrevibacter thaueri  DSM 11995   \n",
       "9     GCF_000320505.1              Methanobrevibacter boviskoreani        JH1   \n",
       "10    GCF_001729965.1                               Methanosphaera       WGK6   \n",
       "11    GCF_003268005.1                               Methanosphaera        BMS   \n",
       "12    GCF_001548675.1                           Methanobrevibacter      YE315   \n",
       "13    GCF_001477655.1                  Methanobrevibacter millerae        SM9   \n",
       "14    GCA_030168595.1             Methanomassiliicoccales archaeon    LGM-DZ1   \n",
       "15    GCF_000404165.1                           Methanobrevibacter       AbM4   \n",
       "16    GCF_000621965.1                   Methanobrevibacter wolinii         SH   \n",
       "17    GCF_900116525.1                               Methanosarcina  DSM 11855   \n",
       "18    GCF_900114585.1                  Methanobrevibacter olleyae   DSM 16632   \n",
       "19    GCA_030168575.1             Methanomassiliicoccales archaeon   LGM-RCC1   \n",
       "20    GCF_000762265.1                  Methanobacterium formicicum       BRM9   \n",
       "21    GCA_902789165.1                uncultured Methanobrevibacter   RUG13441   \n",
       "22    GCA_902762085.1                uncultured Methanobrevibacter   RUG10738   \n",
       "23    GCA_902799105.1                uncultured Methanobrevibacter   RUG14438   \n",
       "24    GCA_902774685.1                uncultured Methanobrevibacter   RUG11990   \n",
       "28    GCA_902775005.1                uncultured Methanobrevibacter   RUG12025   \n",
       "30    GCA_902762095.1                uncultured Methanobrevibacter   RUG10736   \n",
       "33    GCA_902800515.1                uncultured Methanobrevibacter   RUG14577   \n",
       "35    GCA_902787415.1                        methanogenic archaeon   RUG13270   \n",
       "36    GCA_902765105.1                uncultured Methanobrevibacter   RUG11035   \n",
       "37    GCA_902763685.1                        methanogenic archaeon   RUG10898   \n",
       "39    GCA_902765875.1                uncultured Methanobrevibacter   RUG11103   \n",
       "41    GCA_902784715.1                uncultured Methanobrevibacter   RUG12995   \n",
       "43    GCA_902781855.1                uncultured Methanobrevibacter   RUG12711   \n",
       "44    GCA_902757195.1                uncultured Methanobrevibacter   RUG10238   \n",
       "47    GCA_017540705.1  Candidatus Methanomethylophilaceae archaeon   RGIG6915   \n",
       "48    GCA_002509015.1                           Methanobrevibacter     UBA188   \n",
       "49    GCA_902775725.1                uncultured Methanobrevibacter   RUG12100   \n",
       "58    GCA_902763115.1                uncultured Methanobrevibacter   RUG10830   \n",
       "59    GCA_902758425.1                uncultured Methanobrevibacter   RUG10359   \n",
       "62    GCA_902786095.1                uncultured Methanobrevibacter   RUG13139   \n",
       "63    GCA_902759335.1                uncultured Methanobrevibacter   RUG10447   \n",
       "65    GCA_017535165.1  Candidatus Methanomethylophilaceae archaeon   RGIG6752   \n",
       "66    GCA_902756745.1                uncultured Methanobrevibacter   RUG10188   \n",
       "67    GCA_902757275.1                uncultured Methanobrevibacter   RUG10244   \n",
       "69    GCA_902785855.1                uncultured Methanobrevibacter   RUG13115   \n",
       "71    GCA_017524805.1  Candidatus Methanomethylophilaceae archaeon   RGIG6768   \n",
       "72    GCA_902802065.1                uncultured Methanobrevibacter   RUG14729   \n",
       "73    GCA_902800035.1                uncultured Methanobrevibacter   RUG14542   \n",
       "74    GCA_902798335.1                uncultured Methanobrevibacter   RUG14349   \n",
       "77    GCA_902800615.1                uncultured Methanobrevibacter   RUG14594   \n",
       "78    GCA_002505605.1                           Methanobrevibacter     UBA187   \n",
       "79    GCA_902757365.1                uncultured Methanobrevibacter   RUG10252   \n",
       "83    GCA_902774605.1                uncultured Methanobrevibacter   RUG11987   \n",
       "84    GCA_902775675.1                uncultured Methanobrevibacter   RUG12086   \n",
       "85    GCA_902768855.1                uncultured Methanobrevibacter   RUG11403   \n",
       "86    GCA_902798395.1                uncultured Methanobrevibacter   RUG14364   \n",
       "88    GCA_902784195.1                uncultured Methanobrevibacter   RUG12946   \n",
       "89    GCA_902771225.1                uncultured Methanobrevibacter   RUG11657   \n",
       "\n",
       "                                                 host     location  mcrA  \n",
       "0                                              Bovine      Germany     1  \n",
       "1                                                 Cow  New Zealand     1  \n",
       "2                                                Lamb  New Zealand     2  \n",
       "4                                                 Cow    Australia     1  \n",
       "5                                                 Cow      Denmark     1  \n",
       "6                                                 Cow          NaN     1  \n",
       "9                                              Cattle        Korea     1  \n",
       "10                                           Kangaroo    Australia     1  \n",
       "11                                              Steer    Australia     1  \n",
       "12                                             Cattle    Australia     1  \n",
       "13                                             Sheep   New Zealand     2  \n",
       "14                                                NaN        China     1  \n",
       "15                                             Sheep   New Zealand     1  \n",
       "16                                             Bovine          USA     1  \n",
       "17                                             Bovine          USA     1  \n",
       "18                                            Bovine           USA     2  \n",
       "19                                                NaN        China     1  \n",
       "20                                                Cow  New Zealand     2  \n",
       "21                                                Cow     Scotland     1  \n",
       "22                                                Cow     Scotland     2  \n",
       "23                                                Cow     Scotland     1  \n",
       "24                                                Cow     Scotland     2  \n",
       "28                                                Cow     Scotland     1  \n",
       "30                                                Cow     Scotland     1  \n",
       "33                                                Cow     Scotland     2  \n",
       "35                                                Cow     Scotland     1  \n",
       "36                                                Cow     Scotland     1  \n",
       "37                                                Cow     Scotland     1  \n",
       "39                                                Cow     Scotland     1  \n",
       "41                                                Cow     Scotland     1  \n",
       "43                                                Cow     Scotland     1  \n",
       "44                                                Cow     Scotland     1  \n",
       "47  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "48                               Cattle; high methane          NaN     1  \n",
       "49                                                Cow     Scotland     1  \n",
       "58                                                Cow     Scotland     2  \n",
       "59                                                Cow     Scotland     1  \n",
       "62                                                Cow     Scotland     1  \n",
       "63                                                Cow     Scotland     1  \n",
       "65  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "66                                                Cow     Scotland     1  \n",
       "67                                                Cow     Scotland     1  \n",
       "69                                                Cow     Scotland     2  \n",
       "71  Dairy cattle; Water buffalo; Yak; Goat; Sheep;...        China     1  \n",
       "72                                                Cow     Scotland     1  \n",
       "73                                                Cow     Scotland     1  \n",
       "74                                                Cow     Scotland     1  \n",
       "77                                                Cow     Scotland     1  \n",
       "78                               Cattle; high methane          NaN     1  \n",
       "79                                                Cow     Scotland     1  \n",
       "83                                                Cow     Scotland     1  \n",
       "84                                                Cow     Scotland     1  \n",
       "85                                                Cow     Scotland     1  \n",
       "86                                                Cow     Scotland     1  \n",
       "88                                                Cow     Scotland     1  \n",
       "89                                                Cow     Scotland     1  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out['mcrA'] = out['mcrA'].fillna(0).astype(int)\n",
    "out_mcrA = out[out['mcrA'] >= 1]\n",
    "out_mcrA = out_mcrA[['assembly_accession', 'organism_name_x', 'strain_x','host', 'location', 'mcrA']]\n",
    "out_mcrA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "52f52ca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_mcrA.to_csv('out_mcrA.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a99dc7c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a80607b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
