{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8270eb45",
   "metadata": {},
   "source": [
    "A Python script for retrieving given gene sequences in a given pathway (subset of genes) based on the methanogen database, specifically just the ones that were aligned from blast. Creates a fasta file and gtf file for this to be used in building a kallisto index for the differential expression analysis. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aa0ed9",
   "metadata": {},
   "source": [
    "The strains I have identified in the samples based on blastn are:\n",
    "- GCF_001563245.1\n",
    "- GCF_001477655.1\n",
    "- GCF_000012545.1\n",
    "- GCA_902762095.1\n",
    "- GCA_000499765.1\n",
    "\n",
    "The genes of interest include (excludes vhus?):\n",
    "fwdA\n",
    "fmdA\n",
    "fwdB\n",
    "fwdC\n",
    "fmdC\n",
    "fwdD\n",
    "fwdE\n",
    "fmdE\n",
    "fwdF\n",
    "fwdG\n",
    "ftr\n",
    "mch\n",
    "mtd\n",
    "hmd\n",
    "mer\n",
    "mtrA\n",
    "mtrB\n",
    "mtrC\n",
    "mtrD\n",
    "mtrE\n",
    "mtrF\n",
    "mtrG\n",
    "mtrH\n",
    "mcrA\n",
    "mcrB\n",
    "mcrG\n",
    "mrtA\n",
    "mrtB\n",
    "mrtG\n",
    "hdrA1\n",
    "hdrA2\n",
    "hdrD\n",
    "hdrE\n",
    "mvhA\n",
    "vhcA\n",
    "mvhG\n",
    "vhcG\n",
    "mvhD\n",
    "vhcD\n",
    "fdhA\n",
    "fdhB\n",
    "pta\n",
    "acs\n",
    "cdhC\n",
    "cdhE\n",
    "acsC\n",
    "cdhD\n",
    "acsD\n",
    "mtaA\n",
    "mtaB\n",
    "mtaC\n",
    "mtbA\n",
    "mtmB\n",
    "mtbB\n",
    "mtbC\n",
    "mttB\n",
    "mttC\n",
    "mtbA\n",
    "mtmB\n",
    "mtmC\n",
    "mtbB\n",
    "mtbC\n",
    "mttB\n",
    "mttC\n",
    "mtbA\n",
    "mtmB\n",
    "mtmC\n",
    "mtbB\n",
    "mtbC\n",
    "mttB\n",
    "mttC\n",
    "hdrB1\n",
    "hdrB2\n",
    "hdrC1\n",
    "hdrC2\n",
    "mtmC\n",
    "ackA\n",
    "ACSS1_2\n",
    "mtmC\n",
    "fwdH\n",
    "fmdB\n",
    "fmdD\n",
    "fmdF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "a84427e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/gracedai/Documents/MethanogeneDB\n"
     ]
    }
   ],
   "source": [
    "#Check that we are in the correct working directory\n",
    "import os \n",
    "print(os.getcwd())\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "1032c329",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/gracedai/Documents/MethanogeneDB\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "b263d8ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Users/gracedai/Documents/MethanogeneDB/GCA_000970225.1/GCF_000970225.1_ASM97022v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000092305.1/GCF_000092305.1_ASM9230v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970165.1/GCF_000970165.1_ASM97016v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170265.1/GCF_024170265.1_ASM2417026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502925.1/GCA_002502925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000091665.1/GCF_000091665.1_ASM9166v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_020885915.1/GCF_020885915.1_ASM2088591v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017535165.1/GCA_017535165.1_ASM1753516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000166095.1/GCF_000166095.1_ASM16609v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412935.1/GCA_017412935.1_ASM1741293v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508885.1/GCA_002508885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633965.1/GCF_029633965.1_ASM2963396v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114585.1/GCF_900114585.1_IMG-taxon_2593339150_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744315.1/GCF_000744315.1_ASM74431v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_028743435.1/GCF_028743435.1_ASM2874343v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002356395.1/GCA_002356395.1_ASM235639v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168575.1/GCA_030168575.1_ASM3016857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970325.1/GCF_000970325.1_ASM97032v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502965.1/GCA_002502965.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970265.1/GCF_000970265.1_ASM97026v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025525.1/GCF_000025525.1_ASM2552v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001729965.1/GCF_001729965.1_ASM172996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900215215.1/GCF_900215215.1_IMG-taxon_2703719067_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902800035.1/GCA_902800035.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000746075.1/GCF_000746075.1_ASM74607v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000196655.1/GCF_000196655.1_ASM19665v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013725.1/GCF_000013725.1_ASM1372v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_946463545.1/GCF_946463545.1_Mth-DSM-2095-CCS-WG_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003111625.1/GCF_003111625.1_ASM311162v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002252585.1/GCF_002252585.1_ASM225258v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017935945.1/GCA_017935945.1_ASM1793594v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000953115.1/GCA_000953115.1_DSM1535_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_021184045.1/GCF_021184045.1_ASM2118404v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017539065.1/GCA_017539065.1_ASM1753906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503105.1/GCA_002503105.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000217995.1/GCF_000217995.1_ASM21799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017447225.1/GCA_017447225.1_ASM1744722v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000251105.1/GCF_000251105.1_ASM25110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711905.1/GCF_000711905.1_ASM71190v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011005.1/GCF_000011005.1_ASM1100v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024167805.1/GCA_024167805.1_ASM2416780v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495885.1/GCA_002495885.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024808815.1/GCF_024808815.1_ASM2480881v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017540705.1/GCA_017540705.1_ASM1754070v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025397995.1/GCF_025397995.1_ASM2539799v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095815.1/GCA_900095815.1_SIV6_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000621965.1/GCF_000621965.1_ASM62196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813675.1/GCF_002813675.1_ASM281367v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017432245.1/GCA_017432245.1_ASM1743224v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468935.1/GCA_017468935.1_ASM1746893v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873415.1/GCF_017873415.1_ASM1787341v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000744455.1/GCF_000744455.1_ASM74445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969925.1/GCF_000969925.1_ASM96992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000327485.1/GCF_000327485.1_ASM32748v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902763685.1/GCA_902763685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496295.1/GCA_002496295.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900111645.1/GCF_900111645.1_IMG-taxon_2642422537_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024625.1/GCF_000024625.1_ASM2462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900109595.1/GCF_900109595.1_IMG-taxon_2654588138_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508925.1/GCA_002508925.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584625.1/GCA_003584625.1_ASM358462v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017504565.1/GCA_017504565.1_ASM1750456v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026016205.1/GCA_026016205.1_ASM2601620v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002287175.1/GCF_002287175.1_ASM228717v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151225.1/GCA_000151225.1_ASM15122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902795935.1/GCA_902795935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875315.1/GCF_017875315.1_ASM1787531v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555045.1/GCF_027555045.1_ASM2755504v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874455.1/GCF_017874455.1_ASM1787445v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563305.1/GCA_001563305.1_ASM156330v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506325.1/GCA_002506325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508825.1/GCA_002508825.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017399555.1/GCA_017399555.1_ASM1739955v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000306725.1/GCA_000306725.1_ASM30672v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506905.1/GCA_002506905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827225.1/GCF_902827225.1_Methanocaldococcus_lauensis_SG7_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214725.1/GCF_000214725.1_ASM21472v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009914355.1/GCF_009914355.1_ASM991435v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491285.1/GCA_003491285.1_ASM349128v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017512765.1/GCA_017512765.1_ASM1751276v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001481295.1/GCF_001481295.1_ASM148129v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509405.1/GCA_002509405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015765.1/GCF_000015765.1_ASM1576v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000300255.2/GCF_000300255.2_ASM30025v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171695.1/GCA_905171695.1_GoM-Arc1_E37_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000014945.1/GCF_000014945.1_ASM1494v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002496945.1/GCA_002496945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970185.1/GCF_000970185.1_ASM97018v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000220645.1/GCF_000220645.1_ASM22064v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030168595.1/GCA_030168595.1_ASM3016859v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902775675.1/GCA_902775675.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002973515.1/GCF_002973515.1_ASM297351v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001560915.1/GCA_001560915.1_ASM156091v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000179575.2/GCF_000179575.2_ASM17957v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509645.1/GCA_002509645.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001889405.1/GCF_001889405.1_ASM188940v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017549625.1/GCA_017549625.1_ASM1754962v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029854155.1/GCF_029854155.1_ASM2985415v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000235565.1/GCF_000235565.1_ASM23556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813695.1/GCF_002813695.1_ASM281369v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247815.1/GCA_030247815.1_ASM3024781v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003052125.1/GCF_003052125.1_ASM305212v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_905171685.1/GCA_905171685.1_GoM-Arc1_E50_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900177455.1/GCF_900177455.1_IMG-taxon_2706794887_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828535.1/GCF_000828535.1_ASM82853v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017461225.1/GCA_017461225.1_ASM1746122v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017524805.1/GCA_017524805.1_ASM1752480v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017508065.1/GCA_017508065.1_ASM1750806v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017408885.1/GCA_017408885.1_ASM1740888v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018141185.1/GCF_018141185.1_ASM1814118v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509655.1/GCA_002509655.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023016325.1/GCF_023016325.1_ASM2301632v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000021965.1/GCF_000021965.1_ASM2196v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505245.1/GCA_002505245.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009917665.1/GCF_009917665.1_ASM991766v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000404225.1/GCF_000404225.1_ASM40422v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508935.1/GCA_002508935.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813085.1/GCF_002813085.1_ASM281308v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000969885.1/GCF_000969885.1_ASM96988v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828575.1/GCF_000828575.1_ASM82857v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498705.1/GCA_002498705.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900095295.1/GCF_900095295.1_MCBB_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902775725.1/GCA_902775725.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900100715.1/GCF_900100715.1_IMG-taxon_2642422561_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434555.1/GCA_017434555.1_ASM1743455v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498605.1/GCA_002498605.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900101055.1/GCF_900101055.1_IMG-taxon_2634166298_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026625165.1/GCA_026625165.1_ASM2662516v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002504405.1/GCA_002504405.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506565.1/GCA_002506565.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505345.1/GCA_002505345.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_016907005.1/GCF_016907005.1_ASM1690700v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900114835.1/GCF_900114835.1_IMG-taxon_2642422540_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900116525.1/GCF_900116525.1_IMG-taxon_2595698250_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002945325.1/GCF_002945325.1_ASM294532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002509425.1/GCA_002509425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000011585.1/GCF_000011585.1_ASM1158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902782945.1/GCA_902782945.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017537385.1/GCA_017537385.1_ASM1753738v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023167465.1/GCF_023167465.1_ASM2316746v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017625.1/GCF_000017625.1_ASM1762v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494805.1/GCA_002494805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669925.1/GCF_019669925.1_ASM1966992v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017622775.1/GCA_017622775.1_ASM1762277v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019263745.1/GCF_019263745.1_ASM1926374v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902757365.1/GCA_902757365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902775005.1/GCA_902775005.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498365.1/GCA_002498365.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017434525.1/GCA_017434525.1_ASM1743452v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000151245.1/GCA_000151245.1_ASM15124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508845.1/GCA_002508845.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002505275.1/GCA_002505275.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508905.1/GCA_002508905.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000328665.1/GCF_000328665.1_ASM32866v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553445.1/GCA_017553445.1_ASM1755344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019669965.1/GCF_019669965.1_ASM1966996v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498805.1/GCA_002498805.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503785.1/GCA_002503785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016125.1/GCF_000016125.1_ASM1612v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875815.1/GCF_017875815.1_ASM1787581v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003584605.1/GCA_003584605.1_ASM358460v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_019262145.1/GCA_019262145.1_ASM1926214v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900103415.1/GCF_900103415.1_IMG-taxon_2593339167_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875625.1/GCF_017875625.1_ASM1787562v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875565.1/GCF_017875565.1_ASM1787556v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027555065.1/GCF_027555065.1_ASM2755506v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017357525.1/GCF_017357525.1_ASM1735752v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000350305.1/GCA_000350305.1_ASM35030v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902827205.1/GCA_902827205.1_Methanocaldococcus_lauensis_SG1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_030247745.1/GCA_030247745.1_ASM3024774v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508625.1/GCA_002508625.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017468685.1/GCA_017468685.1_ASM1746868v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000828555.1/GCF_000828555.1_ASM82855v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000275865.1/GCA_000275865.1_ASM27586v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000015825.1/GCF_000015825.1_ASM1582v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_027554905.1/GCF_027554905.1_ASM2755490v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875325.1/GCF_017875325.1_ASM1787532v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000430905.1/GCF_000430905.1_ASM43090v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024662215.1/GCF_024662215.1_ASM2466221v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000016525.1/GCF_000016525.1_ASM1652v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_026684035.1/GCF_026684035.1_ASM2668403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017553245.1/GCA_017553245.1_ASM1755324v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000063445.1/GCF_000063445.1_ASM6344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498765.1/GCA_002498765.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017874375.1/GCF_017874375.1_ASM1787437v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495325.1/GCA_002495325.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000739065.1/GCF_000739065.1_ASM73906v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498425.1/GCA_002498425.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_022846155.1/GCF_022846155.1_ASM2284615v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017561385.1/GCA_017561385.1_ASM1756138v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498375.1/GCA_002498375.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007185.1/GCF_000007185.1_ASM718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017165.1/GCF_000017165.1_ASM1716v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000762265.1/GCF_000762265.1_ASM76226v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003711245.1/GCF_003711245.1_ASM371124v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000147875.1/GCF_000147875.1_ASM14787v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000745485.1/GCF_000745485.1_ASM74548v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002508855.1/GCA_002508855.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000214415.1/GCF_000214415.1_ASM21441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013388255.1/GCF_013388255.1_ASM1338825v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000376965.1/GCF_000376965.1_ASM37696v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902383905.1/GCF_902383905.1_UHGG_MGYG-HGUT-02160_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002495685.1/GCA_002495685.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002503135.1/GCA_002503135.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970245.1/GCF_000970245.1_ASM97024v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000304355.2/GCF_000304355.2_Mb_MS2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000013445.1/GCF_000013445.1_ASM1344v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017411925.1/GCA_017411925.1_ASM1741192v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873625.1/GCF_017873625.1_ASM1787362v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000711215.1/GCF_000711215.1_ASM71121v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003491305.1/GCA_003491305.1_ASM349130v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000320505.1/GCF_000320505.1_ASM32050v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017407825.1/GCA_017407825.1_ASM1740782v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970005.1/GCF_000970005.1_ASM97000v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761105.1/GCF_013761105.1_ASM1376110v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970205.1/GCF_000970205.1_ASM97020v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_004363215.1/GCF_004363215.1_ASM436321v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024807035.1/GCF_024807035.1_ASM2480703v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017519045.1/GCA_017519045.1_ASM1751904v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761045.1/GCF_013761045.1_ASM1376104v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017873855.1/GCF_017873855.1_ASM1787385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001304615.2/GCF_001304615.2_ASM130461v2_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017412625.1/GCA_017412625.1_ASM1741262v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_023617305.1/GCF_023617305.1_ASM2361730v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000970045.1/GCF_000970045.1_ASM97004v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_009911715.1/GCA_009911715.1_ASM991171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025449255.1/GCF_025449255.1_ASM2544925v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017875395.1/GCF_017875395.1_ASM1787539v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_025914035.1/GCF_025914035.1_ASM2591403v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003385755.1/GCF_003385755.1_ASM338575v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017633855.1/GCA_017633855.1_ASM1763385v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000191585.1/GCF_000191585.1_ASM19158v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_013761055.1/GCF_013761055.1_ASM1376105v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000007065.1/GCF_000007065.1_ASM706v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002498785.1/GCA_002498785.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902789165.1/GCA_902789165.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900079125.1/GCA_900079125.1_MBBA_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902765875.1/GCA_902765875.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002813655.1/GCF_002813655.1_ASM281365v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017940815.1/GCA_017940815.1_ASM1794081v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017387505.1/GCA_017387505.1_ASM1738750v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000204415.1/GCF_000204415.1_ASM20441v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002506995.1/GCA_002506995.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000800805.1/GCF_000800805.1_ASM80080v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_003351865.1/GCF_003351865.1_ASM335186v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017410345.1/GCA_017410345.1_ASM1741034v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002494585.1/GCA_002494585.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000017185.1/GCF_000017185.1_ASM1718v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017473945.1/GCA_017473945.1_ASM1747394v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_002502465.1/GCA_002502465.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_024170505.1/GCF_024170505.1_ASM2417050v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017409525.1/GCA_017409525.1_ASM1740952v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_001458655.1/GCF_001458655.1_Mb9_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_017406825.1/GCA_017406825.1_ASM1740682v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000008645.1/GCF_000008645.1_ASM864v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000145295.1/GCF_000145295.1_ASM14529v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900036045.1/GCF_900036045.1_Methanoculleus_sp_MAB1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_900106905.1/GCF_900106905.1_IMG-taxon_2642422606_annotated_assembly_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000023985.1/GCF_000023985.1_ASM2398v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014203295.1/GCF_014203295.1_ASM1420329v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000341715.1/GCF_000341715.1_ASM34171v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_018502485.1/GCF_018502485.1_ASM1850248v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_029633915.1/GCF_029633915.1_ASM2963391v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_008033705.1/GCF_008033705.1_ASM803370v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_014207315.1/GCF_014207315.1_ASM1420731v1_genomic.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902802065.1/GCA_902802065.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_902787415.1/GCA_902787415.1.gff', '/Users/gracedai/Documents/MethanogeneDB/GCA_000025865.1/GCF_000025865.1_ASM2586v1_genomic.gff']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "281"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_files = []   \n",
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gff.gz\"):\n",
    "            gunzip(os.path.join(root, file))\n",
    "        if file.endswith(\".gff\"):\n",
    "            gff_files.append(os.path.join(root, file))\n",
    "print(gff_files)\n",
    "len(gff_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f642bd3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# store fasta files\n",
    "# methanogen_files = ['/Users/gracedai/Documents/MethanogeneDB/GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff',\n",
    "#                    '/Users/gracedai/Documents/MethanogeneDB/GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff',\n",
    "#                    '/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs/GCF_902762095.1_Rumen_uncultured_genome_RUG10736_genomic.gtf',\n",
    "#                    '/Users/gracedai/Documents/MethanogeneDB/GCA_000499765.1/GCA_000499765.1_Methanobacterium_genomic.gff', \n",
    "#                    '/Users/gracedai/Documents/MethanogeneDB/GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff']\n",
    "\n",
    "# List of strings\n",
    "names_list = ['GCA_902775005.1', 'GCF_000024185.1', 'GCA_902775675.1', 'GCA_902789165.1', 'GCF_001563245.1', 'GCF_001548675.1',\n",
    "'GCA_902800035.1', 'GCA_902755425.1', 'GCA_902775725.1', 'GCA_902762085.1', 'GCA_902765875.1', 'GCA_902784195.1',\n",
    "'GCF_001477655.1', 'GCF_003814835.1', 'GCF_023617305.1', 'GCA_902771225.1', 'GCA_902800615.1', 'GCF_022846155.1',\n",
    "'GCF_022846175.1', 'GCF_002813085.1', 'GCF_000016525.1', 'GCF_028743435.1', 'GCA_017427765.1', 'GCA_902757365.1',\n",
    "'GCA_002509015.1', 'GCF_003268005.1', 'GCA_017410345.1', 'GCA_017633855.1', 'GCF_000012545.1', 'GCF_902384015.1',\n",
    "'GCA_902757195.1', 'GCA_902802065.1', 'GCA_902798335.1', 'GCA_902774685.1', 'GCA_902762085.1', 'GCA_902799105.1',\n",
    "'GCA_902756745.1', 'GCA_002505605.1', 'GCA_902765105.1', 'GCA_902781855.1', 'GCA_902800515.1', 'GCF_003814835.1',\n",
    "'GCA_002497655.1', 'GCF_017875395.1', 'GCA_902774685.1', 'GCF_020885915.1', 'GCA_002509325.1', 'GCA_017553445.1',\n",
    "'GCF_019669925.1', 'GCF_000300255.2', 'GCF_003711245.1', 'GCF_902387285.1', 'GCF_900109595.1', 'GCA_902763115.1',\n",
    "'GCF_002973515.1', 'GCA_002494585.1', 'GCF_017875565.1', 'GCF_017875625.1', 'GCF_024807035.1', 'GCA_024167805.1',\n",
    "'GCF_000191585.1', 'GCF_900114585.1', 'GCA_902759335.1', 'GCA_902787415.1', 'GCA_902763115.1', 'GCF_003111625.1',\n",
    "'GCF_900109595.1', 'GCF_000023985.1', 'GCA_002502805.1', 'GCA_902763685.1', 'GCF_001563245.1', 'GCF_002813655.1',\n",
    "'GCF_002813675.1', 'GCF_002813695.1', 'GCF_021184045.1', 'GCA_902762095.1', 'GCF_000404165.1'] \n",
    "\n",
    "# Check if any file path contains a substring from the list\n",
    "matching_files = [file_path for file_path in gff_files if any(name in file_path for name in names_list)]\n",
    "\n",
    "# for matching_file in matching_files:\n",
    "#     print(matching_file)\n",
    "\n",
    "len(names_list)\n",
    "len(matching_files)\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "1316e212",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sh import gunzip"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b63f759",
   "metadata": {},
   "source": [
    "# Iterate through each of files, pull the genes and put into fasta file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "e3288df2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#methanogenesis pathways, for co2, acetate, amines\n",
    "gene_ids = [\n",
    "    \"fwdA\", \"fmdA\", \"fwdB\", \"fwdC\", \"fmdC\", \"fwdD\", \"fwdE\", \"fmdE\", \"fwdF\", \"fwdG\",\n",
    "    \"ftr\", \"mch\", \"mtd\", \"hmd\", \"mer\", \"mtrA\", \"mtrB\", \"mtrC\", \"mtrD\", \"mtrE\", \"mtrF\",\n",
    "    \"mtrG\", \"mtrH\", \"mcrA\", \"mcrB\", \"mcrG\", \"hdrA1\", \"hdrA2\", \"hdrD\", \"hdrE\",\n",
    "    \"mvhA\", \"vhcA\", \"mvhG\", \"vhcG\", \"mvhD\", \"vhcD\", \"fdhA\", \"fdhB\", \"pta\", \"acs\",\n",
    "    \"cdhC\", \"cdhE\", \"acsC\", \"cdhD\", \"acsD\", \"mtaA\", \"mtaB\", \"mtaC\", \"mtbA\", \"mtmB\",\n",
    "    \"mtbB\", \"mtbC\", \"mttB\", \"mttC\", \"mtbA\", \"mtmB\", \"mtmC\", \"mtbB\", \"mtbC\", \"mttB\",\n",
    "    \"mttC\", \"mtbA\", \"mtmB\", \"mtmC\", \"mtbB\", \"mtbC\", \"mttB\", \"mttC\", \"mtbA\", \"mtmB\",\n",
    "    \"mtmC\", \"mtbB\", \"mtbC\", \"mttB\", \"mttC\", \"hdrB1\", \"hdrB2\", \"hdrC1\", \"hdrC2\", \"mtmC\", \"ackA\", \n",
    "    \"ACSS1_2\", \"mtmC\", \"fwdH\", \"fmdB\", \"fmdD\", \"fmdF\", \"mrtA\", \"mrtB\", \"mrtG\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "db977e60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>hdrD</td>\n",
       "      <td>27924</td>\n",
       "      <td>28628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mtd</td>\n",
       "      <td>9766</td>\n",
       "      <td>10596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mch</td>\n",
       "      <td>31746</td>\n",
       "      <td>32699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68079</td>\n",
       "      <td>68543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68630</td>\n",
       "      <td>68860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>3309</td>\n",
       "      <td>4637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1763</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>7195</td>\n",
       "      <td>8202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1764</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>2779</td>\n",
       "      <td>3885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1765</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbC</td>\n",
       "      <td>3891</td>\n",
       "      <td>4529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1766</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>1409</td>\n",
       "      <td>2422</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1767 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession  gene  start   stop\n",
       "0       GCA_902774685.1  hdrD  27924  28628\n",
       "1       GCA_902774685.1   mtd   9766  10596\n",
       "2       GCA_902774685.1   mch  31746  32699\n",
       "3       GCA_902774685.1   ftr  68079  68543\n",
       "4       GCA_902774685.1   ftr  68630  68860\n",
       "...                 ...   ...    ...    ...\n",
       "1762    GCA_902787415.1  mcrB   3309   4637\n",
       "1763    GCA_902787415.1  mtbA   7195   8202\n",
       "1764    GCA_902787415.1  mtbA   2779   3885\n",
       "1765    GCA_902787415.1  mtbC   3891   4529\n",
       "1766    GCA_902787415.1  mtbA   1409   2422\n",
       "\n",
       "[1767 rows x 4 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "\n",
    "# def extract_gene_id(line):\n",
    "#     match = re.search(r'gene_id \"([^\"]+)\"', line)\n",
    "#     if match:\n",
    "#         return match.group(1)\n",
    "#     else:\n",
    "#         return \"z\"\n",
    "\n",
    "\n",
    "#create dataframe to store the output \n",
    "df = pd.DataFrame(columns=[\"assembly_accession\", \"gene\", \"start\", \"stop\"])\n",
    "\n",
    "# iterate through the GFF files\n",
    "for path_name in matching_files:\n",
    "    species = path_name[40:55]  #keep species name from the path_name\n",
    "    \n",
    "    with open(path_name, \"r\") as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                fields = line.strip().split('\\t')\n",
    "                \n",
    "                for gene in gene_ids:\n",
    "                    if \"gene=\"+gene in line or (\"RefSeq\tgene\" in line and '\"'+gene+'\"' in line): #specify desired gene\n",
    "                        try:\n",
    "                            start = int(fields[3])\n",
    "                            stop = int(fields[4])\n",
    "                        except ValueError:\n",
    "                            # Handle the case where conversion to int fails\n",
    "                            start = int(fields[2])\n",
    "                            stop = int(fields[3])\n",
    "                        \n",
    "                        new_row = pd.DataFrame({\"assembly_accession\": [species],\n",
    "                                                \"gene\": [gene],\n",
    "                                                \"start\": [start],\n",
    "                                                \"stop\": [stop]})\n",
    "\n",
    "                        df = pd.concat([df, new_row], ignore_index=True)\n",
    "                        df = df.drop_duplicates()\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "b8061a8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'SM9_RS06605'"
      ]
     },
     "execution_count": 300,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "line = 'NZ_CP011266.1\tRefSeq\tgene\t1503908\t1505557\t.\t-\t.\tgene_id \"SM9_RS06605\"; transcript_id \"\"; db_xref \"GeneID:26736302\"; gbkey \"Gene\"; gene \"mcrA\"; gene_biotype \"protein_coding\"; locus_tag \"SM9_RS06605\"; old_locus_tag \"sm9_1353\";'\n",
    "re.search(r'gene_id \"([^\"]+)\"', line).group(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "4e6ce289",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['assembly_accession'] = df['assembly_accession'].apply(lambda x: 'GCA_902762095.1' if 'kallisto' in x else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "29ad58cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">GCA_902800615.1</th>\n",
       "      <th>mtrF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrG</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrH</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhD</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhG</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"24\" valign=\"top\">GCA_902802065.1</th>\n",
       "      <th>fdhA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fdhB</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ftr</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fwdC</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fwdG</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hdrD</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hmd</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mch</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mcrA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mcrB</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mcrG</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mer</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtd</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrB</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrC</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrD</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrE</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mtrH</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhD</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mvhG</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vhcD</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         start  stop\n",
       "assembly_accession gene             \n",
       "GCA_902800615.1    mtrF      1     1\n",
       "                   mtrG      1     1\n",
       "                   mtrH      2     2\n",
       "                   mvhA      1     1\n",
       "                   mvhD      1     1\n",
       "                   mvhG      1     1\n",
       "GCA_902802065.1    fdhA      2     2\n",
       "                   fdhB      2     2\n",
       "                   ftr       2     2\n",
       "                   fwdC      1     1\n",
       "                   fwdG      1     1\n",
       "                   hdrD      1     1\n",
       "                   hmd       1     1\n",
       "                   mch       1     1\n",
       "                   mcrA      1     1\n",
       "                   mcrB      1     1\n",
       "                   mcrG      1     1\n",
       "                   mer       1     1\n",
       "                   mtd       1     1\n",
       "                   mtrA      3     3\n",
       "                   mtrB      1     1\n",
       "                   mtrC      1     1\n",
       "                   mtrD      1     1\n",
       "                   mtrE      1     1\n",
       "                   mtrF      1     1\n",
       "                   mtrH      1     1\n",
       "                   mvhA      1     1\n",
       "                   mvhD      1     1\n",
       "                   mvhG      1     1\n",
       "                   vhcD      1     1"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_grouped = df.groupby(by=[\"assembly_accession\", \"gene\"]).count()\n",
    "df_grouped.sort_values(by=\"assembly_accession\", ascending=False)\n",
    "df_grouped.tail(n=30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "24924002",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>hdrD</td>\n",
       "      <td>27924</td>\n",
       "      <td>28628</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mtd</td>\n",
       "      <td>9766</td>\n",
       "      <td>10596</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mch</td>\n",
       "      <td>31746</td>\n",
       "      <td>32699</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68079</td>\n",
       "      <td>68543</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68630</td>\n",
       "      <td>68860</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>3309</td>\n",
       "      <td>4637</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1763</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>7195</td>\n",
       "      <td>8202</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1764</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>2779</td>\n",
       "      <td>3885</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1765</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbC</td>\n",
       "      <td>3891</td>\n",
       "      <td>4529</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1766</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>1409</td>\n",
       "      <td>2422</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1767 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession  gene  start   stop                  organism_name  \\\n",
       "0       GCA_902774685.1  hdrD  27924  28628  uncultured Methanobrevibacter   \n",
       "1       GCA_902774685.1   mtd   9766  10596  uncultured Methanobrevibacter   \n",
       "2       GCA_902774685.1   mch  31746  32699  uncultured Methanobrevibacter   \n",
       "3       GCA_902774685.1   ftr  68079  68543  uncultured Methanobrevibacter   \n",
       "4       GCA_902774685.1   ftr  68630  68860  uncultured Methanobrevibacter   \n",
       "...                 ...   ...    ...    ...                            ...   \n",
       "1762    GCA_902787415.1  mcrB   3309   4637          methanogenic archaeon   \n",
       "1763    GCA_902787415.1  mtbA   7195   8202          methanogenic archaeon   \n",
       "1764    GCA_902787415.1  mtbA   2779   3885          methanogenic archaeon   \n",
       "1765    GCA_902787415.1  mtbC   3891   4529          methanogenic archaeon   \n",
       "1766    GCA_902787415.1  mtbA   1409   2422          methanogenic archaeon   \n",
       "\n",
       "        strain  \n",
       "0     RUG11990  \n",
       "1     RUG11990  \n",
       "2     RUG11990  \n",
       "3     RUG11990  \n",
       "4     RUG11990  \n",
       "...        ...  \n",
       "1762  RUG13270  \n",
       "1763  RUG13270  \n",
       "1764  RUG13270  \n",
       "1765  RUG13270  \n",
       "1766  RUG13270  \n",
       "\n",
       "[1767 rows x 6 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#insert columns with organism and strain name\n",
    "species = pd.read_csv('/Users/gracedai/Downloads/both.csv')\n",
    "species = species[['assembly_accession', 'organism_name', 'strain']]\n",
    "df = pd.merge(df, species, left_on='assembly_accession', right_on='assembly_accession', how='left')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb8bf9a0",
   "metadata": {},
   "source": [
    "# Gather gene sequences and put into database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f040986e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#os.chdir(\"/Users/gracedai/Documents/asmg labs files/fna sequences\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "94e23059",
   "metadata": {},
   "outputs": [],
   "source": [
    "for root, dirs, files in os.walk(os.getcwd()):\n",
    "    for file in files:\n",
    "        if file.endswith(\".gz\"):\n",
    "            #os.remove(os.path.join(root, file))\n",
    "            gunzip(os.path.join(root, file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3568c7a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/Users/gracedai/Documents/MethanogeneDB')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b0cc37d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to find fasta file in a folder and extract the gene sequence\n",
    "def get_sequence(folder_name, start, stop):\n",
    "    fasta_file = [file for file in os.listdir(folder_name) if file.endswith(\".fna\")]\n",
    "    if not fasta_file:\n",
    "        return None\n",
    "    fasta_path = os.path.join(folder_name, fasta_file[0])\n",
    "    with open(fasta_path, \"r\") as file:\n",
    "        lines = file.readlines()\n",
    "        sequence = \"\".join(line.strip() for line in lines[1:] if not line.startswith(\">\"))\n",
    "    return sequence[start-1:stop] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "2ab5d043",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TTATATTTTTTGAGGTTCCAATCCTAATTGTTTAATCACATCAGTCACATCTGCATAATTACCATAAGTTGGTGATCCTACTCCTTTCACTTCTAATGGATAATTGTCTGCTATAACTGTAGCTAAATTATTGGCTCCCGCATTTAATGTATATTTGACATTTCTAGGACCAACAGTCGGTGTTGGAACGGTAATTCTTATATGTGGATACATTATCCTTGTAACAGCTATGAATTTTAACTGTTCCTTTAATGGGAATGGCGGATGATTTGCCATTGGTGTGTCTTCATAAGGATTAAAACCCATAATTGGTATTTCACCTAATGTATCAAAGTTATTTAAAAATCTTAAATGTTCTATTCTATCTTCAGCACTTTCTCCTAATCCCAGAAGCAGTCCTGAGGATAATTCTATGCCTGTATCACTTATCATTTGACATACATCAATTCTCTGCTGAAGTGTTTCTCCAGGTTTTGCTTCATTATAGACATCTTCATTTATTGTTTCTAGGTTGCAACAGATAGTATCACAGTTTAAATCATATAATTGATTTATGGATTTTTCTGTTAAATCTCCTCCGACATTAACCAATACTTCCAAATCCGTATTTTCCTTTACAATTCTGGTTGCATTAACTGCTTGTTTTCCTTTATATCCATATCCTCCGGAGCAACTAACTCTGGGAATGCCAGCCTCTTCAATTGATTTGGCGGCATTAAAAATTTCTTCATCGGATTTATAAAATGCATTATAATAACCTTTGGAAGAAGTCTCTTCGGCAAAACCGCAGTATTTGCATCTAGGTTGGATTTGACATTTGTTGGTGATGTGTATTGTTGATGTTAATTTAATAACTTTGGAAATGTTGTCCCTAATATTAACGGCAGTTTCAAAAAGTTTTTGCAAATCATTTTCGTTTTCTATATTTAATAATTCTAAAAACTCTTCATCTGTTAATTTTTTATTTTTGCTTGCTTTTTCTAATATGTCATCAATCAA\n"
     ]
    }
   ],
   "source": [
    "#test function\n",
    "out = get_sequence(\"GCA_001477655.1\", 517037, 518035)\n",
    "print(out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bea2e142",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>hdrD</td>\n",
       "      <td>27924</td>\n",
       "      <td>28628</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TCATTTAAGCTCCTCCTCTTCAGAGTCAAGATAGCCTAAAACAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mtd</td>\n",
       "      <td>9766</td>\n",
       "      <td>10596</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TATTATAAAAGAATTTTTTTAATCTATTTTTTGCATTTTCAGATAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mch</td>\n",
       "      <td>31746</td>\n",
       "      <td>32699</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>AGTTAAAAAAAAGAAAAAATGAATCTGCACAATTGCACAAAAATGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68079</td>\n",
       "      <td>68543</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TTAAAAAATAAAAAAAATAGAAGTAAAAAAAGAATAAGGATAGAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68630</td>\n",
       "      <td>68860</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>GACTGATTCCACTTGAGACAACAACTCAGGAGTGTATCCATTAGCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>3309</td>\n",
       "      <td>4637</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCTTTGCCCCAAGTGCGGACACACAGAAGCGTACTACGTCATCAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1763</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>7195</td>\n",
       "      <td>8202</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCAAGCAGTACCTCGACAGACTCTTGGACAATCTGGGACTAGCGGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1764</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>2779</td>\n",
       "      <td>3885</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCATGAAGAAGGCCAAGTCCAAGTCCAAGGCCAAGAAGAAGCGCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1765</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbC</td>\n",
       "      <td>3891</td>\n",
       "      <td>4529</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>GACCACATCAAACTCATGGCCGATCCCAACGGATTCGAGCTCTCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1766</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>1409</td>\n",
       "      <td>2422</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>TCATGGGCTTCGAGAAGAGGAGGTACGCACTCCCCGTCATCGCAGT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1767 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession  gene  start   stop                  organism_name  \\\n",
       "0       GCA_902774685.1  hdrD  27924  28628  uncultured Methanobrevibacter   \n",
       "1       GCA_902774685.1   mtd   9766  10596  uncultured Methanobrevibacter   \n",
       "2       GCA_902774685.1   mch  31746  32699  uncultured Methanobrevibacter   \n",
       "3       GCA_902774685.1   ftr  68079  68543  uncultured Methanobrevibacter   \n",
       "4       GCA_902774685.1   ftr  68630  68860  uncultured Methanobrevibacter   \n",
       "...                 ...   ...    ...    ...                            ...   \n",
       "1762    GCA_902787415.1  mcrB   3309   4637          methanogenic archaeon   \n",
       "1763    GCA_902787415.1  mtbA   7195   8202          methanogenic archaeon   \n",
       "1764    GCA_902787415.1  mtbA   2779   3885          methanogenic archaeon   \n",
       "1765    GCA_902787415.1  mtbC   3891   4529          methanogenic archaeon   \n",
       "1766    GCA_902787415.1  mtbA   1409   2422          methanogenic archaeon   \n",
       "\n",
       "        strain                                           Sequence  \n",
       "0     RUG11990  TCATTTAAGCTCCTCCTCTTCAGAGTCAAGATAGCCTAAAACAAAT...  \n",
       "1     RUG11990  TATTATAAAAGAATTTTTTTAATCTATTTTTTGCATTTTCAGATAA...  \n",
       "2     RUG11990  AGTTAAAAAAAAGAAAAAATGAATCTGCACAATTGCACAAAAATGA...  \n",
       "3     RUG11990  TTAAAAAATAAAAAAAATAGAAGTAAAAAAAGAATAAGGATAGAAA...  \n",
       "4     RUG11990  GACTGATTCCACTTGAGACAACAACTCAGGAGTGTATCCATTAGCT...  \n",
       "...        ...                                                ...  \n",
       "1762  RUG13270  CCTTTGCCCCAAGTGCGGACACACAGAAGCGTACTACGTCATCAGG...  \n",
       "1763  RUG13270  CCAAGCAGTACCTCGACAGACTCTTGGACAATCTGGGACTAGCGGG...  \n",
       "1764  RUG13270  CCATGAAGAAGGCCAAGTCCAAGTCCAAGGCCAAGAAGAAGCGCTG...  \n",
       "1765  RUG13270  GACCACATCAAACTCATGGCCGATCCCAACGGATTCGAGCTCTCCT...  \n",
       "1766  RUG13270  TCATGGGCTTCGAGAAGAGGAGGTACGCACTCCCCGTCATCGCAGT...  \n",
       "\n",
       "[1767 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get the fasta files and sequences, put them in a column and merge with previous df\n",
    "# store the gene sequences\n",
    "sequences = []\n",
    "\n",
    "for _, row in df.iterrows():\n",
    "    folder_name = row[\"assembly_accession\"]\n",
    "    #print(folder_name)\n",
    "    start, stop = row[\"start\"], row[\"stop\"]\n",
    "    sequence = get_sequence(folder_name, start, stop) #extract the gene sequence using function\n",
    "    sequences.append(sequence)\n",
    "\n",
    "# add column of sequences to existing dataframe\n",
    "df[\"Sequence\"] = sequences\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "13c61a33",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_020885915.1</td>\n",
       "      <td>cdhC</td>\n",
       "      <td>491485</td>\n",
       "      <td>492891</td>\n",
       "      <td>Candidatus Methanoliparum</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>ATGGCTGAAGAGTTTCCCTTTGATATAAGTCCGATGTATGAAGGGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_020885915.1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>21556</td>\n",
       "      <td>22344</td>\n",
       "      <td>Candidatus Methanoliparum</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>TCACATCTTCTCGTTTATCAACTCAGGATTAAAACCACCCATGACT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_020885915.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>110977</td>\n",
       "      <td>112311</td>\n",
       "      <td>Candidatus Methanoliparum</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>ATGGCAAAATATAAAGACAAAATAGATTTATACAGTGATAGGGGAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_020885915.1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>113474</td>\n",
       "      <td>114238</td>\n",
       "      <td>Candidatus Methanoliparum</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>ATGGTAGAGTATAAAAAACAATATTACCCTGGCGCGTCTGTTGTTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_020885915.1</td>\n",
       "      <td>mcrA</td>\n",
       "      <td>114258</td>\n",
       "      <td>115922</td>\n",
       "      <td>Candidatus Methanoliparum</td>\n",
       "      <td>LAM-1</td>\n",
       "      <td>ATGGTAGAGGATAAAAAGTTTATAAAAGCATTGTCTAAGAAATTTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>mvhA</td>\n",
       "      <td>34091</td>\n",
       "      <td>35518</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>TTAGATTTTTTTAATCATATTTCCTTCACTGTCATATACTTCTACA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>mvhG</td>\n",
       "      <td>35520</td>\n",
       "      <td>36446</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>TTATTTTCCTCCTTTTTGCATTTTAGCAGGGACTAAAGCAGCTGGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>GCA_902775005.1</td>\n",
       "      <td>mvhD</td>\n",
       "      <td>36467</td>\n",
       "      <td>36892</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12025</td>\n",
       "      <td>TTATCCTTCAGCATCTAATTGTGCTTTTAATGGGGAAGGTCCTAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>GCA_902784195.1</td>\n",
       "      <td>mtrF</td>\n",
       "      <td>67929</td>\n",
       "      <td>68135</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG12946</td>\n",
       "      <td>ATTTTATCTAGAACTGCCTTGTCATCTGCTGTGCTTATGATTAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>GCA_902757365.1</td>\n",
       "      <td>mvhA</td>\n",
       "      <td>74628</td>\n",
       "      <td>76052</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG10252</td>\n",
       "      <td>CGTAAACATAAGGGACGACGATTATGCGGATTCCGGATACAGTGCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1767 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession  gene   start    stop                  organism_name  \\\n",
       "33      GCA_020885915.1  cdhC  491485  492891      Candidatus Methanoliparum   \n",
       "27      GCA_020885915.1  mcrG   21556   22344      Candidatus Methanoliparum   \n",
       "28      GCA_020885915.1  mcrB  110977  112311      Candidatus Methanoliparum   \n",
       "29      GCA_020885915.1  mcrG  113474  114238      Candidatus Methanoliparum   \n",
       "30      GCA_020885915.1  mcrA  114258  115922      Candidatus Methanoliparum   \n",
       "...                 ...   ...     ...     ...                            ...   \n",
       "890     GCA_902775005.1  mvhA   34091   35518  uncultured Methanobrevibacter   \n",
       "891     GCA_902775005.1  mvhG   35520   36446  uncultured Methanobrevibacter   \n",
       "892     GCA_902775005.1  mvhD   36467   36892  uncultured Methanobrevibacter   \n",
       "1022    GCA_902784195.1  mtrF   67929   68135  uncultured Methanobrevibacter   \n",
       "883     GCA_902757365.1  mvhA   74628   76052  uncultured Methanobrevibacter   \n",
       "\n",
       "        strain                                           Sequence  \n",
       "33       LAM-1  ATGGCTGAAGAGTTTCCCTTTGATATAAGTCCGATGTATGAAGGGG...  \n",
       "27       LAM-1  TCACATCTTCTCGTTTATCAACTCAGGATTAAAACCACCCATGACT...  \n",
       "28       LAM-1  ATGGCAAAATATAAAGACAAAATAGATTTATACAGTGATAGGGGAA...  \n",
       "29       LAM-1  ATGGTAGAGTATAAAAAACAATATTACCCTGGCGCGTCTGTTGTTA...  \n",
       "30       LAM-1  ATGGTAGAGGATAAAAAGTTTATAAAAGCATTGTCTAAGAAATTTG...  \n",
       "...        ...                                                ...  \n",
       "890   RUG12025  TTAGATTTTTTTAATCATATTTCCTTCACTGTCATATACTTCTACA...  \n",
       "891   RUG12025  TTATTTTCCTCCTTTTTGCATTTTAGCAGGGACTAAAGCAGCTGGT...  \n",
       "892   RUG12025  TTATCCTTCAGCATCTAATTGTGCTTTTAATGGGGAAGGTCCTAAA...  \n",
       "1022  RUG12946  ATTTTATCTAGAACTGCCTTGTCATCTGCTGTGCTTATGATTAAAT...  \n",
       "883   RUG10252  CGTAAACATAAGGGACGACGATTATGCGGATTCCGGATACAGTGCA...  \n",
       "\n",
       "[1767 rows x 7 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_name'] = df['organism_name'].astype(str)\n",
    "df = df.rename(columns={'Isolate': 'strain'})\n",
    "df_sorted = df.sort_values('organism_name', ascending=True)\n",
    "df_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "cefb5125",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>hdrD</td>\n",
       "      <td>27924</td>\n",
       "      <td>28628</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TCATTTAAGCTCCTCCTCTTCAGAGTCAAGATAGCCTAAAACAAAT...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mtd</td>\n",
       "      <td>9766</td>\n",
       "      <td>10596</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TATTATAAAAGAATTTTTTTAATCTATTTTTTGCATTTTCAGATAA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>mch</td>\n",
       "      <td>31746</td>\n",
       "      <td>32699</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>AGTTAAAAAAAAGAAAAAATGAATCTGCACAATTGCACAAAAATGA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68079</td>\n",
       "      <td>68543</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>TTAAAAAATAAAAAAAATAGAAGTAAAAAAAGAATAAGGATAGAAA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_902774685.1</td>\n",
       "      <td>ftr</td>\n",
       "      <td>68630</td>\n",
       "      <td>68860</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG11990</td>\n",
       "      <td>GACTGATTCCACTTGAGACAACAACTCAGGAGTGTATCCATTAGCT...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>3309</td>\n",
       "      <td>4637</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCTTTGCCCCAAGTGCGGACACACAGAAGCGTACTACGTCATCAGG...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1763</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>7195</td>\n",
       "      <td>8202</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCAAGCAGTACCTCGACAGACTCTTGGACAATCTGGGACTAGCGGG...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1764</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>2779</td>\n",
       "      <td>3885</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>CCATGAAGAAGGCCAAGTCCAAGTCCAAGGCCAAGAAGAAGCGCTG...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1765</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbC</td>\n",
       "      <td>3891</td>\n",
       "      <td>4529</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>GACCACATCAAACTCATGGCCGATCCCAACGGATTCGAGCTCTCCT...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1766</th>\n",
       "      <td>GCA_902787415.1</td>\n",
       "      <td>mtbA</td>\n",
       "      <td>1409</td>\n",
       "      <td>2422</td>\n",
       "      <td>methanogenic archaeon</td>\n",
       "      <td>RUG13270</td>\n",
       "      <td>TCATGGGCTTCGAGAAGAGGAGGTACGCACTCCCCGTCATCGCAGT...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1767 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession  gene  start   stop                  organism_name  \\\n",
       "0       GCA_902774685.1  hdrD  27924  28628  uncultured Methanobrevibacter   \n",
       "1       GCA_902774685.1   mtd   9766  10596  uncultured Methanobrevibacter   \n",
       "2       GCA_902774685.1   mch  31746  32699  uncultured Methanobrevibacter   \n",
       "3       GCA_902774685.1   ftr  68079  68543  uncultured Methanobrevibacter   \n",
       "4       GCA_902774685.1   ftr  68630  68860  uncultured Methanobrevibacter   \n",
       "...                 ...   ...    ...    ...                            ...   \n",
       "1762    GCA_902787415.1  mcrB   3309   4637          methanogenic archaeon   \n",
       "1763    GCA_902787415.1  mtbA   7195   8202          methanogenic archaeon   \n",
       "1764    GCA_902787415.1  mtbA   2779   3885          methanogenic archaeon   \n",
       "1765    GCA_902787415.1  mtbC   3891   4529          methanogenic archaeon   \n",
       "1766    GCA_902787415.1  mtbA   1409   2422          methanogenic archaeon   \n",
       "\n",
       "        strain                                           Sequence  count  \n",
       "0     RUG11990  TCATTTAAGCTCCTCCTCTTCAGAGTCAAGATAGCCTAAAACAAAT...      1  \n",
       "1     RUG11990  TATTATAAAAGAATTTTTTTAATCTATTTTTTGCATTTTCAGATAA...      1  \n",
       "2     RUG11990  AGTTAAAAAAAAGAAAAAATGAATCTGCACAATTGCACAAAAATGA...      1  \n",
       "3     RUG11990  TTAAAAAATAAAAAAAATAGAAGTAAAAAAAGAATAAGGATAGAAA...      1  \n",
       "4     RUG11990  GACTGATTCCACTTGAGACAACAACTCAGGAGTGTATCCATTAGCT...      2  \n",
       "...        ...                                                ...    ...  \n",
       "1762  RUG13270  CCTTTGCCCCAAGTGCGGACACACAGAAGCGTACTACGTCATCAGG...      1  \n",
       "1763  RUG13270  CCAAGCAGTACCTCGACAGACTCTTGGACAATCTGGGACTAGCGGG...      1  \n",
       "1764  RUG13270  CCATGAAGAAGGCCAAGTCCAAGTCCAAGGCCAAGAAGAAGCGCTG...      2  \n",
       "1765  RUG13270  GACCACATCAAACTCATGGCCGATCCCAACGGATTCGAGCTCTCCT...      4  \n",
       "1766  RUG13270  TCATGGGCTTCGAGAAGAGGAGGTACGCACTCCCCGTCATCGCAGT...      3  \n",
       "\n",
       "[1767 rows x 8 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add count column \n",
    "df['count'] = df.groupby(by=[\"assembly_accession\", \"gene\"]).cumcount() + 1\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d74169f9",
   "metadata": {},
   "source": [
    "# Output dataframes to a formatted fasta file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "13586802",
   "metadata": {},
   "outputs": [],
   "source": [
    "#out_fasta should be \"name.fasta\" to ensure the output file is in fasta format\n",
    "def to_fasta(df, out_fasta):\n",
    "    df[\"organism_name\"] = df[\"organism_name\"].str.replace(\" \",\"-\")\n",
    "    df[\"strain\"] = df[\"strain\"].str.replace(\" \",\"-\")\n",
    "    with open(out_fasta, 'w') as file:\n",
    "        for _, row in df.iterrows():\n",
    "            header = f'>{row[\"assembly_accession\"]}|{row[\"organism_name\"]}|{row[\"strain\"]}|{row[\"gene\"]}|{row[\"start\"]}|{row[\"stop\"]}|{row[\"orientation\"]}'\n",
    "            sequence = row[\"Sequence\"]\n",
    "            file.write(f'{header}\\n{sequence}\\n')\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "7c79cf0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export to local fasta file\n",
    "to_fasta(df, \"kallisto_db_full.fa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2100975f",
   "metadata": {},
   "source": [
    "# Create Customm GTF file to match kallisto_db.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "77faade1",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/Users/gracedai/Documents/asmg_labs_files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f14df477",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/gracedai/Documents/asmg_labs_files'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "334a7b05",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re \n",
    "import pandas as pd\n",
    "\n",
    "id_df = pd.DataFrame(columns=['gene_id', 'gene', 'start', 'stop'])\n",
    "\n",
    "def extract_gene_id(line):\n",
    "    match = re.search(r'gene_id \"([^\"]+)\"', line)\n",
    "    if match:\n",
    "        return match.group(1)\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "def to_gtf(df, folder_name, out_gtf):\n",
    "    with open(out_gtf, 'w') as out_file:\n",
    "        index = 0\n",
    "        for _, row in df.iterrows():\n",
    "            assembly_accession = row['assembly_accession']\n",
    "            gtf_file = [file for file in os.listdir(folder_name) if assembly_accession[3:14] in file]\n",
    "            \n",
    "            if not gtf_file:\n",
    "                print(f\"No matching GTF file for {assembly_accession}\")\n",
    "                continue\n",
    "            \n",
    "            gtf_path = os.path.join(folder_name, gtf_file[0])\n",
    "            start = row['start']\n",
    "            stop = row['stop']\n",
    "            gene = row['gene']\n",
    "            \n",
    "            matched = False \n",
    "            with open(gtf_path, 'r') as gtf_file:\n",
    "                for line in gtf_file:\n",
    "                    if not line.startswith(\"#\"):\n",
    "                        fields = line.strip().split('\\t')\n",
    "                        gene_id = extract_gene_id(line)\n",
    "                  \n",
    "                        if fields[3] == str(start) and fields[4] == str(stop):\n",
    "                            out_file.write(line)\n",
    "                            id_df.loc[index, 'gene_id'] = gene_id \n",
    "                            id_df.loc[index, 'start'] = start\n",
    "                            id_df.loc[index, 'stop'] = stop \n",
    "                            id_df.loc[index, 'gene'] = gene \n",
    "                            matched = True\n",
    "                            index+=1\n",
    "                        elif fields[2] == str(start) and fields[3] == str(start):\n",
    "                            out_file.write(line)\n",
    "                            id_df.loc[index, 'gene_id'] = gene_id \n",
    "                            id_df.loc[index, 'start'] = start\n",
    "                            id_df.loc[index, 'stop'] = stop \n",
    "                            id_df.loc[index, 'gene'] = gene \n",
    "                            matched = True\n",
    "                            index+=1\n",
    "                        \n",
    "            if not matched:\n",
    "                print(f\"No matching entry for {assembly_accession} and gene {row['gene']} {start} {stop}\")\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "213e1d9b",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#test the function\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m testdf \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[38;5;241m.\u001b[39mhead(n\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m10\u001b[39m)\n\u001b[1;32m      3\u001b[0m to_gtf(testdf, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtest\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      4\u001b[0m id_df\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "#test the function\n",
    "testdf = df.head(n=10)\n",
    "to_gtf(testdf, \"/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs\", \"test\")\n",
    "id_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "139f8f45",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[9], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m to_gtf(\u001b[43mdf\u001b[49m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mkallisto_db.gtf\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      2\u001b[0m id_df\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "to_gtf(df, \"/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs\", \"kallisto_db.gtf\")\n",
    "id_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "840273b9",
   "metadata": {},
   "source": [
    "# Edit fasta file to contain unique gene ids from the above gtf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "3d751c27",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>count</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>517037</td>\n",
       "      <td>518035</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>TTATATTTTTTGAGGTTCCAATCCTAATTGTTTAATCACATCAGTC...</td>\n",
       "      <td>1</td>\n",
       "      <td>SM9_RS01950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>518159</td>\n",
       "      <td>519172</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>TTATTTGTCTTTACCTAAGTAATCTAATACGGTAGGAACGATTTCT...</td>\n",
       "      <td>2</td>\n",
       "      <td>SM9_RS01955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>519586</td>\n",
       "      <td>521085</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...</td>\n",
       "      <td>3</td>\n",
       "      <td>SM9_RS01960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mer</td>\n",
       "      <td>537737</td>\n",
       "      <td>538699</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>TTAGAAACTTGCAATTACGTCTCCTAATAATTTAATGGATTCTTTT...</td>\n",
       "      <td>1</td>\n",
       "      <td>SM9_RS02065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>768353</td>\n",
       "      <td>768907</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>CTAAGCTTCAGCTTCTTCTTTTTTATTTACAATAACTAATTTTTTG...</td>\n",
       "      <td>1</td>\n",
       "      <td>SM9_RS03260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>mtrD</td>\n",
       "      <td>1662278</td>\n",
       "      <td>1662979</td>\n",
       "      <td>Methanobrevibacter-olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>TTAAATACCTCCAATCATTAAAACAATGAAAATTGCAGCTACAAGA...</td>\n",
       "      <td>1</td>\n",
       "      <td>YLM1_RS06950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>mtrE</td>\n",
       "      <td>1662992</td>\n",
       "      <td>1663864</td>\n",
       "      <td>Methanobrevibacter-olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>TTATTCCTCATATGGTCCGAATTTTTCTCTTGCAGACTTTTCAAGT...</td>\n",
       "      <td>1</td>\n",
       "      <td>YLM1_RS06955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>mcrA</td>\n",
       "      <td>1664648</td>\n",
       "      <td>1666300</td>\n",
       "      <td>Methanobrevibacter-olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>TTATTTTGCTGGAATGATGATAGATCTTTCACCTGCAGGTTCGAAT...</td>\n",
       "      <td>2</td>\n",
       "      <td>YLM1_RS06960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>mcrG</td>\n",
       "      <td>1666316</td>\n",
       "      <td>1667068</td>\n",
       "      <td>Methanobrevibacter-olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>CTATAATTCTGGGTTGTAACCAAAATTAGATCTGGTGACGTGTATT...</td>\n",
       "      <td>2</td>\n",
       "      <td>YLM1_RS06965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>GCA_001563245.1</td>\n",
       "      <td>mcrB</td>\n",
       "      <td>1668246</td>\n",
       "      <td>1669577</td>\n",
       "      <td>Methanobrevibacter-olleyae</td>\n",
       "      <td>YLM1</td>\n",
       "      <td>TTAGATGTCACCTTTTATATCAGCAGCTGCTAAAGCAACATATTTG...</td>\n",
       "      <td>2</td>\n",
       "      <td>YLM1_RS06980</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession  gene    start     stop                organism_name  \\\n",
       "0      GCA_001477655.1   hmd   517037   518035  Methanobrevibacter-millerae   \n",
       "1      GCA_001477655.1   hmd   518159   519172  Methanobrevibacter-millerae   \n",
       "2      GCA_001477655.1   hmd   519586   521085  Methanobrevibacter-millerae   \n",
       "3      GCA_001477655.1   mer   537737   538699  Methanobrevibacter-millerae   \n",
       "4      GCA_001477655.1  mtrA   768353   768907  Methanobrevibacter-millerae   \n",
       "..                 ...   ...      ...      ...                          ...   \n",
       "108    GCA_001563245.1  mtrD  1662278  1662979   Methanobrevibacter-olleyae   \n",
       "109    GCA_001563245.1  mtrE  1662992  1663864   Methanobrevibacter-olleyae   \n",
       "110    GCA_001563245.1  mcrA  1664648  1666300   Methanobrevibacter-olleyae   \n",
       "111    GCA_001563245.1  mcrG  1666316  1667068   Methanobrevibacter-olleyae   \n",
       "112    GCA_001563245.1  mcrB  1668246  1669577   Methanobrevibacter-olleyae   \n",
       "\n",
       "    strain                                           Sequence  count  \\\n",
       "0      SM9  TTATATTTTTTGAGGTTCCAATCCTAATTGTTTAATCACATCAGTC...      1   \n",
       "1      SM9  TTATTTGTCTTTACCTAAGTAATCTAATACGGTAGGAACGATTTCT...      2   \n",
       "2      SM9  ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...      3   \n",
       "3      SM9  TTAGAAACTTGCAATTACGTCTCCTAATAATTTAATGGATTCTTTT...      1   \n",
       "4      SM9  CTAAGCTTCAGCTTCTTCTTTTTTATTTACAATAACTAATTTTTTG...      1   \n",
       "..     ...                                                ...    ...   \n",
       "108   YLM1  TTAAATACCTCCAATCATTAAAACAATGAAAATTGCAGCTACAAGA...      1   \n",
       "109   YLM1  TTATTCCTCATATGGTCCGAATTTTTCTCTTGCAGACTTTTCAAGT...      1   \n",
       "110   YLM1  TTATTTTGCTGGAATGATGATAGATCTTTCACCTGCAGGTTCGAAT...      2   \n",
       "111   YLM1  CTATAATTCTGGGTTGTAACCAAAATTAGATCTGGTGACGTGTATT...      2   \n",
       "112   YLM1  TTAGATGTCACCTTTTATATCAGCAGCTGCTAAAGCAACATATTTG...      2   \n",
       "\n",
       "          gene_id  \n",
       "0     SM9_RS01950  \n",
       "1     SM9_RS01955  \n",
       "2     SM9_RS01960  \n",
       "3     SM9_RS02065  \n",
       "4     SM9_RS03260  \n",
       "..            ...  \n",
       "108  YLM1_RS06950  \n",
       "109  YLM1_RS06955  \n",
       "110  YLM1_RS06960  \n",
       "111  YLM1_RS06965  \n",
       "112  YLM1_RS06980  \n",
       "\n",
       "[113 rows x 9 columns]"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new = pd.merge(df, id_df, on=['gene', 'start', 'stop'], how='right')\n",
    "df_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "64708dd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#out_fasta should be \"name.fasta\" to ensure the output file is in fasta format\n",
    "def to_fasta2(df, out_fasta):\n",
    "    df[\"organism_name\"] = df[\"organism_name\"].str.replace(\" \",\"-\")\n",
    "    df[\"strain\"] = df[\"strain\"].str.replace(\" \",\"-\")\n",
    "    with open(out_fasta, 'w') as file:\n",
    "        for _, row in df.iterrows():\n",
    "            header = f'>{row[\"gene_id\"]}|{row[\"organism_name\"]}|{row[\"strain\"]}|{row[\"gene\"]}|{row[\"count\"]}|{row[\"start\"]}|{row[\"stop\"]}'\n",
    "            sequence = row[\"Sequence\"]\n",
    "            file.write(f'{header}\\n{sequence}\\n')\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "52d3b34f",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df_new' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[49], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#export to local fasta file\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m to_fasta2(\u001b[43mdf_new\u001b[49m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mkallisto_db2.fa\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df_new' is not defined"
     ]
    }
   ],
   "source": [
    "#export to local fasta file\n",
    "to_fasta2(df_new, \"kallisto_db2.fa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be6f991a",
   "metadata": {},
   "source": [
    "# Get sequences of interest for 3NOP analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d2f86e02",
   "metadata": {},
   "outputs": [],
   "source": [
    "#os.chdir(\"/Users/gracedai/Documents/asmg_labs_files/kallisto_gtfs\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "0a8ecad4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#GCA numbers to keep \n",
    "accession_ids = [\"GCA_001548675.1\", \"GCA_003814835.1\", \"GCA_902775675.1\", \"GCA_902789165.1\", \n",
    "                 \"GCA_902775005.1\", \"GCA_902775725.1\", \"GCA_000024185.1\", \"GCA_001477655.1\", \n",
    "                \"GCA_902765875.1\", \"GCA_902757365.1\", \"GCA_902800035.1\", \"GCA_902802065.1\", \n",
    "                \"GCA_000012545.1\", \"GCA_003268005.1\", \"GCA_902387285.1\", \"GCA_017427765.1\", \n",
    "                 \"GCA_022846175.1\", \"GCA_000404165.1\", \"GCA_001563245.1\", \"GCA_902384015.1\"]\n",
    "\n",
    "accession_gffs = [file_path for file_path in gff_files if any(name in file_path for name in accession_ids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "ff94bea0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(accession_gffs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "164a4c39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(accession_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "be643d70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/Users/gracedai/Documents/MethanogeneDB/GCA_001477655.1/GCF_001477655.1_ASM147765v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902800035.1/GCA_902800035.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902387285.1/GCF_902387285.1_UHGG_MGYG-HGUT-02456_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_017427765.1/GCA_017427765.1_ASM1742776v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_001563245.1/GCF_001563245.1_ASM156324v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_000024185.1/GCF_000024185.1_ASM2418v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902775675.1/GCA_902775675.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902384015.1/GCF_902384015.1_UHGG_MGYG-HGUT-02164_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_022846175.1/GCF_022846175.1_ASM2284617v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_000404165.1/GCF_000404165.1_ASM40416v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902775725.1/GCA_902775725.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_003268005.1/GCF_003268005.1_ASM326800v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902757365.1/GCA_902757365.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902775005.1/GCA_902775005.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_003814835.1/GCF_003814835.1_ASM381483v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902789165.1/GCA_902789165.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902765875.1/GCA_902765875.1.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_001548675.1/GCF_001548675.1_ASM154867v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_000012545.1/GCF_000012545.1_ASM1254v1_genomic.gff',\n",
       " '/Users/gracedai/Documents/MethanogeneDB/GCA_902802065.1/GCA_902802065.1.gff']"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "accession_gffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "7a77ca31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>orientation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>517037</td>\n",
       "      <td>518035</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>518159</td>\n",
       "      <td>519172</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>519586</td>\n",
       "      <td>521085</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mer</td>\n",
       "      <td>537737</td>\n",
       "      <td>538699</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>768353</td>\n",
       "      <td>768907</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>11100</td>\n",
       "      <td>12500</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fwdG</td>\n",
       "      <td>13073</td>\n",
       "      <td>13303</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhB</td>\n",
       "      <td>18039</td>\n",
       "      <td>19259</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>19274</td>\n",
       "      <td>21334</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>16144</td>\n",
       "      <td>16677</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>439 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession  gene   start    stop orientation\n",
       "0      GCA_001477655.1   hmd  517037  518035           -\n",
       "1      GCA_001477655.1   hmd  518159  519172           -\n",
       "2      GCA_001477655.1   hmd  519586  521085           +\n",
       "3      GCA_001477655.1   mer  537737  538699           -\n",
       "4      GCA_001477655.1  mtrA  768353  768907           -\n",
       "..                 ...   ...     ...     ...         ...\n",
       "434    GCA_902802065.1  fdhA   11100   12500           -\n",
       "435    GCA_902802065.1  fwdG   13073   13303           -\n",
       "436    GCA_902802065.1  fdhB   18039   19259           -\n",
       "437    GCA_902802065.1  fdhA   19274   21334           -\n",
       "438    GCA_902802065.1  mtrA   16144   16677           -\n",
       "\n",
       "[439 rows x 5 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# need to account for whether the protein is encoded on the forward or reverse strand \n",
    "import pandas as pd\n",
    "import re\n",
    "\n",
    "#create dataframe to store the output \n",
    "protein_df = pd.DataFrame(columns=[\"assembly_accession\", \"gene\", \"start\", \"stop\"])\n",
    "\n",
    "# iterate through the GFF files\n",
    "for path_name in accession_gffs:\n",
    "    species = path_name[40:55]  #keep species name from the path_name\n",
    "    \n",
    "    with open(path_name, \"r\") as file:\n",
    "        for line in file:\n",
    "            if not line.startswith('#'):\n",
    "                fields = line.strip().split('\\t')\n",
    "                \n",
    "                for gene in gene_ids:\n",
    "                    if \"gene=\"+gene in line or (\"RefSeq\tgene\" in line and '\"'+gene+'\"' in line): #specify desired gene\n",
    "                        try:\n",
    "                            start = int(fields[3])\n",
    "                            stop = int(fields[4])\n",
    "                            orientation = fields[6]\n",
    "                        except ValueError:\n",
    "                            # Handle the case where conversion to int fails, for prodigal files\n",
    "                            start = int(fields[2])\n",
    "                            stop = int(fields[3])\n",
    "                            orientation = fields[5]\n",
    "                        \n",
    "                        new_row = pd.DataFrame({\"assembly_accession\": [species],\n",
    "                                                \"gene\": [gene],\n",
    "                                                \"start\": [start],\n",
    "                                                \"stop\": [stop],\n",
    "                                               \"orientation\": [orientation]})\n",
    "\n",
    "                        protein_df = pd.concat([protein_df, new_row], ignore_index=True)\n",
    "                        protein_df = protein_df.drop_duplicates()\n",
    "\n",
    "protein_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "13993abb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "def get_complement(sequence):\n",
    "    #Returns the complement of a DNA sequence\n",
    "    complement = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C'}\n",
    "    return \"\".join(complement[base] for base in sequence)\n",
    "\n",
    "def get_sequence(folder_name, start, stop, orientation):\n",
    "    fasta_file = [file for file in os.listdir(folder_name) if file.endswith(\".fna\")]\n",
    "    if not fasta_file:\n",
    "        return None\n",
    "    fasta_path = os.path.join(folder_name, fasta_file[0])\n",
    "    \n",
    "    with open(fasta_path, \"r\") as file:\n",
    "        lines = file.readlines()\n",
    "        sequence = \"\".join(line.strip() for line in lines[1:] if not line.startswith(\">\"))\n",
    "\n",
    "    extracted_sequence = sequence[start-1:stop]\n",
    "\n",
    "    #if the orientation is negative, reverse and complement the sequence\n",
    "    if orientation == \"-\":\n",
    "        extracted_sequence = get_complement(extracted_sequence[::-1])\n",
    "\n",
    "    return extracted_sequence\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "97aeaf02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATGGCTGATAAAAAATTCTTAGATGCAATGAAACAAAAATTCAGTGAAGATCCAACTGAAAAAACAACCCAATTCTATAACATGGGTGGTTGGACTCAATCTGAAAGAAAAACTGCATTTGTAAATGAAGGTAAAGAAATCGCTGAGAAAAGAGGAATTCCAATGTACAACCCAGACATTGGTACTCCTTTAGGTCAAAGAGCTTTAATGTCTTACCAATTATCTACTACTGATACTTTCGTTGAAGGTGACGATTTACACTTCATTAACAACGCAGCTATCCAACAAGCTTGGGATGACATTAGAAGAACTGTAATCGTAGGTTTAAACACTGCTCACAACGTTCTTGAAAAAAGATTAGGTATTGAAGTTACTCCTGAAACTATTACCAATTACTTAGAAACTGTAAACCACGCTATGCCTGGTGCAGCAGTTGTTCAAGAACACATGGTAGAAACCGACCCATTAGTAGTACAAGATTCCTATGTTAAAGTATTTACTGGTGACGACGAATTAGCAGACGAAATTGACTCTGCATTTGTATTGGACATCAACAAAGAGTTCCCTGAAGAACAAGCTGAAGCTTTAAAAGCTGAAGTTGGCGGAGCTGTATGGCAAGCTGTAAGAATTCCATCTATCGTAGGAAGAGTTTGTGACGGTGGTACCACTTCCAGATGGTCTGCAATGCAAATTGGTATGTCAATGATTTCAGCATACAACCAATGTGCTGGTGAAGGAGCTACTGGTGACTTCGCATACGCATCCAAACACGCAGAAGTTGTTCACATGGGTACTTACTTACCTGTAAGAAGAGCAAGAGCAGAAAACGAACTTGGTGGAGTTCCATTCGGATTCATGGCAGATATCTGTCAAGCTTCCAGAGTCTCACCTGATGACCCTGTAAAAACCTCATTAGAAGTAGTAGCTTTAGGTGCTGCTTTATACGACCAAATTTGGTTAGGTTCTTACATGTCTGGTGGTGTAGGATTCACTCAATATGCTACTGCAGCATATACTGATAATGTATTAGACGACTTCACCTACTATGGTAAAGACTACGTCGAAGACAAATACGGCGACTTATGTTCCGCACCTAACAACATGGACACTGTTTTAGATGTTGGTACTGAAGTAGCATTCTACGCATTAGACCAATACGAAGAATACCCAGCTTTACTTGAAACCCACTTCGGTGGATCTCAAAGAGCTTCCGTTATTTCCGCAGCTGCTGGTTGTTCCACTGCATTCGCTACTGGTAATGCACAAACTGGTTTAAGTGCATGGTACTTAGCTATGTACTTACACAAAGAACAACACTCCAGATTAGGATTCTACGGATACGATTTACAAGATCAATGTGGTGCTGCTAACGTATTCTCCATCAGAAACGATGAAGGATTACCACTCGAAATGAGAGGACCAAACTATCCTAACTACGCAATGAATGTAGGTCACCAAGGTGAATACGCTGGTATCGCACAAGCTCCTCACGCAGCTCGTGGAGACGCTTGGGCTTTCAACCCATTAATCAAAATCGCATTTGCTGACAAAAACTTATGCTTCGACTTCAGCCAAGTACGTGCTGAATTTGCTAAAGGTGCATTAAGAGAGTTCGAACCTGCAGGTGAAAGAACTGCAATCACTCCAGCAAAATAG\n"
     ]
    }
   ],
   "source": [
    "out = get_sequence(\"GCA_001477655.1\", 2218335, 2219990, \"-\")\n",
    "print(out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "33f199fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>orientation</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>strain_x</th>\n",
       "      <th>organism_name_y</th>\n",
       "      <th>strain_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>517037</td>\n",
       "      <td>518035</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>518159</td>\n",
       "      <td>519172</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>519586</td>\n",
       "      <td>521085</td>\n",
       "      <td>+</td>\n",
       "      <td>ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mer</td>\n",
       "      <td>537737</td>\n",
       "      <td>538699</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>768353</td>\n",
       "      <td>768907</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>11100</td>\n",
       "      <td>12500</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fwdG</td>\n",
       "      <td>13073</td>\n",
       "      <td>13303</td>\n",
       "      <td>-</td>\n",
       "      <td>GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhB</td>\n",
       "      <td>18039</td>\n",
       "      <td>19259</td>\n",
       "      <td>-</td>\n",
       "      <td>CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>19274</td>\n",
       "      <td>21334</td>\n",
       "      <td>-</td>\n",
       "      <td>CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>16144</td>\n",
       "      <td>16677</td>\n",
       "      <td>-</td>\n",
       "      <td>TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>439 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession  gene   start    stop orientation  \\\n",
       "0      GCA_001477655.1   hmd  517037  518035           -   \n",
       "1      GCA_001477655.1   hmd  518159  519172           -   \n",
       "2      GCA_001477655.1   hmd  519586  521085           +   \n",
       "3      GCA_001477655.1   mer  537737  538699           -   \n",
       "4      GCA_001477655.1  mtrA  768353  768907           -   \n",
       "..                 ...   ...     ...     ...         ...   \n",
       "434    GCA_902802065.1  fdhA   11100   12500           -   \n",
       "435    GCA_902802065.1  fwdG   13073   13303           -   \n",
       "436    GCA_902802065.1  fdhB   18039   19259           -   \n",
       "437    GCA_902802065.1  fdhA   19274   21334           -   \n",
       "438    GCA_902802065.1  mtrA   16144   16677           -   \n",
       "\n",
       "                                              Sequence  \\\n",
       "0    TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...   \n",
       "1    ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...   \n",
       "2    ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...   \n",
       "3    ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...   \n",
       "4    ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...   \n",
       "..                                                 ...   \n",
       "434  TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...   \n",
       "435  GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...   \n",
       "436  CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...   \n",
       "437  CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...   \n",
       "438  TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...   \n",
       "\n",
       "                   organism_name_x  strain_x                organism_name_y  \\\n",
       "0      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "1      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "2      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "3      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "4      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "..                             ...       ...                            ...   \n",
       "434  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "435  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "436  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "437  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "438  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "\n",
       "     strain_y  \n",
       "0         SM9  \n",
       "1         SM9  \n",
       "2         SM9  \n",
       "3         SM9  \n",
       "4         SM9  \n",
       "..        ...  \n",
       "434  RUG14729  \n",
       "435  RUG14729  \n",
       "436  RUG14729  \n",
       "437  RUG14729  \n",
       "438  RUG14729  \n",
       "\n",
       "[439 rows x 10 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nuc_sequences = []\n",
    "\n",
    "for _, row in protein_df.iterrows():\n",
    "    folder_name = row[\"assembly_accession\"]\n",
    "    #print(folder_name)\n",
    "    start, stop, orientation = row[\"start\"], row[\"stop\"], row[\"orientation\"]\n",
    "    sequence = get_sequence(folder_name, start, stop, orientation) #extract the gene sequence using function\n",
    "    nuc_sequences.append(sequence)\n",
    "\n",
    "# add column of sequences to existing dataframe\n",
    "protein_df[\"Sequence\"] = nuc_sequences\n",
    "protein_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "8a63fdb6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>orientation</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>strain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>517037</td>\n",
       "      <td>518035</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>518159</td>\n",
       "      <td>519172</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>519586</td>\n",
       "      <td>521085</td>\n",
       "      <td>+</td>\n",
       "      <td>ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mer</td>\n",
       "      <td>537737</td>\n",
       "      <td>538699</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>768353</td>\n",
       "      <td>768907</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>11100</td>\n",
       "      <td>12500</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fwdG</td>\n",
       "      <td>13073</td>\n",
       "      <td>13303</td>\n",
       "      <td>-</td>\n",
       "      <td>GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhB</td>\n",
       "      <td>18039</td>\n",
       "      <td>19259</td>\n",
       "      <td>-</td>\n",
       "      <td>CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>19274</td>\n",
       "      <td>21334</td>\n",
       "      <td>-</td>\n",
       "      <td>CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>16144</td>\n",
       "      <td>16677</td>\n",
       "      <td>-</td>\n",
       "      <td>TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>439 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession  gene   start    stop orientation  \\\n",
       "0      GCA_001477655.1   hmd  517037  518035           -   \n",
       "1      GCA_001477655.1   hmd  518159  519172           -   \n",
       "2      GCA_001477655.1   hmd  519586  521085           +   \n",
       "3      GCA_001477655.1   mer  537737  538699           -   \n",
       "4      GCA_001477655.1  mtrA  768353  768907           -   \n",
       "..                 ...   ...     ...     ...         ...   \n",
       "434    GCA_902802065.1  fdhA   11100   12500           -   \n",
       "435    GCA_902802065.1  fwdG   13073   13303           -   \n",
       "436    GCA_902802065.1  fdhB   18039   19259           -   \n",
       "437    GCA_902802065.1  fdhA   19274   21334           -   \n",
       "438    GCA_902802065.1  mtrA   16144   16677           -   \n",
       "\n",
       "                                              Sequence  \\\n",
       "0    TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...   \n",
       "1    ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...   \n",
       "2    ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...   \n",
       "3    ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...   \n",
       "4    ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...   \n",
       "..                                                 ...   \n",
       "434  TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...   \n",
       "435  GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...   \n",
       "436  CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...   \n",
       "437  CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...   \n",
       "438  TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...   \n",
       "\n",
       "                     organism_name    strain  \n",
       "0      Methanobrevibacter millerae       SM9  \n",
       "1      Methanobrevibacter millerae       SM9  \n",
       "2      Methanobrevibacter millerae       SM9  \n",
       "3      Methanobrevibacter millerae       SM9  \n",
       "4      Methanobrevibacter millerae       SM9  \n",
       "..                             ...       ...  \n",
       "434  uncultured Methanobrevibacter  RUG14729  \n",
       "435  uncultured Methanobrevibacter  RUG14729  \n",
       "436  uncultured Methanobrevibacter  RUG14729  \n",
       "437  uncultured Methanobrevibacter  RUG14729  \n",
       "438  uncultured Methanobrevibacter  RUG14729  \n",
       "\n",
       "[439 rows x 8 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_df = protein_df.drop('organism_name_x', axis=1)\n",
    "protein_df = protein_df.drop('strain_x', axis=1)\n",
    "protein_df = protein_df.rename(columns={'organism_name_y': 'organism_name', 'strain_y': 'strain'})\n",
    "protein_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "ee92918d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>orientation</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>strain_x</th>\n",
       "      <th>organism_name_y</th>\n",
       "      <th>strain_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>517037</td>\n",
       "      <td>518035</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>518159</td>\n",
       "      <td>519172</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>hmd</td>\n",
       "      <td>519586</td>\n",
       "      <td>521085</td>\n",
       "      <td>+</td>\n",
       "      <td>ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mer</td>\n",
       "      <td>537737</td>\n",
       "      <td>538699</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001477655.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>768353</td>\n",
       "      <td>768907</td>\n",
       "      <td>-</td>\n",
       "      <td>ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...</td>\n",
       "      <td>Methanobrevibacter-millerae</td>\n",
       "      <td>SM9</td>\n",
       "      <td>Methanobrevibacter millerae</td>\n",
       "      <td>SM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>11100</td>\n",
       "      <td>12500</td>\n",
       "      <td>-</td>\n",
       "      <td>TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fwdG</td>\n",
       "      <td>13073</td>\n",
       "      <td>13303</td>\n",
       "      <td>-</td>\n",
       "      <td>GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhB</td>\n",
       "      <td>18039</td>\n",
       "      <td>19259</td>\n",
       "      <td>-</td>\n",
       "      <td>CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>fdhA</td>\n",
       "      <td>19274</td>\n",
       "      <td>21334</td>\n",
       "      <td>-</td>\n",
       "      <td>CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>GCA_902802065.1</td>\n",
       "      <td>mtrA</td>\n",
       "      <td>16144</td>\n",
       "      <td>16677</td>\n",
       "      <td>-</td>\n",
       "      <td>TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...</td>\n",
       "      <td>uncultured-Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "      <td>uncultured Methanobrevibacter</td>\n",
       "      <td>RUG14729</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>439 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession  gene   start    stop orientation  \\\n",
       "0      GCA_001477655.1   hmd  517037  518035           -   \n",
       "1      GCA_001477655.1   hmd  518159  519172           -   \n",
       "2      GCA_001477655.1   hmd  519586  521085           +   \n",
       "3      GCA_001477655.1   mer  537737  538699           -   \n",
       "4      GCA_001477655.1  mtrA  768353  768907           -   \n",
       "..                 ...   ...     ...     ...         ...   \n",
       "434    GCA_902802065.1  fdhA   11100   12500           -   \n",
       "435    GCA_902802065.1  fwdG   13073   13303           -   \n",
       "436    GCA_902802065.1  fdhB   18039   19259           -   \n",
       "437    GCA_902802065.1  fdhA   19274   21334           -   \n",
       "438    GCA_902802065.1  mtrA   16144   16677           -   \n",
       "\n",
       "                                              Sequence  \\\n",
       "0    TTGATTGATGACATATTAGAAAAAGCAAGCAAAAATAAAAAATTAA...   \n",
       "1    ATGAAAGTAGCAATTTTAGGTGCAGGATGTTACAGAACTCATGCTG...   \n",
       "2    ATGTATGACTTGATAAAAGAAGCAGTTCATGATGATGATGCAGCTA...   \n",
       "3    ATGAAGTTCGGTATAGAATTCGTACCTCAAATACCATTAAATGAAA...   \n",
       "4    ATGGCTGATAAGAAATCTCCTGCAGATGGCTGGCCAGTAATCAGTG...   \n",
       "..                                                 ...   \n",
       "434  TTGCAGGTCTGATGCTGGCTTCTAGTGGTTCCGGCTGCATTTACAG...   \n",
       "435  GGCACTCCTGGAAAGAATGTTGCAATGGGAACCTATGCTCCTATGG...   \n",
       "436  CTTTGAGTTCGTTTTCATTTAATATTTCAGCATCGAACCTTGCACC...   \n",
       "437  CTCAGGACATTGCACCTTGTGATCTGTGTTAAAGAGAGGGGTATAC...   \n",
       "438  TATCCGATTCATCTGTACCGGGAGTATTGATTGCAACCACCCGTCC...   \n",
       "\n",
       "                   organism_name_x  strain_x                organism_name_y  \\\n",
       "0      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "1      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "2      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "3      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "4      Methanobrevibacter-millerae       SM9    Methanobrevibacter millerae   \n",
       "..                             ...       ...                            ...   \n",
       "434  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "435  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "436  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "437  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "438  uncultured-Methanobrevibacter  RUG14729  uncultured Methanobrevibacter   \n",
       "\n",
       "     strain_y  \n",
       "0         SM9  \n",
       "1         SM9  \n",
       "2         SM9  \n",
       "3         SM9  \n",
       "4         SM9  \n",
       "..        ...  \n",
       "434  RUG14729  \n",
       "435  RUG14729  \n",
       "436  RUG14729  \n",
       "437  RUG14729  \n",
       "438  RUG14729  \n",
       "\n",
       "[439 rows x 10 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species = pd.read_csv('/Users/gracedai/Downloads/both.csv')\n",
    "species = species[['assembly_accession', 'organism_name', 'strain']]\n",
    "protein_df = pd.merge(protein_df, species, left_on='assembly_accession', right_on='assembly_accession', how='left')\n",
    "protein_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "6ff06fd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "to_fasta(protein_df, \"all_seq_oriented.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "af5ca005",
   "metadata": {},
   "outputs": [],
   "source": [
    "# use all_seq_oriented.fa for all the sequences of interest \n",
    "\n",
    "gene = \"mcrA\"\n",
    "input_fasta = 'all_seq_oriented.fa'\n",
    "output_fasta = '3NOP_seqs.fa'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "58ab5a48",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_fasta(input_fasta, output_fasta):\n",
    "    with open(input_fasta, 'r') as infile, open(output_fasta, 'w') as outfile:\n",
    "        keep = False\n",
    "        for line in infile:\n",
    "            if line.startswith(\">\"):\n",
    "                keep = gene in line\n",
    "            if keep:\n",
    "                outfile.write(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "48e0cb1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_fasta(input_fasta, output_fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "a13d5d9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "6dddde3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to match with gff files and take the reverse of the sequence, then translate that as well...\n",
    "\n",
    "def to_protein(input_fasta, output_fasta):\n",
    "    with open(input_fasta, \"r\") as infile, open(output_fasta, \"w\") as outfile:\n",
    "        for record in SeqIO.parse(infile, \"fasta\"):\n",
    "            try:\n",
    "                protein_seq = record.seq.translate()\n",
    "                protein_record = SeqRecord(protein_seq, id=record.id)\n",
    "                SeqIO.write(protein_record, outfile, \"fasta\")\n",
    "            except Exception as e:\n",
    "                print(f\"Error translating record {record.id}: {e}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "41cabe8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_fasta_2 = \"3NOP_seqs.fa\"\n",
    "output_fasta_2 = \"3NOP_protein_seqs_mcrA.fa\"\n",
    "\n",
    "to_protein(input_fasta_2, output_fasta_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3609f7b9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
